diff --git a/statsmodels/discrete/conditional_models.py b/statsmodels/discrete/conditional_models.py
--- a/statsmodels/discrete/conditional_models.py
+++ b/statsmodels/discrete/conditional_models.py
@@ -122,7 +122,12 @@ def fit(self,
             disp=disp,
             skip_hessian=skip_hessian)
 
-        crslt = ConditionalResults(self, rslt.params, rslt.cov_params(), 1)
+        if skip_hessian:
+            cov_params = None
+        else:
+            cov_params = rslt.cov_params()
+
+        crslt = ConditionalResults(self, rslt.params, cov_params, 1)
         crslt.method = method
         crslt.nobs = self.nobs
         crslt.n_groups = self._n_groups
@@ -232,8 +237,7 @@ class ConditionalLogit(_ConditionalModel):
 
     def __init__(self, endog, exog, missing='none', **kwargs):
 
-        super().__init__(
-            endog, exog, missing=missing, **kwargs)
+        super().__init__(endog, exog, missing=missing, **kwargs)
 
         if np.any(np.unique(self.endog) != np.r_[0, 1]):
             msg = "endog must be coded as 0, 1"
