diff --git a/pandas/core/indexes/range.py b/pandas/core/indexes/range.py
--- a/pandas/core/indexes/range.py
+++ b/pandas/core/indexes/range.py
@@ -772,8 +772,7 @@ def equals(self, other: object) -> bool:
             return self._range == other._range
         return super().equals(other)
 
-    # error: Signature of "sort_values" incompatible with supertype "Index"
-    @overload  # type: ignore[override]
+    @overload
     def sort_values(
         self,
         *,
@@ -791,7 +790,7 @@ def sort_values(
         ascending: bool = ...,
         na_position: NaPosition = ...,
         key: Callable | None = ...,
-    ) -> tuple[Self, np.ndarray | RangeIndex]: ...
+    ) -> tuple[Self, np.ndarray]: ...
 
     @overload
     def sort_values(
@@ -801,7 +800,7 @@ def sort_values(
         ascending: bool = ...,
         na_position: NaPosition = ...,
         key: Callable | None = ...,
-    ) -> Self | tuple[Self, np.ndarray | RangeIndex]: ...
+    ) -> Self | tuple[Self, np.ndarray]: ...
 
     def sort_values(
         self,
@@ -810,7 +809,7 @@ def sort_values(
         ascending: bool = True,
         na_position: NaPosition = "last",
         key: Callable | None = None,
-    ) -> Self | tuple[Self, np.ndarray | RangeIndex]:
+    ) -> Self | tuple[Self, np.ndarray]:
         if key is not None:
             return super().sort_values(
                 return_indexer=return_indexer,
@@ -831,10 +830,10 @@ def sort_values(
 
         if return_indexer:
             if inverse_indexer:
-                rng = range(len(self) - 1, -1, -1)
+                indexer = np.arange(len(self) - 1, -1, -1, dtype=np.intp)
             else:
-                rng = range(len(self))
-            return sorted_index, RangeIndex(rng)
+                indexer = np.arange(len(self), dtype=np.intp)
+            return sorted_index, indexer
         else:
             return sorted_index
 
