diff --git a/darts/models/forecasting/torch_forecasting_model.py b/darts/models/forecasting/torch_forecasting_model.py
--- a/darts/models/forecasting/torch_forecasting_model.py
+++ b/darts/models/forecasting/torch_forecasting_model.py
@@ -1320,6 +1320,8 @@ def _train(
                 val_dataloaders=val_loader,
                 ckpt_path=ckpt_path,
             )
+        else:
+            trainer.strategy.connect(model)
         self.model = model
         self.trainer = trainer
 
