diff --git a/arrow/parser.py b/arrow/parser.py
index 62a670eb9..243fd1721 100644
--- a/arrow/parser.py
+++ b/arrow/parser.py
@@ -348,10 +348,18 @@ def _parse_token(self, token, value, parts):
             parts["day"] = int(value)
 
         elif token == "dddd":
-            parts["day_of_week"] = self.locale.day_names.index(value) - 1
+            # locale day names are 1-indexed
+            day_of_week = [x.lower() for x in self.locale.day_names].index(
+                value.lower()
+            )
+            parts["day_of_week"] = day_of_week - 1
 
         elif token == "ddd":
-            parts["day_of_week"] = self.locale.day_abbreviations.index(value) - 1
+            # locale day abbreviations are 1-indexed
+            day_of_week = [x.lower() for x in self.locale.day_abbreviations].index(
+                value.lower()
+            )
+            parts["day_of_week"] = day_of_week - 1
 
         elif token.upper() in ["HH", "H"]:
             parts["hour"] = int(value)
diff --git a/tests/test_parser.py b/tests/test_parser.py
index b93034290..9fb4e68f3 100644
--- a/tests/test_parser.py
+++ b/tests/test_parser.py
@@ -659,28 +659,52 @@ def test_parse_ddd_and_dddd(self):
             == expected
         )
 
+    def test_parse_ddd_and_dddd_ignore_case(self):
+        # Regression test for issue #851
+        expected = datetime(2019, 6, 24)
+        assert (
+            self.parser.parse("MONDAY, June 24, 2019", "dddd, MMMM DD, YYYY")
+            == expected
+        )
+
+    def test_parse_ddd_and_dddd_then_format(self):
         # Regression test for issue #446
         arw_formatter = formatter.DateTimeFormatter()
-        arw_formatter.format(self.parser.parse("Mon", "ddd"), "ddd") == "Mon"
-        arw_formatter.format(self.parser.parse("Monday", "dddd"), "dddd") == "Monday"
-        arw_formatter.format(self.parser.parse("Tue", "ddd"), "ddd") == "Tue"
-        arw_formatter.format(self.parser.parse("Tuesday", "dddd"), "dddd") == "Tuesday"
-        arw_formatter.format(self.parser.parse("Wed", "ddd"), "ddd") == "Wed"
-        arw_formatter.format(
-            self.parser.parse("Wednesday", "dddd"), "dddd"
-        ) == "Wednesday"
-        arw_formatter.format(self.parser.parse("Thu", "ddd"), "ddd") == "Thu"
-        arw_formatter.format(
-            self.parser.parse("Thursday", "dddd"), "dddd"
-        ) == "Thursday"
-        arw_formatter.format(self.parser.parse("Fri", "ddd"), "ddd") == "Fri"
-        arw_formatter.format(self.parser.parse("Friday", "dddd"), "dddd") == "Friday"
-        arw_formatter.format(self.parser.parse("Sat", "ddd"), "ddd") == "Sat"
-        arw_formatter.format(
-            self.parser.parse("Saturday", "dddd"), "dddd"
-        ) == "Saturday"
-        arw_formatter.format(self.parser.parse("Sun", "ddd"), "ddd") == "Sun"
-        arw_formatter.format(self.parser.parse("Sunday", "dddd"), "dddd") == "Sunday"
+        assert arw_formatter.format(self.parser.parse("Mon", "ddd"), "ddd") == "Mon"
+        assert (
+            arw_formatter.format(self.parser.parse("Monday", "dddd"), "dddd")
+            == "Monday"
+        )
+        assert arw_formatter.format(self.parser.parse("Tue", "ddd"), "ddd") == "Tue"
+        assert (
+            arw_formatter.format(self.parser.parse("Tuesday", "dddd"), "dddd")
+            == "Tuesday"
+        )
+        assert arw_formatter.format(self.parser.parse("Wed", "ddd"), "ddd") == "Wed"
+        assert (
+            arw_formatter.format(self.parser.parse("Wednesday", "dddd"), "dddd")
+            == "Wednesday"
+        )
+        assert arw_formatter.format(self.parser.parse("Thu", "ddd"), "ddd") == "Thu"
+        assert (
+            arw_formatter.format(self.parser.parse("Thursday", "dddd"), "dddd")
+            == "Thursday"
+        )
+        assert arw_formatter.format(self.parser.parse("Fri", "ddd"), "ddd") == "Fri"
+        assert (
+            arw_formatter.format(self.parser.parse("Friday", "dddd"), "dddd")
+            == "Friday"
+        )
+        assert arw_formatter.format(self.parser.parse("Sat", "ddd"), "ddd") == "Sat"
+        assert (
+            arw_formatter.format(self.parser.parse("Saturday", "dddd"), "dddd")
+            == "Saturday"
+        )
+        assert arw_formatter.format(self.parser.parse("Sun", "ddd"), "ddd") == "Sun"
+        assert (
+            arw_formatter.format(self.parser.parse("Sunday", "dddd"), "dddd")
+            == "Sunday"
+        )
 
     def test_parse_HH_24(self):
         assert self.parser.parse(
