diff --git a/tests/tracking/test_rest_tracking.py b/tests/tracking/test_rest_tracking.py
index 6019a3878d843..bc879c8b71217 100644
--- a/tests/tracking/test_rest_tracking.py
+++ b/tests/tracking/test_rest_tracking.py
@@ -1899,6 +1899,7 @@ def test_create_model_version_with_validation_regex(db_uri: str):
             os.environ.copy()
             | {
                 "MLFLOW_CREATE_MODEL_VERSION_SOURCE_VALIDATION_REGEX": r"^mlflow-artifacts:/.*$",
+                "MLFLOW_SERVER_ENABLE_JOB_EXECUTION": "false",
             }
         ),
     ) as proc:
