diff --git a/statsmodels/tsa/ardl/tests/test_ardl.py b/statsmodels/tsa/ardl/tests/test_ardl.py
index 87fe8a6b7cd..9e6c902ef1c 100644
--- a/statsmodels/tsa/ardl/tests/test_ardl.py
+++ b/statsmodels/tsa/ardl/tests/test_ardl.py
@@ -823,10 +823,15 @@ def test_resids_ardl_uecm():
     assert_allclose(uecm_res.resid, ardl_res.resid)
 
 
-@pytest.mark.parameterize("y_lags",[None, 1, 2])
-@pytest.mark.parameterize("x_lags",[None, 1, 2])
-def test_ardl_trend_ctt(data, y_lags, x_lags):
+@pytest.mark.parametrize("y_lags", [None, 1, 2])
+@pytest.mark.parametrize("x_lags", [None, 1, 2])
+@pytest.mark.parametrize("causal", [True, False])
+def test_ardl_trend_ctt(data, y_lags, x_lags, causal):
     """Test ARDL with trend='ctt'."""
-    res = ARDL(data.y, None, data.x, None, trend="ctt").fit()
-    assert res.params.shape[0] == (3 + y_lags if y_lags else 0 + x_lags if x_lags else 0)
+    res = ARDL(data.y, y_lags, data.x, x_lags, trend="ctt", causal=causal).fit()
+    n_x = data.x.shape[1]
+    n_params = 3
+    n_params += y_lags if y_lags else 0
+    n_params += n_x * (int(not causal) + x_lags) if x_lags else 0
+    assert res.params.shape[0] == n_params
     check_results(res)
