diff --git a/docs/news.rst b/docs/news.rst
index 539fb3d6a..4ac9cc863 100644
--- a/docs/news.rst
+++ b/docs/news.rst
@@ -10,6 +10,12 @@
 Current release
 ---------------
 
+Psycopg 3.3.2
+^^^^^^^^^^^^^
+
+Fix race condition in adapters at startup (:ticket:`#1230`).
+
+
 Psycopg 3.3.1
 ^^^^^^^^^^^^^
 
diff --git a/psycopg/psycopg/_adapters_map.py b/psycopg/psycopg/_adapters_map.py
index da19cbacc..a986d77e4 100644
--- a/psycopg/psycopg/_adapters_map.py
+++ b/psycopg/psycopg/_adapters_map.py
@@ -217,9 +217,10 @@ def get_dumper(self, cls: type, format: PyFormat) -> type[Dumper]:
 
             # If the adapter is not found, look for its name as a string
             fqn = scls.__module__ + "." + scls.__qualname__
-            if fqn in dmap:
+            if (d := dmap.get(fqn)) is not None:
                 # Replace the class name with the class itself
-                d = dmap[scls] = dmap.pop(fqn)
+                dmap[scls] = d
+                dmap.pop(fqn, None)
                 return d
 
         format = PyFormat(format)
