diff --git a/tests/data_context/cloud_data_context/test_include_rendered_content.py b/tests/data_context/cloud_data_context/test_include_rendered_content.py
index 586582a37bf7..5242c5bb4d6b 100644
--- a/tests/data_context/cloud_data_context/test_include_rendered_content.py
+++ b/tests/data_context/cloud_data_context/test_include_rendered_content.py
@@ -39,7 +39,7 @@ def test_cloud_backed_data_context_add_or_update_expectation_suite_include_rende
 
     empty_expectation_suite = ExpectationSuite(expectation_suite_name="test_suite")
     with mock.patch(
-        "great_expectations.data_context.store.gx_cloud_store_backend.GXCloudStoreBackend.has_key"
+        "great_expectations.data_context.store.gx_cloud_store_backend.GXCloudStoreBackend._get"
     ), mock.patch(
         "great_expectations.data_context.store.gx_cloud_store_backend.GXCloudStoreBackend._set",
         return_value=cloud_ref,
