diff --git a/featuretools/entityset/entityset.py b/featuretools/entityset/entityset.py
--- a/featuretools/entityset/entityset.py
+++ b/featuretools/entityset/entityset.py
@@ -1115,9 +1115,9 @@ def add_last_time_indexes(self, updated_dataframes=None):
         child_cols = defaultdict(dict)
         for r in self.relationships:
             children[r._parent_dataframe_name].append(r.child_dataframe)
-            child_cols[r._parent_dataframe_name][
-                r._child_dataframe_name
-            ] = r.child_column
+            child_cols[r._parent_dataframe_name][r._child_dataframe_name] = (
+                r.child_column
+            )
 
         updated_dataframes = updated_dataframes or []
         if updated_dataframes:

diff --git a/featuretools/feature_base/cache.py b/featuretools/feature_base/cache.py
--- a/featuretools/feature_base/cache.py
+++ b/featuretools/feature_base/cache.py
@@ -3,6 +3,7 @@
 
 Custom caching class, currently used for FeatureBase
 """
+
 # needed for defaultdict annotation if < python 3.9
 from __future__ import annotations
 

diff --git a/featuretools/feature_base/features_serializer.py b/featuretools/feature_base/features_serializer.py
--- a/featuretools/feature_base/features_serializer.py
+++ b/featuretools/feature_base/features_serializer.py
@@ -123,12 +123,12 @@ def _feature_definitions(self):
                         # being converted to strings, but integer dict values are not.
                         primitives_dict_key = str(primitive_number)
                         primitive_id_to_key[primitive_id] = primitives_dict_key
-                        self._primitives_dict[
+                        self._primitives_dict[primitives_dict_key] = (
+                            serialize_primitive(primitive)
+                        )
+                        self._features_dict[name]["arguments"]["primitive"] = (
                             primitives_dict_key
-                        ] = serialize_primitive(primitive)
-                        self._features_dict[name]["arguments"][
-                            "primitive"
-                        ] = primitives_dict_key
+                        )
                         primitive_number += 1
                     else:
                         # Primitive we have seen already - use existing primitive_id key
