diff --git a/sktime/clustering/k_medoids.py b/sktime/clustering/k_medoids.py
--- a/sktime/clustering/k_medoids.py
+++ b/sktime/clustering/k_medoids.py
@@ -129,9 +129,14 @@ def _compute_new_cluster_centers(
             curr_indexes = np.where(assignment_indexes == i)[0]
             distance_matrix = np.zeros((len(curr_indexes), len(curr_indexes)))
             for j in range(len(curr_indexes)):
+                curr_j = curr_indexes[j]
                 for k in range(len(curr_indexes)):
-                    distance_matrix[j, k] = self._precomputed_pairwise[j, k]
-            result = medoids(X[curr_indexes], self._precomputed_pairwise)
+                    distance_matrix[j, k] = self._precomputed_pairwise[
+                        curr_j, curr_indexes[k]
+                    ]
+            result = medoids(
+                X[curr_indexes], precomputed_pairwise_distance=distance_matrix
+            )
             if result.shape[0] > 0:
                 new_centers[i, :] = result
         return new_centers
