diff --git a/pydantic-core/src/serializers/type_serializers/model.rs b/pydantic-core/src/serializers/type_serializers/model.rs
index 5c00b101476..9fc6918beba 100644
--- a/pydantic-core/src/serializers/type_serializers/model.rs
+++ b/pydantic-core/src/serializers/type_serializers/model.rs
@@ -15,6 +15,7 @@ use super::{
 };
 use crate::build_tools::py_schema_err;
 use crate::build_tools::{ExtraBehavior, py_schema_error_type};
+use crate::common::missing_sentinel::get_missing_sentinel_object;
 use crate::definitions::DefinitionsBuilder;
 use crate::serializers::SerializationState;
 use crate::serializers::shared::DoSerialize;
@@ -217,18 +218,31 @@ impl ModelSerializer {
         let py: Python<'_> = model.py();
         let mut attrs = model.getattr(intern!(py, "__dict__"))?.cast_into::<PyDict>()?;
 
+        // If excluding unset fields, mask any fields not in `__pydantic_fields_set__` with the
+        // missing sentinel so that the fields serializer will exclude them from the output.
+        // The `GeneralFieldsSerializer` makes sure to exclude the value, so any user-defined
+        // serializer won't be called.
         if extra.exclude_unset {
             let fields_set = model
                 .getattr(intern!(py, "__pydantic_fields_set__"))?
                 .cast_into::<PySet>()?;
 
-            let new_attrs = attrs.copy()?;
-            for key in new_attrs.keys() {
+            // if nothing actually unset, avoid copying the dict
+            let mut new_attrs = None;
+            let mut missing_sentinel = None;
+
+            for key in attrs.keys() {
                 if !fields_set.contains(&key)? {
-                    new_attrs.del_item(key)?;
+                    let missing_sentinel = missing_sentinel.get_or_insert_with(|| get_missing_sentinel_object(py));
+                    let new_attrs = match &new_attrs {
+                        Some(new_attrs) => new_attrs,
+                        None => new_attrs.insert(attrs.copy()?),
+                    };
+                    new_attrs.set_item(key, missing_sentinel.to_owned())?;
                 }
             }
-            attrs = new_attrs;
+
+            attrs = new_attrs.unwrap_or(attrs);
         }
 
         if self.has_extra {
diff --git a/tests/types/test_union.py b/tests/types/test_union.py
index 0e333dcde42..1006689fd34 100644
--- a/tests/types/test_union.py
+++ b/tests/types/test_union.py
@@ -1,4 +1,4 @@
-from typing import ClassVar, Literal, Union
+from typing import ClassVar, Literal, Optional, Union
 
 from typing_extensions import TypedDict
 
@@ -76,6 +76,25 @@ class Container2(pydantic.BaseModel):
     assert MyModel.field_a_serializer_calls == 2
 
 
+def test_exclude_unset_in_nested_union():
+    class Cat(pydantic.BaseModel):
+        type: Literal['cat']
+        color: Optional[str] = None  # field with default
+
+    class Dog(pydantic.BaseModel):
+        type: Literal['dog']
+
+    class Zoo(pydantic.BaseModel):
+        animals: list[Union[Cat, Dog]]
+
+    cat = Cat(type='cat')
+    zoo = Zoo(animals=[cat])
+
+    assert zoo.model_dump() == {'animals': [{'type': 'cat', 'color': None}]}
+    assert cat.model_dump(exclude_unset=True) == {'type': 'cat'}
+    assert zoo.model_dump(exclude_unset=True) == {'animals': [{'type': 'cat'}]}
+
+
 def test_list_union_omit():
     OmitList = list[Union[pydantic.OnErrorOmit[int], pydantic.OnErrorOmit[bool]]]
     ta = pydantic.TypeAdapter(OmitList)
