diff --git a/providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py b/providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py
--- a/providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py
+++ b/providers/amazon/src/airflow/providers/amazon/aws/hooks/base_aws.py
@@ -412,8 +412,15 @@ def _fetch_saml_assertion_using_http_spegno_auth(self, saml_config: dict[str, An
     def _get_web_identity_credential_fetcher(
         self,
     ) -> botocore.credentials.AssumeRoleWithWebIdentityCredentialFetcher:
-        base_session = self.basic_session._session or botocore.session.get_session()
-        client_creator = base_session.create_client
+        session_config = self.config
+        endpoint_url = self.conn.get_service_endpoint_url("sts", sts_connection_assume=True)
+
+        def client_creator(service_name, **kwargs):
+            config = kwargs.pop("config", None)
+            if session_config:
+                config = session_config.merge(config) if config else session_config
+            return self.basic_session.client(service_name, config=config, endpoint_url=endpoint_url, **kwargs)
+
         federation = str(self.extra_config.get("assume_role_with_web_identity_federation"))
 
         web_identity_token_loader = {
