diff --git a/featuretools/primitives/utils.py b/featuretools/primitives/utils.py
--- a/featuretools/primitives/utils.py
+++ b/featuretools/primitives/utils.py
@@ -8,6 +8,7 @@
 from woodwork.column_schema import ColumnSchema
 
 import featuretools
+from featuretools.primitives import NumberOfCommonWords
 from featuretools.primitives.base import (
     AggregationPrimitive,
     PrimitiveBase,
@@ -343,6 +344,8 @@ def serialize_primitive(primitive):
     """build a dictionary with the data necessary to construct the given primitive"""
     args_dict = {name: val for name, val in primitive.get_arguments()}
     cls = type(primitive)
+    if cls == NumberOfCommonWords and "word_set" in args_dict:
+        args_dict["word_set"] = list(args_dict["word_set"])
     return {
         "type": cls.__name__,
         "module": cls.__module__,
@@ -385,6 +388,10 @@ def deserialize_primitive(self, primitive_dict):
                 'Primitive "%s" in module "%s" not found' % (class_name, module_name),
             )
         arguments = primitive_dict["arguments"]
+        if cls == NumberOfCommonWords and "word_set" in arguments:
+            # We converted word_set from a set to a list to make it serializable,
+            # we should convert it back now.
+            arguments["word_set"] = set(arguments["word_set"])
         primitive_instance = cls(**arguments)
 
         return primitive_instance
