diff --git a/mlflow/gateway/cli.py b/mlflow/gateway/cli.py
--- a/mlflow/gateway/cli.py
+++ b/mlflow/gateway/cli.py
@@ -4,7 +4,7 @@
 from mlflow.gateway.config import _validate_config
 from mlflow.gateway.runner import run_app
 from mlflow.telemetry.events import GatewayStartEvent
-from mlflow.telemetry.track import record_usage_event
+from mlflow.telemetry.track import _record_event
 from mlflow.utils.os import is_windows
 
 
@@ -44,8 +44,8 @@ def commands():
     default=2,
     help="The number of workers.",
 )
-@record_usage_event(GatewayStartEvent)
 def start(config_path: str, host: str, port: str, workers: int):
     if is_windows():
         raise click.ClickException("MLflow AI Gateway does not support Windows.")
+    _record_event(GatewayStartEvent, {})
     run_app(config_path=config_path, host=host, port=port, workers=workers)
