diff --git a/.all-contributorsrc b/.all-contributorsrc
index 0b226de6265..bf8791166bd 100644
--- a/.all-contributorsrc
+++ b/.all-contributorsrc
@@ -3260,6 +3260,7 @@
       "avatar_url": "https://avatars.githubusercontent.com/Sohaib-Ahmed21",
       "profile": "https://github.com/Sohaib-Ahmed21",
       "contributions": [
+        "bug",
         "code"
       ]
     },
diff --git a/CONTRIBUTORS.md b/CONTRIBUTORS.md
index d3f122ba716..b3d7dc23b17 100644
--- a/CONTRIBUTORS.md
+++ b/CONTRIBUTORS.md
@@ -351,7 +351,9 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
     <tr>
       <td align="center" valign="top" width="11.11%"><a href="https://github.com/sbuse"><img src="https://avatars.githubusercontent.com/u/24408707?v=4?s=100" width="100px;" alt="Simon B."/><br /><sub><b>Simon B.</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=sbuse" title="Code">💻</a></td>
       <td align="center" valign="top" width="11.11%"><a href="https://github.com/slavik57"><img src="https://avatars.githubusercontent.com/u/6184997?v=4?s=100" width="100px;" alt="Slava Shpitalny"/><br /><sub><b>Slava Shpitalny</b></sub></a><br /><a href="#maintenance-slavik57" title="Maintenance">🚧</a></td>
-      <td align="center" valign="top" width="11.11%"><a href="https://github.com/Sohaib-Ahmed21"><img src="https://avatars.githubusercontent.com/Sohaib-Ahmed21?s=100" width="100px;" alt="Sohaib Ahmed"/><br /><sub><b>Sohaib Ahmed</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=Sohaib-Ahmed21" title="Code">💻</a></td>
+    </tr>
+    <tr>
+      <td align="center" valign="top" width="11.11%"><a href="https://github.com/Sohaib-Ahmed21"><img src="https://avatars.githubusercontent.com/Sohaib-Ahmed21?s=100" width="100px;" alt="Sohaib Ahmed"/><br /><sub><b>Sohaib Ahmed</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=Sohaib-Ahmed21" title="Code">💻</a> <a href="https://github.com/sktime/sktime/issues?q=author%3ASohaib-Ahmed21" title="Bug reports">🐛</a> </td>
       <td align="center" valign="top" width="11.11%"><a href="https://www.linkedin.com/in/solomon-botchway-a1383821b/"><img src="https://avatars.githubusercontent.com/u/62394255?v=4?s=100" width="100px;" alt="Solomon Botchway"/><br /><sub><b>Solomon Botchway</b></sub></a><br /><a href="#maintenance-snnbotchway" title="Maintenance">🚧</a></td>
       <td align="center" valign="top" width="11.11%"><a href="https://github.com/khrapovs"><img src="https://avatars.githubusercontent.com/u/3774663?v=4?s=100" width="100px;" alt="Stanislav Khrapov"/><br /><sub><b>Stanislav Khrapov</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=khrapovs" title="Code">💻</a></td>
       <td align="center" valign="top" width="11.11%"><a href="https://github.com/steenrotsman"><img src="https://avatars.githubusercontent.com/u/78110080?s=400&v=4?s=100" width="100px;" alt="Stijn J. Rotman"/><br /><sub><b>Stijn J. Rotman</b></sub></a><br /><a href="https://github.com/sktime/sktime/commits?author=steenrotsman" title="Code">💻</a> <a href="https://github.com/sktime/sktime/commits?author=steenrotsman" title="Documentation">📖</a></td>
diff --git a/sktime/forecasting/tests/test_all_forecasters.py b/sktime/forecasting/tests/test_all_forecasters.py
index 2b538b6d8ed..3f28d3338fd 100644
--- a/sktime/forecasting/tests/test_all_forecasters.py
+++ b/sktime/forecasting/tests/test_all_forecasters.py
@@ -967,7 +967,7 @@ def test_global_forecasting_multiindex_hier(self, estimator_instance):
         max_prediction_length = 3
         fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)
         X_train, y_train, X_test, y_test = self._multiindex_hier_data(
-            max_prediction_length
+            max_prediction_length, data_length=15
         )
         estimator_instance.fit(y_train, X_train, fh=fh)
 
@@ -986,7 +986,9 @@ def test_global_forecasting_multiindex(self, estimator_instance):
 
         max_prediction_length = 3
         fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)
-        X_train, y_train, X_test, y_test = self._multiindex_data(max_prediction_length)
+        X_train, y_train, X_test, y_test = self._multiindex_data(
+            max_prediction_length, data_length=15
+        )
 
         estimator_instance.fit(y_train, X_train, fh=fh)
 
@@ -1033,7 +1035,7 @@ def test_global_forecasting_hier_series(self, estimator_instance):
         max_prediction_length = 3
         fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)
         X_train, y_train, X_test, y_test = self._multiindex_hier_data(
-            max_prediction_length
+            max_prediction_length, data_length=15
         )
         estimator_instance.fit(y_train, X_train, fh=fh)
 
@@ -1057,7 +1059,9 @@ def test_global_forecasting_different_timestamps(self, estimator_instance):
 
         max_prediction_length = 3
         fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)
-        X_train, y_train, X_test, y_test = self._multiindex_data(max_prediction_length)
+        X_train, y_train, X_test, y_test = self._multiindex_data(
+            max_prediction_length, data_length=15
+        )
         estimator_instance.fit(y_train, X_train, fh=fh)
 
         time_gap = pd.Timedelta(3650, "D")
@@ -1087,7 +1091,9 @@ def test_global_forecasting_no_X(self, estimator_instance):
 
         max_prediction_length = 3
         fh = ForecastingHorizon(range(1, max_prediction_length + 1), is_relative=True)
-        _, y_train, _, y_test = self._multiindex_data(max_prediction_length)
+        _, y_train, _, y_test = self._multiindex_data(
+            max_prediction_length, data_length=15
+        )
 
         estimator_instance.fit(y=y_train, fh=fh)
 
@@ -1130,7 +1136,7 @@ def test_global_predict_interval(
 
         max_prediction_length = np.max(fh_int_oos)
         X_train, y_train, X_test, y_test = self._multiindex_hier_data(
-            max_prediction_length
+            max_prediction_length, data_length=15
         )
 
         estimator_instance.fit(y_train, fh=fh_int_oos, X=X_train)
@@ -1177,7 +1183,7 @@ def test_global_predict_quantiles(self, estimator_instance, fh_int_oos, alpha):
 
         max_prediction_length = np.max(fh_int_oos)
         X_train, y_train, X_test, y_test = self._multiindex_hier_data(
-            max_prediction_length
+            max_prediction_length, data_length=15
         )
 
         estimator_instance.fit(y_train, fh=fh_int_oos, X=X_train)
@@ -1220,7 +1226,7 @@ def test_global_predict_proba(self, estimator_instance, fh_int_oos):
 
         max_prediction_length = np.max(fh_int_oos)
         X_train, y_train, X_test, y_test = self._multiindex_hier_data(
-            max_prediction_length
+            max_prediction_length, data_length=15
         )
 
         estimator_instance.fit(y_train, fh=fh_int_oos, X=X_train)
