diff --git a/src/evaluate/module.py b/src/evaluate/module.py
--- a/src/evaluate/module.py
+++ b/src/evaluate/module.py
@@ -559,7 +559,7 @@ def _infer_feature_from_example(self, example):
                     self._enforce_nested_string_type(features, example)
                     features.encode_example(example)
                     return features
-                except ValueError:
+                except (ValueError, TypeError):
                     continue
         feature_strings = "\n".join([f"Feature option {i}: {feature}" for i, feature in enumerate(self.features)])
         error_msg = (
