diff --git a/arrow/locales.py b/arrow/locales.py
index 6c10134d..0fe28170 100644
--- a/arrow/locales.py
+++ b/arrow/locales.py
@@ -3385,33 +3385,33 @@ class FarsiLocale(Locale):
 
     month_names = [
         "",
-        "January",
-        "February",
-        "March",
-        "April",
-        "May",
-        "June",
-        "July",
-        "August",
-        "September",
-        "October",
-        "November",
-        "December",
+        "ژانویه",
+        "فوریه",
+        "مارس",
+        "آوریل",
+        "مه",
+        "ژوئن",
+        "ژوئیه",
+        "اوت",
+        "سپتامبر",
+        "اکتبر",
+        "نوامبر",
+        "دسامبر",
     ]
     month_abbreviations = [
         "",
-        "Jan",
-        "Feb",
-        "Mar",
-        "Apr",
-        "May",
-        "Jun",
-        "Jul",
-        "Aug",
-        "Sep",
-        "Oct",
-        "Nov",
-        "Dec",
+        "ژانویه",
+        "فوریه",
+        "مارس",
+        "آوریل",
+        "مه",
+        "ژوئن",
+        "ژوئیه",
+        "اوت",
+        "سپتامبر",
+        "اکتبر",
+        "نوامبر",
+        "دسامبر",
     ]
 
     day_names = [
@@ -3424,7 +3424,16 @@ class FarsiLocale(Locale):
         "شنبه",
         "یکشنبه",
     ]
-    day_abbreviations = ["", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]
+    day_abbreviations = [
+        "",
+        "دو شنبه",
+        "سه شنبه",
+        "چهارشنبه",
+        "پنجشنبه",
+        "جمعه",
+        "شنبه",
+        "یکشنبه",
+    ]
 
 
 class HebrewLocale(Locale):
diff --git a/tests/test_locales.py b/tests/test_locales.py
index 3eff7f44..b1916ae0 100644
--- a/tests/test_locales.py
+++ b/tests/test_locales.py
@@ -1463,6 +1463,13 @@ def test_timeframes(self):
         assert self.locale._format_timeframe("months", 2) == "2 ماه"
         assert self.locale._format_timeframe("years", 2) == "2 سال"
 
+    def test_weekday(self):
+        fa = arrow.Arrow(2024, 10, 25, 17, 30, 00)
+        assert self.locale.day_name(fa.isoweekday()) == "جمعه"
+        assert self.locale.day_abbreviation(fa.isoweekday()) == "جمعه"
+        assert self.locale.month_name(fa.month) == "اکتبر"
+        assert self.locale.month_abbreviation(fa.month) == "اکتبر"
+
 
 @pytest.mark.usefixtures("lang_locale")
 class TestNepaliLocale:
