diff --git a/mlflow/langchain/__init__.py b/mlflow/langchain/__init__.py
--- a/mlflow/langchain/__init__.py
+++ b/mlflow/langchain/__init__.py
@@ -17,6 +17,7 @@
 import os
 import shutil
 import types
+from importlib.util import find_spec
 from typing import Any, Dict, List, NamedTuple, Optional, Union
 
 import cloudpickle
@@ -131,13 +132,9 @@ def _get_map_of_special_chain_class_to_loader_arg():
     if version.parse(langchain.__version__) <= version.parse("0.0.246"):
         class_name_to_loader_arg["langchain.chains.SQLDatabaseChain"] = "database"
     else:
-        try:
-            import langchain.experimental
-
+        if find_spec("langchain_experimental"):
+            # Add this entry only if langchain_experimental is installed
             class_name_to_loader_arg["langchain_experimental.sql.SQLDatabaseChain"] = "database"
-        except ImportError:
-            # Users may not have langchain_experimental installed, which is completely normal
-            pass
 
     class_to_loader_arg = {
         _RetrieverChain: "retriever",
