diff --git a/sktime/tests/_config.py b/sktime/tests/_config.py
index c9b31695b87..eaf5a605eeb 100644
--- a/sktime/tests/_config.py
+++ b/sktime/tests/_config.py
@@ -142,6 +142,7 @@
     # stochastic failure of quantile prediction monotonicity, refer to #4420, #4431
     "VAR": ["test_predict_quantiles"],
     "Prophet": ["test_predict_quantiles"],
+    "VECM": ["test_predict_quantiles"],
 }
 
 # We use estimator tags in addition to class hierarchies to further distinguish
