diff --git a/CHANGELOG.md b/CHANGELOG.md
index a4c13e6ab5..3f242fefd5 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,8 @@ but cannot always guarantee backwards compatibility. Changes that may **break co
 
 **Improved**
 
+- `TorchForecastingModel` parameter `torch_metrics` now supports all input metric types from ``torchmetrics.MetricCollection``. Eg. now you can also pass a dictionary or sequence of metrics. [#2958](https://github.com/unit8co/darts/pull/2958) by [CorticallyAI](https://github.com/CorticallyAI).
+
 **Fixed**
 
 - Fixed an issue in `TFTExplainer` where attempting to explain a list of series longer than the model's batch size resulted in an `IndexError`. A more informative error message is now raised instead. [#2957](https://github.com/unit8co/darts/pull/2957) by [Dennis Bader](https://github.com/dennisbader).
diff --git a/darts/models/forecasting/pl_forecasting_module.py b/darts/models/forecasting/pl_forecasting_module.py
index 35bf99f6a3..b45ee0bf0c 100644
--- a/darts/models/forecasting/pl_forecasting_module.py
+++ b/darts/models/forecasting/pl_forecasting_module.py
@@ -85,7 +85,12 @@ def __init__(
         train_sample_shape: Optional[tuple] = None,
         loss_fn: nn.modules.loss._Loss = nn.MSELoss(),
         torch_metrics: Optional[
-            Union[torchmetrics.Metric, torchmetrics.MetricCollection]
+            Union[
+                torchmetrics.Metric,
+                torchmetrics.MetricCollection,
+                Sequence[Union[torchmetrics.Metric, torchmetrics.MetricCollection]],
+                dict[str, Union[torchmetrics.Metric, torchmetrics.MetricCollection]],
+            ]
         ] = None,
         likelihood: Optional[TorchLikelihood] = None,
         optimizer_cls: torch.optim.Optimizer = torch.optim.Adam,
@@ -130,8 +135,8 @@ def __init__(
             This parameter will be ignored for probabilistic models if the ``likelihood`` parameter is specified.
             Default: ``torch.nn.MSELoss()``.
         torch_metrics
-            A torch metric or a ``MetricCollection`` used for evaluation. A full list of available metrics can be found
-            at https://torchmetrics.readthedocs.io/en/latest/. Default: ``None``.
+            A ``torchmetric.Metric`` or a ``MetricCollection`` used for evaluation. A full list of available metrics
+            can be found `here <https://torchmetrics.readthedocs.io/en/latest/>`__. Default: ``None``.
         likelihood
             One of Darts' :meth:`Likelihood <darts.utils.likelihood_models.torch.TorchLikelihood>` models to be used for
             probabilistic forecasts. Default: ``None``.
@@ -799,20 +804,14 @@ def output_chunk_length(self) -> Optional[int]:
 
     @staticmethod
     def configure_torch_metrics(
-        torch_metrics: Union[torchmetrics.Metric, torchmetrics.MetricCollection],
+        torch_metrics: Union[
+            torchmetrics.Metric,
+            torchmetrics.MetricCollection,
+            Sequence[Union[torchmetrics.Metric, torchmetrics.MetricCollection]],
+            dict[str, Union[torchmetrics.Metric, torchmetrics.MetricCollection]],
+        ],
     ) -> torchmetrics.MetricCollection:
         """process the torch_metrics parameter."""
-        if torch_metrics is None:
-            torch_metrics = torchmetrics.MetricCollection([])
-        elif isinstance(torch_metrics, torchmetrics.Metric):
-            torch_metrics = torchmetrics.MetricCollection([torch_metrics])
-        elif isinstance(torch_metrics, torchmetrics.MetricCollection):
-            pass
-        else:
-            raise_log(
-                AttributeError(
-                    "`torch_metrics` only accepts type torchmetrics.Metric or torchmetrics.MetricCollection"
-                ),
-                logger,
-            )
-        return torch_metrics
+        return torchmetrics.MetricCollection(
+            torch_metrics if torch_metrics is not None else []
+        )
diff --git a/darts/tests/models/forecasting/test_torch_forecasting_model.py b/darts/tests/models/forecasting/test_torch_forecasting_model.py
index 2b1a48ef45..d397fa4ff8 100644
--- a/darts/tests/models/forecasting/test_torch_forecasting_model.py
+++ b/darts/tests/models/forecasting/test_torch_forecasting_model.py
@@ -1508,14 +1508,20 @@ def __init__(self, positional_param, named_param=0, *args, **kwargs):
         with pytest.raises(ValueError):
             _ = RnnModelLambda(0, input_chunk_length=12, **invalid_kwargs)
 
-    def test_metrics(self):
-        metric = MeanAbsolutePercentageError()
-        metric_collection = MetricCollection([
-            MeanAbsolutePercentageError(),
-            MeanAbsoluteError(),
-            R2Score(),
-        ])
-
+    @pytest.mark.parametrize(
+        "metric",
+        [
+            MeanAbsolutePercentageError(),  # single metric
+            MetricCollection([
+                MeanAbsolutePercentageError(),
+                MeanAbsoluteError(),
+                R2Score(),
+            ]),  # metric collection
+            {"metric_name": MeanAbsolutePercentageError()},  # dict of metrics
+            [MeanAbsolutePercentageError()],  # sequence of metrics
+        ],
+    )
+    def test_metrics(self, metric):
         model_kwargs = {
             "logger": DummyLogger(),
             "log_every_n_steps": 1,
@@ -1533,18 +1539,6 @@ def test_metrics(self):
         )
         model.fit(self.series)
 
-        # test metric collection
-        model = RNNModel(
-            12,
-            "RNN",
-            10,
-            10,
-            n_epochs=1,
-            torch_metrics=metric_collection,
-            pl_trainer_kwargs=model_kwargs,
-        )
-        model.fit(self.series)
-
         # test multivariate series
         model = RNNModel(
             12,
@@ -1552,7 +1546,7 @@ def test_metrics(self):
             10,
             10,
             n_epochs=1,
-            torch_metrics=metric_collection,
+            torch_metrics=metric,
             pl_trainer_kwargs=model_kwargs,
         )
         model.fit(self.multivariate_series)
@@ -1610,7 +1604,7 @@ def test_metrics_w_likelihood(self):
 
     def test_invalid_metrics(self):
         torch_metrics = ["invalid"]
-        with pytest.raises(AttributeError):
+        with pytest.raises(ValueError):
             model = RNNModel(
                 12,
                 "RNN",
