diff --git a/great_expectations/data_context/data_context/abstract_data_context.py b/great_expectations/data_context/data_context/abstract_data_context.py
index 753502687af7..aa5665c9e6a3 100644
--- a/great_expectations/data_context/data_context/abstract_data_context.py
+++ b/great_expectations/data_context/data_context/abstract_data_context.py
@@ -132,6 +132,7 @@
     send_usage_message,
     usage_statistics_enabled_method,
 )
+from great_expectations.checkpoint import Checkpoint
 
 SQLAlchemyError = sqlalchemy.SQLAlchemyError
 if not SQLAlchemyError:
@@ -143,7 +144,6 @@
 if TYPE_CHECKING:
     from typing_extensions import TypeAlias
 
-    from great_expectations.checkpoint import Checkpoint
     from great_expectations.checkpoint.configurator import ActionDict
     from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
     from great_expectations.core.expectation_configuration import (
@@ -1891,8 +1891,9 @@ def add_checkpoint(  # noqa: PLR0913
             checkpoint=checkpoint,
         )
 
+        result: Checkpoint | CheckpointConfig
         try:
-            return self.checkpoint_store.add_checkpoint(checkpoint)
+            result = self.checkpoint_store.add_checkpoint(checkpoint)
         except gx_exceptions.CheckpointError as e:
             # deprecated-v0.15.50
             warnings.warn(
@@ -1900,7 +1901,15 @@ def add_checkpoint(  # noqa: PLR0913
                 "and will be removed in v0.18. Please use add_or_update_checkpoint instead.",
                 DeprecationWarning,
             )
-            return self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+            result = self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+
+        if isinstance(result, CheckpointConfig):
+            result = Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=result,
+                data_context=self,
+                name=name,
+            )
+        return result
 
     @public_api
     @new_method_or_class(version="0.15.48")
@@ -1916,7 +1925,16 @@ def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
         Returns:
             The updated Checkpoint.
         """
-        return self.checkpoint_store.update_checkpoint(checkpoint)
+        result: Checkpoint | CheckpointConfig = self.checkpoint_store.update_checkpoint(
+            checkpoint
+        )
+        if isinstance(result, CheckpointConfig):
+            result = Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=result,
+                data_context=self,
+                name=result.name,
+            )
+        return result
 
     @overload
     def add_or_update_checkpoint(  # noqa: PLR0913
@@ -2069,7 +2087,16 @@ def add_or_update_checkpoint(  # noqa: C901, PLR0913
             checkpoint=checkpoint,
         )
 
-        return self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+        result: Checkpoint | CheckpointConfig = (
+            self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+        )
+        if isinstance(result, CheckpointConfig):
+            result = Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=result,
+                data_context=self,
+                name=name,
+            )
+        return result
 
     def _resolve_add_checkpoint_args(  # noqa: PLR0913
         self,
diff --git a/great_expectations/data_context/data_context/cloud_data_context.py b/great_expectations/data_context/data_context/cloud_data_context.py
index 082bf783c238..d567de6af70a 100644
--- a/great_expectations/data_context/data_context/cloud_data_context.py
+++ b/great_expectations/data_context/data_context/cloud_data_context.py
@@ -19,6 +19,7 @@
 
 import great_expectations.exceptions as gx_exceptions
 from great_expectations import __version__
+from great_expectations.checkpoint.checkpoint import Checkpoint
 from great_expectations.core import ExpectationSuite
 from great_expectations.core._docs_decorators import public_api
 from great_expectations.core.config_provider import (
@@ -41,6 +42,7 @@
 )
 from great_expectations.data_context.types.base import (
     DEFAULT_USAGE_STATISTICS_URL,
+    CheckpointConfig,
     DataContextConfig,
     DataContextConfigDefaults,
     GXCloudConfig,
@@ -54,7 +56,6 @@
 
 if TYPE_CHECKING:
     from great_expectations.alias_types import PathStr
-    from great_expectations.checkpoint.checkpoint import Checkpoint
     from great_expectations.checkpoint.configurator import ActionDict
     from great_expectations.checkpoint.types.checkpoint_result import CheckpointResult
     from great_expectations.data_context.store.datasource_store import DatasourceStore
@@ -759,8 +760,9 @@ def add_checkpoint(  # noqa: PLR0913
             checkpoint=checkpoint,
         )
 
+        result: Checkpoint | CheckpointConfig
         try:
-            return self.checkpoint_store.add_checkpoint(checkpoint)
+            result = self.checkpoint_store.add_checkpoint(checkpoint)
         except gx_exceptions.CheckpointError as e:
             # deprecated-v0.16.16
             warnings.warn(
@@ -768,7 +770,15 @@ def add_checkpoint(  # noqa: PLR0913
                 "and will be removed in v0.18. Please use add_or_update_checkpoint instead.",
                 DeprecationWarning,
             )
-            return self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+            result = self.checkpoint_store.add_or_update_checkpoint(checkpoint)
+
+        if isinstance(result, CheckpointConfig):
+            result = Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=result,
+                data_context=self,
+                name=name,
+            )
+        return result
 
     def _determine_default_action_list(self) -> Sequence[ActionDict]:
         default_actions = super()._determine_default_action_list()
diff --git a/great_expectations/data_context/store/checkpoint_store.py b/great_expectations/data_context/store/checkpoint_store.py
index 1cfd7499e074..73293f62ba62 100644
--- a/great_expectations/data_context/store/checkpoint_store.py
+++ b/great_expectations/data_context/store/checkpoint_store.py
@@ -155,7 +155,7 @@ def get_checkpoint(
 
         return checkpoint_config
 
-    def add_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+    def add_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint | CheckpointConfig:
         """Persist a stand-alone Checkpoint object.
 
         Args:
@@ -177,7 +177,9 @@ def add_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
                 f"A Checkpoint named {checkpoint.name} already exists."
             )
 
-    def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+    def update_checkpoint(
+        self, checkpoint: Checkpoint
+    ) -> Checkpoint | CheckpointConfig:
         """Use a stand-alone Checkpoint object to update a persisted value.
 
         Args:
@@ -199,7 +201,9 @@ def update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
                 f"Could not find an existing Checkpoint named {checkpoint.name}."
             )
 
-    def add_or_update_checkpoint(self, checkpoint: Checkpoint) -> Checkpoint:
+    def add_or_update_checkpoint(
+        self, checkpoint: Checkpoint
+    ) -> Checkpoint | CheckpointConfig:
         """Use a stand-alone Checkpoint object to either add or update a persisted value.
 
         Args:
@@ -227,18 +231,20 @@ def _persist_checkpoint(
         key: GXCloudIdentifier | ConfigurationIdentifier,
         checkpoint: Checkpoint,
         persistence_fn: Callable,
-    ) -> Checkpoint:
+    ) -> Checkpoint | CheckpointConfig:
         checkpoint_ref = persistence_fn(key=key, value=checkpoint.get_config())
         if isinstance(checkpoint_ref, GXCloudResourceRef):
-            # update parts of config that may have been updated by cloud (ids, default actions, etc.)
-            cloud_id = checkpoint_ref.id
-            checkpoint.config.ge_cloud_id = cloud_id
-            checkpoint.config.validations = checkpoint_ref.response["data"][
-                "attributes"
-            ]["checkpoint_config"].get("validations")
-            checkpoint.config.action_list = checkpoint_ref.response["data"][
-                "attributes"
-            ]["checkpoint_config"].get("action_list")
+            # return CheckpointConfig from cloud POST response to account for any defaults/new ids added in cloud
+            checkpoint_config = checkpoint_ref.response["data"]["attributes"][
+                "checkpoint_config"
+            ]
+            checkpoint_config["ge_cloud_id"] = checkpoint_config.pop("id")
+            return self.deserialize(checkpoint_config)
+        elif self.ge_cloud_mode:
+            # if in cloud mode and checkpoint_ref is not a GXCloudResourceRef, a PUT operation occurred
+            # re-fetch and return CheckpointConfig from cloud to account for any defaults/new ids added in cloud
+            return self.get_checkpoint(name=checkpoint.name, id=None)
+
         return checkpoint
 
     @public_api
diff --git a/great_expectations/data_context/store/gx_cloud_store_backend.py b/great_expectations/data_context/store/gx_cloud_store_backend.py
index 08bdc67072a7..3f542b9c1d2e 100644
--- a/great_expectations/data_context/store/gx_cloud_store_backend.py
+++ b/great_expectations/data_context/store/gx_cloud_store_backend.py
@@ -542,6 +542,25 @@ def remove_key(self, key):
                 f"Unable to delete object in GX Cloud Store Backend: {repr(e)}"
             )
 
+    def _update(self, key, value, **kwargs):
+        existing = self._get(key)
+        if key[1] is None:
+            key = (key[0], existing["data"]["id"], key[2])
+
+        return self.set(key=key, value=value, **kwargs)
+
+    def _add_or_update(self, key, value, **kwargs):
+        try:
+            existing = self._get(key)
+        except StoreBackendError as e:
+            logger.info(f"Could not find object associated with key {key}: {e}")
+            existing = None
+        if existing is not None:
+            id = key[1] if key[1] is not None else existing["data"]["id"]
+            key = (key[0], id, key[2])
+            return self.set(key=key, value=value, **kwargs)
+        return self.add(key=key, value=value, **kwargs)
+
     def _has_key(self, key: Tuple[str, ...]) -> bool:
         try:
             _ = self._get(key)
diff --git a/tests/data_context/cloud_data_context/test_checkpoint_crud.py b/tests/data_context/cloud_data_context/test_checkpoint_crud.py
index 52a1debd6b14..263c51c1343f 100644
--- a/tests/data_context/cloud_data_context/test_checkpoint_crud.py
+++ b/tests/data_context/cloud_data_context/test_checkpoint_crud.py
@@ -5,6 +5,7 @@
 import pandas as pd
 import pytest
 
+from great_expectations.checkpoint import Checkpoint
 from great_expectations.core.batch import RuntimeBatchRequest
 from great_expectations.data_context.cloud_constants import GXCloudRESTResource
 from great_expectations.data_context.data_context.cloud_data_context import (
@@ -18,6 +19,7 @@
     checkpointConfigSchema,
 )
 from great_expectations.data_context.types.resource_identifiers import GXCloudIdentifier
+from great_expectations.exceptions import StoreBackendError
 from great_expectations.util import get_context
 from tests.data_context.conftest import MockResponse
 
@@ -66,7 +68,7 @@ def _mocked_post_response(*args, **kwargs):
                 "data": {
                     "id": checkpoint_id,
                     "attributes": {
-                        "checkpoint_config": checkpoint_config_with_ids,
+                        "checkpoint_config": copy.deepcopy(checkpoint_config_with_ids),
                         "class_name": "Checkpoint",
                         "created_by_id": created_by_id,
                         "default_validation_id": "4cb29141-db66-4dac-a74b-8360779e3da3",
@@ -115,7 +117,7 @@ def _mocked_get_response(*args, **kwargs):
             {
                 "data": {
                     "attributes": {
-                        "checkpoint_config": checkpoint_config_with_ids,
+                        "checkpoint_config": copy.deepcopy(checkpoint_config_with_ids),
                         "class_name": "Checkpoint",
                         "created_by_id": created_by_id,
                         "default_validation_id": "4cb29141-db66-4dac-a74b-8360779e3da3",
@@ -149,7 +151,9 @@ def _mocked_get_by_name_response(*args, **kwargs):
                 "data": [
                     {
                         "attributes": {
-                            "checkpoint_config": checkpoint_config_with_ids,
+                            "checkpoint_config": copy.deepcopy(
+                                checkpoint_config_with_ids
+                            ),
                             "class_name": "Checkpoint",
                             "created_by_id": created_by_id,
                             "default_validation_id": "4cb29141-db66-4dac-a74b-8360779e3da3",
@@ -310,8 +314,8 @@ def test_add_checkpoint_updates_existing_checkpoint_in_cloud_backend(
         assert mock_get.call_count == 3
         mock_get.assert_called_with(
             mock.ANY,  # requests.Session object
-            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
-            params=None,
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints",
+            params={"name": "oss_test_checkpoint"},
         )
 
         expected_checkpoint_config["ge_cloud_id"] = checkpoint_id
@@ -358,9 +362,9 @@ def test_cloud_backed_data_context_add_or_update_checkpoint_adds(
     with mock.patch(
         "requests.Session.post", autospec=True, side_effect=mocked_post_response
     ) as mock_post, mock.patch(
-        "great_expectations.data_context.store.GXCloudStoreBackend._has_key",
+        "great_expectations.data_context.store.GXCloudStoreBackend._get",
         autospec=True,
-        return_value=False,
+        side_effect=StoreBackendError("Does not exist."),
     ) as _:
         checkpoint = context.add_or_update_checkpoint(**checkpoint_config)
 
@@ -400,7 +404,7 @@ def test_cloud_backed_data_context_add_or_update_checkpoint_updates_when_id_pres
     checkpoint_id: str,
     validation_ids: Tuple[str, str],
     checkpoint_config_with_ids: dict,
-    mocked_post_response: Callable[[], MockResponse],
+    mocked_put_response: Callable[[], MockResponse],
     mocked_get_response: Callable[[], MockResponse],
     ge_cloud_base_url: str,
     ge_cloud_organization_id: str,
@@ -414,11 +418,11 @@ def test_cloud_backed_data_context_add_or_update_checkpoint_updates_when_id_pres
     validation_id_1, validation_id_2 = validation_ids
 
     with mock.patch(
-        "requests.Session.put", autospec=True, side_effect=mocked_post_response
+        "requests.Session.put", autospec=True, side_effect=mocked_put_response
     ) as mock_put, mock.patch(
-        "great_expectations.data_context.store.StoreBackend.has_key",
+        "requests.Session.get",
         autospec=True,
-        return_value=True,
+        side_effect=mocked_get_response,
     ) as _:
         checkpoint = context.add_or_update_checkpoint(**checkpoint_config_with_ids)
 
@@ -454,6 +458,197 @@ def test_cloud_backed_data_context_add_or_update_checkpoint_updates_when_id_pres
     assert checkpoint.validations[1]["id"] == validation_id_2
 
 
+@pytest.mark.cloud
+@pytest.mark.integration
+def test_cloud_backed_data_context_add_or_update_checkpoint_updates_when_id_not_present(
+    empty_cloud_data_context: CloudDataContext,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config_with_ids: dict,
+    mocked_put_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    """
+    A Cloud-backed context should save to a Cloud-backed CheckpointStore when calling `add_checkpoint`.
+    When saving, it should use the id from the response to create the checkpoint.
+    """
+    context = empty_cloud_data_context
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.Session.put", autospec=True, side_effect=mocked_put_response
+    ) as mock_put, mock.patch(
+        "requests.Session.get",
+        autospec=True,
+        side_effect=mocked_get_response,
+    ) as _:
+        checkpoint_config = copy.deepcopy(checkpoint_config_with_ids)
+        checkpoint_config.pop("id")
+        checkpoint = context.add_or_update_checkpoint(**checkpoint_config)
+
+        # Round trip through schema to mimic updates made during store serialization process
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(**checkpoint_config)
+        )
+
+        mock_put.assert_called_once_with(
+            mock.ANY,  # requests.Session object
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
+            json={
+                "data": {
+                    "type": "checkpoint",
+                    "attributes": {
+                        "checkpoint_config": expected_checkpoint_config,
+                        "organization_id": ge_cloud_organization_id,
+                    },
+                    "id": checkpoint_id,
+                },
+            },
+        )
+
+    assert checkpoint.config.validations[0]["id"] == validation_id_1
+    assert checkpoint.validations[0]["id"] == validation_id_1
+
+    assert checkpoint.config.validations[1]["id"] == validation_id_2
+    assert checkpoint.validations[1]["id"] == validation_id_2
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+def test_cloud_backed_data_context_update_checkpoint_updates_when_id_present(
+    empty_cloud_data_context: CloudDataContext,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config_with_ids: dict,
+    mocked_put_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    """
+    A Cloud-backed context should update a Cloud-backed CheckpointStore when calling `update_checkpoint`.
+    """
+    context = empty_cloud_data_context
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.Session.put", autospec=True, side_effect=mocked_put_response
+    ) as mock_put, mock.patch(
+        "requests.Session.get",
+        autospec=True,
+        side_effect=mocked_get_response,
+    ) as _:
+        ge_cloud_id = checkpoint_config_with_ids.pop("id")
+        checkpoint = context.update_checkpoint(
+            Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=CheckpointConfig(
+                    ge_cloud_id=ge_cloud_id,
+                    **checkpoint_config_with_ids,
+                ),
+                data_context=context,
+                name=checkpoint_config_with_ids["name"],
+            )
+        )
+
+        # Round trip through schema to mimic updates made during store serialization process
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(ge_cloud_id=ge_cloud_id, **checkpoint_config_with_ids)
+        )
+        expected_checkpoint_config["ge_cloud_id"] = checkpoint_id
+
+        mock_put.assert_called_once_with(
+            mock.ANY,  # requests.Session object
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
+            json={
+                "data": {
+                    "type": "checkpoint",
+                    "attributes": {
+                        "checkpoint_config": expected_checkpoint_config,
+                        "organization_id": ge_cloud_organization_id,
+                    },
+                    "id": checkpoint_id,
+                },
+            },
+        )
+
+    assert checkpoint.ge_cloud_id == checkpoint_id
+    assert checkpoint.config.ge_cloud_id == checkpoint_id
+
+    assert checkpoint.config.validations[0]["id"] == validation_id_1
+    assert checkpoint.validations[0]["id"] == validation_id_1
+
+    assert checkpoint.config.validations[1]["id"] == validation_id_2
+    assert checkpoint.validations[1]["id"] == validation_id_2
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+def test_cloud_backed_data_context_update_checkpoint_updates_when_id_not_present(
+    empty_cloud_data_context: CloudDataContext,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config_with_ids: dict,
+    mocked_put_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    """
+    A Cloud-backed context should update a Cloud-backed CheckpointStore when calling `update_checkpoint`.
+    """
+    context = empty_cloud_data_context
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.Session.put", autospec=True, side_effect=mocked_put_response
+    ) as mock_put, mock.patch(
+        "requests.Session.get",
+        autospec=True,
+        side_effect=mocked_get_response,
+    ) as _:
+        checkpoint_config = copy.deepcopy(checkpoint_config_with_ids)
+        checkpoint_config.pop("id")
+
+        checkpoint = context.update_checkpoint(
+            Checkpoint.instantiate_from_config_with_runtime_args(
+                checkpoint_config=CheckpointConfig(**checkpoint_config),
+                data_context=context,
+                name=checkpoint_config["name"],
+            )
+        )
+
+        # Round trip through schema to mimic updates made during store serialization process
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(**checkpoint_config)
+        )
+
+        mock_put.assert_called_once_with(
+            mock.ANY,  # requests.Session object
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
+            json={
+                "data": {
+                    "type": "checkpoint",
+                    "attributes": {
+                        "checkpoint_config": expected_checkpoint_config,
+                        "organization_id": ge_cloud_organization_id,
+                    },
+                    "id": checkpoint_id,
+                },
+            },
+        )
+
+    assert checkpoint.config.validations[0]["id"] == validation_id_1
+    assert checkpoint.validations[0]["id"] == validation_id_1
+
+    assert checkpoint.config.validations[1]["id"] == validation_id_2
+    assert checkpoint.validations[1]["id"] == validation_id_2
+
+
 @pytest.mark.xfail(
     reason="GX Cloud E2E tests are currently failing due to a schema issue with DataContextVariables; xfailing for purposes of the 0.15.20 release",
     run=True,
diff --git a/tests/data_context/cloud_data_context/test_expectation_suite_crud.py b/tests/data_context/cloud_data_context/test_expectation_suite_crud.py
index 8e5a1a81e9a1..84fc581808c9 100644
--- a/tests/data_context/cloud_data_context/test_expectation_suite_crud.py
+++ b/tests/data_context/cloud_data_context/test_expectation_suite_crud.py
@@ -660,7 +660,7 @@ def test_add_or_update_expectation_suite_updates_existing_obj(
     ) as mock_put:
         context.add_or_update_expectation_suite(expectation_suite=suite)
 
-    mock_get.assert_called_once()  # check if resource exists
+    assert mock_get.call_count == 2  # check if resource exists, get updated resource
     mock_put.assert_called_once()  # persist resource
 
 
