diff --git a/torch/_dynamo/variables/higher_order_ops.py b/torch/_dynamo/variables/higher_order_ops.py
--- a/torch/_dynamo/variables/higher_order_ops.py
+++ b/torch/_dynamo/variables/higher_order_ops.py
@@ -3899,13 +3899,10 @@ def _call_function(
         score_mod_node, score_mod_lifted_args = self.create_wrapped_node(
             tx, query, score_mod, "score_mod"
         )
-        mask_fn = block_mask.items[-1]
-        if mask_fn.is_python_constant():
-            mask_callable = mask_fn.as_python_constant()
-            if mask_callable is None:
-                mask_callable = torch.nn.attention.flex_attention.noop_mask
+        mask_fn = block_mask.items[-1]  # type: ignore[attr-defined]
+        if mask_fn.is_python_constant() and mask_fn.as_python_constant() is None:
             mask_fn = UserFunctionVariable(
-                mask_callable,
+                torch.nn.attention.flex_attention.noop_mask,
                 source=mask_fn.source,
             )
         mask_fn_node, mask_fn_lifted_args = self.create_wrapped_node(
