diff --git a/featuretools/primitives/standard/transform/binary/greater_than_equal_to_scalar.py b/featuretools/primitives/standard/transform/binary/greater_than_equal_to_scalar.py
--- a/featuretools/primitives/standard/transform/binary/greater_than_equal_to_scalar.py
+++ b/featuretools/primitives/standard/transform/binary/greater_than_equal_to_scalar.py
@@ -1,8 +1,5 @@
-import numpy as np
-import pandas as pd
-import pandas.api.types as pdtypes
 from woodwork.column_schema import ColumnSchema
-from woodwork.logical_types import BooleanNullable, Datetime, Ordinal
+from woodwork.logical_types import BooleanNullable
 
 from featuretools.primitives.base.transform_primitive_base import TransformPrimitive
 from featuretools.utils.gen_utils import Library
@@ -23,11 +20,7 @@ class GreaterThanEqualToScalar(TransformPrimitive):
     """
 
     name = "greater_than_equal_to_scalar"
-    input_types = [
-        [ColumnSchema(semantic_tags={"numeric"})],
-        [ColumnSchema(logical_type=Datetime)],
-        [ColumnSchema(logical_type=Ordinal)],
-    ]
+    input_types = [ColumnSchema(semantic_tags={"numeric"})]
     return_type = ColumnSchema(logical_type=BooleanNullable)
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
 
@@ -39,11 +32,6 @@ def __init__(self, value=0):
 
     def get_function(self):
         def greater_than_equal_to_scalar(vals):
-            if (
-                pdtypes.is_categorical_dtype(vals)
-                and self.value not in vals.cat.categories
-            ):
-                return vals.where(pd.isnull, np.nan)
             return vals >= self.value
 
         return greater_than_equal_to_scalar

diff --git a/featuretools/primitives/standard/transform/binary/greater_than_scalar.py b/featuretools/primitives/standard/transform/binary/greater_than_scalar.py
--- a/featuretools/primitives/standard/transform/binary/greater_than_scalar.py
+++ b/featuretools/primitives/standard/transform/binary/greater_than_scalar.py
@@ -1,8 +1,5 @@
-import numpy as np
-import pandas as pd
-import pandas.api.types as pdtypes
 from woodwork.column_schema import ColumnSchema
-from woodwork.logical_types import BooleanNullable, Datetime, Ordinal
+from woodwork.logical_types import BooleanNullable
 
 from featuretools.primitives.base.transform_primitive_base import TransformPrimitive
 from featuretools.utils.gen_utils import Library
@@ -23,11 +20,7 @@ class GreaterThanScalar(TransformPrimitive):
     """
 
     name = "greater_than_scalar"
-    input_types = [
-        [ColumnSchema(semantic_tags={"numeric"})],
-        [ColumnSchema(logical_type=Datetime)],
-        [ColumnSchema(logical_type=Ordinal)],
-    ]
+    input_types = [ColumnSchema(semantic_tags={"numeric"})]
     return_type = ColumnSchema(logical_type=BooleanNullable)
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
 
@@ -37,11 +30,6 @@ def __init__(self, value=0):
 
     def get_function(self):
         def greater_than_scalar(vals):
-            if (
-                pdtypes.is_categorical_dtype(vals)
-                and self.value not in vals.cat.categories
-            ):
-                return vals.where(pd.isnull, np.nan)
             return vals > self.value
 
         return greater_than_scalar

diff --git a/featuretools/primitives/standard/transform/binary/less_than_equal_to_scalar.py b/featuretools/primitives/standard/transform/binary/less_than_equal_to_scalar.py
--- a/featuretools/primitives/standard/transform/binary/less_than_equal_to_scalar.py
+++ b/featuretools/primitives/standard/transform/binary/less_than_equal_to_scalar.py
@@ -1,8 +1,5 @@
-import numpy as np
-import pandas as pd
-import pandas.api.types as pdtypes
 from woodwork.column_schema import ColumnSchema
-from woodwork.logical_types import BooleanNullable, Datetime, Ordinal
+from woodwork.logical_types import BooleanNullable
 
 from featuretools.primitives.base.transform_primitive_base import TransformPrimitive
 from featuretools.utils.gen_utils import Library
@@ -23,11 +20,7 @@ class LessThanEqualToScalar(TransformPrimitive):
     """
 
     name = "less_than_equal_to_scalar"
-    input_types = [
-        [ColumnSchema(semantic_tags={"numeric"})],
-        [ColumnSchema(logical_type=Datetime)],
-        [ColumnSchema(logical_type=Ordinal)],
-    ]
+    input_types = [ColumnSchema(semantic_tags={"numeric"})]
     return_type = ColumnSchema(logical_type=BooleanNullable)
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
 
@@ -39,11 +32,6 @@ def __init__(self, value=0):
 
     def get_function(self):
         def less_than_equal_to_scalar(vals):
-            if (
-                pdtypes.is_categorical_dtype(vals)
-                and self.value not in vals.cat.categories
-            ):
-                return vals.where(pd.isnull, np.nan)
             return vals <= self.value
 
         return less_than_equal_to_scalar

diff --git a/featuretools/primitives/standard/transform/binary/less_than_scalar.py b/featuretools/primitives/standard/transform/binary/less_than_scalar.py
--- a/featuretools/primitives/standard/transform/binary/less_than_scalar.py
+++ b/featuretools/primitives/standard/transform/binary/less_than_scalar.py
@@ -1,8 +1,5 @@
-import numpy as np
-import pandas as pd
-import pandas.api.types as pdtypes
 from woodwork.column_schema import ColumnSchema
-from woodwork.logical_types import BooleanNullable, Datetime, Ordinal
+from woodwork.logical_types import BooleanNullable
 
 from featuretools.primitives.base.transform_primitive_base import TransformPrimitive
 from featuretools.utils.gen_utils import Library
@@ -23,11 +20,7 @@ class LessThanScalar(TransformPrimitive):
     """
 
     name = "less_than_scalar"
-    input_types = [
-        [ColumnSchema(semantic_tags={"numeric"})],
-        [ColumnSchema(logical_type=Datetime)],
-        [ColumnSchema(logical_type=Ordinal)],
-    ]
+    input_types = [ColumnSchema(semantic_tags={"numeric"})]
     return_type = ColumnSchema(logical_type=BooleanNullable)
     compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
 
@@ -37,11 +30,6 @@ def __init__(self, value=0):
 
     def get_function(self):
         def less_than_scalar(vals):
-            if (
-                pdtypes.is_categorical_dtype(vals)
-                and self.value not in vals.cat.categories
-            ):
-                return vals.where(pd.isnull, np.nan)
             return vals < self.value
 
         return less_than_scalar
