diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -7418,6 +7418,15 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         x = cbook._reshape_2D(x, 'x')
         nx = len(x)  # number of datasets
 
+        for arr in x:
+            if len(arr) > 0 and isinstance(
+                arr[0], (datetime.timedelta, np.timedelta64)
+            ):
+                raise TypeError(
+                    "Axes.hist does not currently support timedelta inputs. "
+                    "Convert to numeric values  (e.g., .total_seconds()) first."
+                )
+
         # Process unit information.  _process_unit_info sets the unit and
         # converts the first dataset; then we convert each following dataset
         # one at a time.
