diff --git a/tests/field_defaults/tests.py b/tests/field_defaults/tests.py
index 7f85d946f6bb..c4b978d1693a 100644
--- a/tests/field_defaults/tests.py
+++ b/tests/field_defaults/tests.py
@@ -1,4 +1,4 @@
-from datetime import datetime
+from datetime import UTC, datetime
 from decimal import Decimal
 from math import pi
 
@@ -15,8 +15,7 @@
 )
 from django.db.models.functions import Collate
 from django.db.models.lookups import GreaterThan
-from django.test import SimpleTestCase, TestCase, override_settings, skipUnlessDBFeature
-from django.utils import timezone
+from django.test import SimpleTestCase, TestCase, skipUnlessDBFeature
 
 from .models import (
     Article,
@@ -77,7 +76,6 @@ def test_null_db_default(self):
             self.assertIsNone(obj2.null)
 
     @skipUnlessDBFeature("supports_expression_defaults")
-    @override_settings(USE_TZ=True)
     def test_db_default_function(self):
         m = DBDefaultsFunction.objects.create()
         expected_num_queries = (
@@ -85,7 +83,7 @@ def test_db_default_function(self):
         )
         with self.assertNumQueries(expected_num_queries):
             self.assertAlmostEqual(m.number, pi)
-            self.assertEqual(m.year, timezone.now().year)
+            self.assertEqual(m.year, datetime.now(UTC).year)
             self.assertAlmostEqual(m.added, pi + 4.5)
             self.assertEqual(m.multiple_subfunctions, 4.5)
 
@@ -175,13 +173,12 @@ def test_bulk_create_mixed_db_defaults(self):
         self.assertCountEqual(headlines, ["Default headline", "Something else"])
 
     @skipUnlessDBFeature("supports_expression_defaults")
-    @override_settings(USE_TZ=True)
     def test_bulk_create_mixed_db_defaults_function(self):
         instances = [DBDefaultsFunction(), DBDefaultsFunction(year=2000)]
         DBDefaultsFunction.objects.bulk_create(instances)
 
         years = DBDefaultsFunction.objects.values_list("year", flat=True)
-        self.assertCountEqual(years, [2000, timezone.now().year])
+        self.assertCountEqual(years, [2000, datetime.now(UTC).year])
 
     @skipUnlessDBFeature("supports_expression_defaults")
     def test_full_clean(self):
