diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index fb36c7ab36..1b8fe4e8de 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -7,7 +7,7 @@ repos:
       - id: end-of-file-fixer
       - id: trailing-whitespace
   - repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.12.0
+    rev: v0.14.8
     hooks:
       - id: ruff-check
         args: [ --fix ]
diff --git a/boto3/resources/model.py b/boto3/resources/model.py
index 9d61fea965..fbb053ff96 100644
--- a/boto3/resources/model.py
+++ b/boto3/resources/model.py
@@ -366,7 +366,7 @@ def _load_name_with_category(self, names, name, category, snake_case=True):
             name = xform_name(name)
 
         if name in names:
-            logger.debug(f'Renaming {self.name} {category} {name}')
+            logger.debug('Renaming %s %s %s', self.name, category, name)
             self._renamed[(category, name)] = name + '_' + category
             name += '_' + category
 
diff --git a/boto3/s3/transfer.py b/boto3/s3/transfer.py
index c42ae4b178..65f506f4dc 100644
--- a/boto3/s3/transfer.py
+++ b/boto3/s3/transfer.py
@@ -175,13 +175,17 @@ def create_transfer_manager(client, config, osutil=None):
         crt_transfer_manager = create_crt_transfer_manager(client, config)
         if crt_transfer_manager is not None:
             logger.debug(
-                f"Using CRT client. pid: {getpid()}, thread: {threading.get_ident()}"
+                "Using CRT client. pid: %s, thread: %s",
+                getpid(),
+                threading.get_ident(),
             )
             return crt_transfer_manager
 
     # If we don't resolve something above, fallback to the default.
     logger.debug(
-        f"Using default client. pid: {getpid()}, thread: {threading.get_ident()}"
+        "Using default client. pid: %s, thread: %s",
+        getpid(),
+        threading.get_ident(),
     )
     return _create_default_transfer_manager(client, config, osutil)
 
@@ -214,9 +218,11 @@ def _should_use_crt(config):
         return True
 
     logger.debug(
-        "Opting out of CRT Transfer Manager. Preferred client: "
-        f"{pref_transfer_client}, CRT available: {HAS_CRT}, "
-        f"Instance Optimized: {is_optimized_instance}."
+        "Opting out of CRT Transfer Manager. "
+        "Preferred client: %s, CRT available: %s, Instance Optimized: %s",
+        pref_transfer_client,
+        HAS_CRT,
+        is_optimized_instance,
     )
     return False
 
diff --git a/pyproject.toml b/pyproject.toml
index d13a48b6bd..ccf6b38b02 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -43,7 +43,15 @@ target-version = "py39"
 # Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
 # Unlike Flake8, Ruff doesn't enable pycodestyle warnings (`W`) or
 # McCabe complexity (`C901`) by default.
-select = ["E4", "E7", "E9", "F", "I", "UP"]
+select = [
+    "E4", # pycodestyle
+    "E7", # pycodestyle
+    "E9", # pycodestyle
+    "F", # pyflakes
+    "G", # flake8-logging-format
+    "I", # isort
+    "UP", # pyupgrade
+]
 ignore = []
 
 # Allow fix for all enabled rules (when `--fix`) is provided.
