diff --git a/napari/layers/shapes/shapes.py b/napari/layers/shapes/shapes.py
index 017eab1e31c..5e40be23902 100644
--- a/napari/layers/shapes/shapes.py
+++ b/napari/layers/shapes/shapes.py
@@ -78,6 +78,7 @@
 from napari.utils.events import Event
 from napari.utils.events.custom_types import Array
 from napari.utils.misc import ensure_iterable
+from napari.utils.notifications import show_warning
 from napari.utils.translations import trans
 
 DEFAULT_COLOR_CYCLE = np.array([[1, 0, 1, 1], [0, 1, 0, 1]])
@@ -2642,18 +2643,31 @@ def _finish_drawing(self, event=None) -> None:
                     self._data_view.edit(index, vertices[:-1])
             if self._mode in {Mode.ADD_POLYGON, Mode.ADD_POLYGON_LASSO}:
                 vertices = self._data_view.shapes[index].data
+                if self._mode == Mode.ADD_POLYGON_LASSO:
+                    prev_vertices = len(vertices)
+                    vertices = rdp(
+                        vertices,
+                        epsilon=get_settings().experimental.rdp_epsilon,
+                    )
+                    if len(vertices) <= 3 and prev_vertices > 3:
+                        # https://github.com/napari/napari/issues/7903
+                        show_warning(
+                            trans._(
+                                'Polygons must have three or more vertices. '
+                                'Lasso polygons are simplified using the '
+                                'RDP algorithm, which may cause polygons '
+                                'smaller than RDP epsilon to disappear. If  '
+                                'you face issues drawing small polygons, '
+                                'try reducing napari > Settings > '
+                                'Experimental > RDP epsilon. '
+                            ),
+                        )
                 if len(vertices) <= 3:
                     self._data_view.remove(index)
                     # Clear selected data to prevent issues.
                     # See https://github.com/napari/napari/pull/6912#discussion_r1601169680
                     self.selected_data.clear()
-                elif self._mode == Mode.ADD_POLYGON:
-                    self._data_view.edit(index, vertices[:-1])
                 else:
-                    vertices = rdp(
-                        vertices,
-                        epsilon=get_settings().experimental.rdp_epsilon,
-                    )
                     self._data_view.edit(
                         index,
                         vertices[:-1],
