diff --git a/dask/dataframe/io/parquet/arrow.py b/dask/dataframe/io/parquet/arrow.py
index 458f1e7905f..9c231f9bdfd 100644
--- a/dask/dataframe/io/parquet/arrow.py
+++ b/dask/dataframe/io/parquet/arrow.py
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+import itertools
 import json
 import operator
 import textwrap
@@ -284,9 +285,19 @@ def _get_rg_statistics(row_group, col_names):
     statistics for all columns.
     """
 
-    row_group_schema = {
-        col_name: i for i, col_name in enumerate(row_group.schema.names)
-    }
+    row_group_schema = dict(
+        zip(
+            row_group.schema.names,
+            itertools.accumulate(
+                [
+                    # Need to account for multi-field struct columns
+                    max(row_group.schema.types[i].num_fields, 1)
+                    for i in range(len(row_group.schema.names) - 1)
+                ],
+                initial=0,
+            ),
+        )
+    )
 
     def name_stats(column_name):
         col = row_group.metadata.column(row_group_schema[column_name])
diff --git a/dask/dataframe/io/tests/test_parquet.py b/dask/dataframe/io/tests/test_parquet.py
index 3cba1bda64f..75e445db981 100644
--- a/dask/dataframe/io/tests/test_parquet.py
+++ b/dask/dataframe/io/tests/test_parquet.py
@@ -919,6 +919,48 @@ def test_append_dict_column(tmpdir):
     assert_eq(expect, result)
 
 
+@PYARROW_MARK
+def test_filter_with_struct_column(tmpdir):
+    # Check filtering on a table containing
+    # a multi-field struct column.
+    table = pa.Table.from_arrays(
+        [
+            pa.array(
+                [
+                    {"subfield1": 10, "subfield2": 12},
+                    {"subfield1": 20, "subfield2": 12},
+                    {"subfield1": 30, "subfield2": 12},
+                ]
+            ),
+            pa.array(["aa", "bb", "bb"]),
+        ],
+        schema=pa.schema(
+            [
+                (
+                    "nested_column",
+                    pa.struct([("subfield1", pa.int32()), ("subfield2", pa.int32())]),
+                ),
+                ("id", pa.string()),
+            ]
+        ),
+    )
+    fn = str(tmpdir) + "file.parq"
+    pq.write_table(table, fn)
+
+    pdf = table.to_pandas()
+    assert_eq(dd.read_parquet(fn), pdf)
+    assert_eq(
+        dd.read_parquet(fn, filters=[("id", "not in", ["bb"])]),
+        pdf[pdf["id"] != "bb"],
+        check_index=False,
+    )
+    assert_eq(
+        dd.read_parquet(fn, filters=[("id", "in", ["bb"])]),
+        pdf[pdf["id"] == "bb"],
+        check_index=False,
+    )
+
+
 def test_ordering(tmpdir, write_engine, read_engine):
     tmp = str(tmpdir)
     df = pd.DataFrame(
