diff --git a/great_expectations/data_context/data_context/cloud_data_context.py b/great_expectations/data_context/data_context/cloud_data_context.py
index b8bdcb7fbcf9..489b315aab65 100644
--- a/great_expectations/data_context/data_context/cloud_data_context.py
+++ b/great_expectations/data_context/data_context/cloud_data_context.py
@@ -764,15 +764,16 @@ def add_checkpoint(  # noqa: PLR0913
             checkpoint=checkpoint,
         )
 
-        checkpoint_config = self.checkpoint_store.create(
-            checkpoint_config=checkpoint.config
-        )
-
-        from great_expectations.checkpoint.checkpoint import Checkpoint
-
-        return Checkpoint.instantiate_from_config_with_runtime_args(
-            checkpoint_config=checkpoint_config, data_context=self  # type: ignore[arg-type]
-        )
+        try:
+            return self.checkpoint_store.add_checkpoint(checkpoint)
+        except gx_exceptions.CheckpointError as e:
+            # deprecated-v0.16.16
+            warnings.warn(
+                f"{e.message}; using add_checkpoint to overwrite an existing value is deprecated as of v0.16.16 "
+                "and will be removed in v0.18. Please use add_or_update_checkpoint instead.",
+                DeprecationWarning,
+            )
+            return self.checkpoint_store.add_or_update_checkpoint(checkpoint)
 
     def list_checkpoints(self) -> Union[List[str], List[ConfigurationIdentifier]]:
         return self.checkpoint_store.list_checkpoints(ge_cloud_mode=True)
diff --git a/great_expectations/data_context/store/checkpoint_store.py b/great_expectations/data_context/store/checkpoint_store.py
index 45c579b89f9e..39ae3adb1024 100644
--- a/great_expectations/data_context/store/checkpoint_store.py
+++ b/great_expectations/data_context/store/checkpoint_store.py
@@ -19,7 +19,6 @@
     DataContextConfigDefaults,
 )
 from great_expectations.data_context.types.refs import (
-    GXCloudIDAwareRef,
     GXCloudResourceRef,
 )
 from great_expectations.data_context.types.resource_identifiers import (  # noqa: TCH001
@@ -251,9 +250,16 @@ def _persist_checkpoint(
         persistence_fn: Callable,
     ) -> Checkpoint:
         checkpoint_ref = persistence_fn(key=key, value=checkpoint.get_config())
-        if isinstance(checkpoint_ref, GXCloudIDAwareRef):
+        if isinstance(checkpoint_ref, GXCloudResourceRef):
+            # update parts of config that may have been updated by cloud (ids, default actions, etc.)
             cloud_id = checkpoint_ref.id
             checkpoint.config.ge_cloud_id = cloud_id
+            checkpoint.config.validations = checkpoint_ref.response["data"][
+                "attributes"
+            ]["checkpoint_config"].get("validations")
+            checkpoint.config.action_list = checkpoint_ref.response["data"][
+                "attributes"
+            ]["checkpoint_config"].get("action_list")
         return checkpoint
 
     @public_api
diff --git a/pyproject.toml b/pyproject.toml
index b9e7cb8acc06..4906d52d65da 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -407,6 +407,9 @@ filterwarnings = [
     # Example Actual Warning: (found in test_data_context)
     # UserWarning: Warning. An existing `config_variables.yml` was found here
     'ignore: Warning. An existing `config_variables.yml` was found here:UserWarning',
+    # Example Actual Warning: DeprecationWarning: A Checkpoint named oss_test_checkpoint already exists.;
+    # using add_checkpoint to overwrite an existing value is deprecated as of v0.16.16 and will be removed in v0.18. Please use add_or_update_checkpoint instead.
+    'ignore: A Checkpoint named :DeprecationWarning',
     # --------------------------------------- Great Expectations Deprecation Warnings ----------------------------------
 
     # --------------------------------------- TEMPORARY IGNORES --------------------------------------------------------
diff --git a/tests/data_context/cloud_data_context/test_checkpoint_crud.py b/tests/data_context/cloud_data_context/test_checkpoint_crud.py
index fc6a26564d40..52a1debd6b14 100644
--- a/tests/data_context/cloud_data_context/test_checkpoint_crud.py
+++ b/tests/data_context/cloud_data_context/test_checkpoint_crud.py
@@ -50,15 +50,32 @@ def checkpoint_config_with_ids(
 
 @pytest.fixture
 def mocked_post_response(
-    mock_response_factory: Callable, checkpoint_id: str, validation_ids: Tuple[str, str]
+    mock_response_factory: Callable,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config_with_ids: dict,
 ) -> Callable[[], MockResponse]:
     validation_id_1, validation_id_2 = validation_ids
 
     def _mocked_post_response(*args, **kwargs):
+        created_by_id = "c06ac6a2-52e0-431e-b878-9df624edc8b8"
+        organization_id = "046fe9bc-c85b-4e95-b1af-e4ce36ba5384"
+
         return mock_response_factory(
             {
                 "data": {
                     "id": checkpoint_id,
+                    "attributes": {
+                        "checkpoint_config": checkpoint_config_with_ids,
+                        "class_name": "Checkpoint",
+                        "created_by_id": created_by_id,
+                        "default_validation_id": "4cb29141-db66-4dac-a74b-8360779e3da3",
+                        "description": "My First checkpoint.",
+                        "id": checkpoint_id,
+                        "name": "oss_test_checkpoint",
+                        "organization_id": f"{organization_id}",
+                    },
+                    "type": "checkpoint",
                     "validations": [
                         {"id": validation_id_1},
                         {"id": validation_id_2},
@@ -226,8 +243,10 @@ def test_cloud_backed_data_context_add_checkpoint(
     with mock.patch(
         "requests.Session.post", autospec=True, side_effect=mocked_post_response
     ) as mock_post, mock.patch(
-        "requests.Session.get", autospec=True, side_effect=mocked_get_response
-    ) as mock_get:
+        "great_expectations.data_context.store.GXCloudStoreBackend._has_key",
+        autospec=True,
+        return_value=False,
+    ) as _:
         checkpoint = context.add_checkpoint(**checkpoint_config)
 
         # Round trip through schema to mimic updates made during store serialization process
@@ -249,12 +268,6 @@ def test_cloud_backed_data_context_add_checkpoint(
             },
         )
 
-        mock_get.assert_called_with(
-            mock.ANY,  # requests.Session object
-            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
-            params={"name": checkpoint_config["name"]},
-        )
-
     assert checkpoint.ge_cloud_id == checkpoint_id
     assert checkpoint.config.ge_cloud_id == checkpoint_id
 
@@ -280,15 +293,12 @@ def test_add_checkpoint_updates_existing_checkpoint_in_cloud_backend(
     context = empty_cloud_data_context
 
     with mock.patch(
-        "requests.Session.post", autospec=True, side_effect=mocked_post_response
-    ) as mock_post, mock.patch(
         "requests.Session.put", autospec=True, side_effect=mocked_put_response
     ) as mock_put, mock.patch(
         "requests.Session.get", autospec=True, side_effect=mocked_get_response
     ) as mock_get:
-        checkpoint_1 = context.add_checkpoint(**checkpoint_config)
-        checkpoint_2 = context.add_checkpoint(
-            ge_cloud_id=checkpoint_1.ge_cloud_id, **checkpoint_config
+        checkpoint = context.add_checkpoint(
+            ge_cloud_id=checkpoint_id, **checkpoint_config
         )
 
         # Round trip through schema to mimic updates made during store serialization process
@@ -296,10 +306,20 @@ def test_add_checkpoint_updates_existing_checkpoint_in_cloud_backend(
             CheckpointConfig(**checkpoint_config)
         )
 
-        # Called during creation of `checkpoint_1`
-        mock_post.assert_called_once_with(
+        # Always called by store after POST and PATCH calls
+        assert mock_get.call_count == 3
+        mock_get.assert_called_with(
             mock.ANY,  # requests.Session object
-            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints",
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
+            params=None,
+        )
+
+        expected_checkpoint_config["ge_cloud_id"] = checkpoint_id
+
+        # Called during creation of `checkpoint` (which is `checkpoint_1` but updated)
+        mock_put.assert_called_once_with(
+            mock.ANY,  # requests.Session object
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
             json={
                 "data": {
                     "type": "checkpoint",
@@ -307,21 +327,108 @@ def test_add_checkpoint_updates_existing_checkpoint_in_cloud_backend(
                         "checkpoint_config": expected_checkpoint_config,
                         "organization_id": ge_cloud_organization_id,
                     },
+                    "id": checkpoint_id,
                 },
             },
         )
 
-        # Always called by store after POST and PATCH calls
-        assert mock_get.call_count == 2
-        mock_get.assert_called_with(
+    assert checkpoint.ge_cloud_id == checkpoint_id
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+def test_cloud_backed_data_context_add_or_update_checkpoint_adds(
+    empty_cloud_data_context: CloudDataContext,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config: dict,
+    mocked_post_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    """
+    A Cloud-backed context should save to a Cloud-backed CheckpointStore when calling `add_checkpoint`.
+    When saving, it should use the id from the response to create the checkpoint.
+    """
+    context = empty_cloud_data_context
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.Session.post", autospec=True, side_effect=mocked_post_response
+    ) as mock_post, mock.patch(
+        "great_expectations.data_context.store.GXCloudStoreBackend._has_key",
+        autospec=True,
+        return_value=False,
+    ) as _:
+        checkpoint = context.add_or_update_checkpoint(**checkpoint_config)
+
+        # Round trip through schema to mimic updates made during store serialization process
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(**checkpoint_config)
+        )
+
+        mock_post.assert_called_with(
             mock.ANY,  # requests.Session object
-            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
-            params={"name": checkpoint_config["name"]},
+            f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints",
+            json={
+                "data": {
+                    "type": "checkpoint",
+                    "attributes": {
+                        "checkpoint_config": expected_checkpoint_config,
+                        "organization_id": ge_cloud_organization_id,
+                    },
+                },
+            },
         )
 
+    assert checkpoint.ge_cloud_id == checkpoint_id
+    assert checkpoint.config.ge_cloud_id == checkpoint_id
+
+    assert checkpoint.config.validations[0]["id"] == validation_id_1
+    assert checkpoint.validations[0]["id"] == validation_id_1
+
+    assert checkpoint.config.validations[1]["id"] == validation_id_2
+    assert checkpoint.validations[1]["id"] == validation_id_2
+
+
+@pytest.mark.cloud
+@pytest.mark.integration
+def test_cloud_backed_data_context_add_or_update_checkpoint_updates_when_id_present(
+    empty_cloud_data_context: CloudDataContext,
+    checkpoint_id: str,
+    validation_ids: Tuple[str, str],
+    checkpoint_config_with_ids: dict,
+    mocked_post_response: Callable[[], MockResponse],
+    mocked_get_response: Callable[[], MockResponse],
+    ge_cloud_base_url: str,
+    ge_cloud_organization_id: str,
+) -> None:
+    """
+    A Cloud-backed context should save to a Cloud-backed CheckpointStore when calling `add_checkpoint`.
+    When saving, it should use the id from the response to create the checkpoint.
+    """
+    context = empty_cloud_data_context
+
+    validation_id_1, validation_id_2 = validation_ids
+
+    with mock.patch(
+        "requests.Session.put", autospec=True, side_effect=mocked_post_response
+    ) as mock_put, mock.patch(
+        "great_expectations.data_context.store.StoreBackend.has_key",
+        autospec=True,
+        return_value=True,
+    ) as _:
+        checkpoint = context.add_or_update_checkpoint(**checkpoint_config_with_ids)
+
+        # Round trip through schema to mimic updates made during store serialization process
+        ge_cloud_id = checkpoint_config_with_ids.pop("id")
+        expected_checkpoint_config = checkpointConfigSchema.dump(
+            CheckpointConfig(ge_cloud_id=ge_cloud_id, **checkpoint_config_with_ids)
+        )
         expected_checkpoint_config["ge_cloud_id"] = checkpoint_id
 
-        # Called during creation of `checkpoint_2` (which is `checkpoint_1` but updated)
         mock_put.assert_called_once_with(
             mock.ANY,  # requests.Session object
             f"{ge_cloud_base_url}/organizations/{ge_cloud_organization_id}/checkpoints/{checkpoint_id}",
@@ -337,7 +444,14 @@ def test_add_checkpoint_updates_existing_checkpoint_in_cloud_backend(
             },
         )
 
-    assert checkpoint_1.ge_cloud_id == checkpoint_2.ge_cloud_id
+    assert checkpoint.ge_cloud_id == checkpoint_id
+    assert checkpoint.config.ge_cloud_id == checkpoint_id
+
+    assert checkpoint.config.validations[0]["id"] == validation_id_1
+    assert checkpoint.validations[0]["id"] == validation_id_1
+
+    assert checkpoint.config.validations[1]["id"] == validation_id_2
+    assert checkpoint.validations[1]["id"] == validation_id_2
 
 
 @pytest.mark.xfail(
