diff --git a/src/quart/wrappers/response.py b/src/quart/wrappers/response.py
index 4257e73..136a485 100644
--- a/src/quart/wrappers/response.py
+++ b/src/quart/wrappers/response.py
@@ -77,11 +77,8 @@ async def __aenter__(self) -> DataBody:
     async def __aexit__(self, exc_type: type, exc_value: BaseException, tb: TracebackType) -> None:
         pass
 
-    def __aiter__(self) -> AsyncIterator:
-        async def _aiter() -> AsyncGenerator[bytes, None]:
-            yield self.data[self.begin : self.end]
-
-        return _aiter()
+    def __aiter__(self) -> AsyncIterator[bytes]:
+        return _DataBodyGen(self)
 
     async def make_conditional(self, begin: int, end: int | None) -> int:
         self.begin = begin
@@ -91,6 +88,19 @@ async def make_conditional(self, begin: int, end: int | None) -> int:
         return len(self.data)
 
 
+class _DataBodyGen(AsyncIterator[bytes]):
+    def __init__(self, data_body: DataBody):
+        self._data_body = data_body
+        self._iterated = False
+
+    async def __anext__(self) -> bytes:
+        if self._iterated:
+            raise StopAsyncIteration
+
+        self._iterated = True
+        return self._data_body.data[self._data_body.begin : self._data_body.end]
+
+
 class IterableBody(ResponseBody):
     def __init__(self, iterable: AsyncGenerator[bytes, None] | Iterable) -> None:
         self.iter: AsyncGenerator[bytes, None]
