diff --git a/packages/python/plotly/codegen/__init__.py b/packages/python/plotly/codegen/__init__.py
--- a/packages/python/plotly/codegen/__init__.py
+++ b/packages/python/plotly/codegen/__init__.py
@@ -269,14 +269,14 @@ def perform_codegen():
     optional_figure_widget_import = f"""
 if sys.version_info < (3, 7):
     try:
-        import ipywidgets
-        from distutils.version import LooseVersion
-        if LooseVersion(ipywidgets.__version__) >= LooseVersion('7.0.0'):
+        import ipywidgets as _ipywidgets
+        from distutils.version import LooseVersion as _LooseVersion
+        if _LooseVersion(_ipywidgets.__version__) >= _LooseVersion("7.0.0"):
             from ..graph_objs._figurewidget import FigureWidget
-        del LooseVersion
-        del ipywidgets
-    except ImportError:
-        pass
+        else:
+            raise ImportError()
+    except Exception:
+        from ..missing_ipywidgets import FigureWidget
 else:
     __all__.append("FigureWidget")
     orig_getattr = __getattr__
@@ -285,12 +285,17 @@ def __getattr__(import_name):
             try:
                 import ipywidgets
                 from distutils.version import LooseVersion
-                if LooseVersion(ipywidgets.__version__) >= LooseVersion('7.0.0'):
+
+                if LooseVersion(ipywidgets.__version__) >= LooseVersion("7.0.0"):
                     from ..graph_objs._figurewidget import FigureWidget
+
                     return FigureWidget
-            except ImportError:
-                    pass
-        
+                else:
+                    raise ImportError()
+            except Exception:
+                from ..missing_ipywidgets import FigureWidget
+                return FigureWidget
+
         return orig_getattr(import_name)
 """
     # ### __all__ ###

diff --git a/packages/python/plotly/plotly/graph_objects/__init__.py b/packages/python/plotly/plotly/graph_objects/__init__.py
--- a/packages/python/plotly/plotly/graph_objects/__init__.py
+++ b/packages/python/plotly/plotly/graph_objects/__init__.py
@@ -261,15 +261,15 @@
 
 if sys.version_info < (3, 7):
     try:
-        import ipywidgets
-        from distutils.version import LooseVersion
+        import ipywidgets as _ipywidgets
+        from distutils.version import LooseVersion as _LooseVersion
 
-        if LooseVersion(ipywidgets.__version__) >= LooseVersion("7.0.0"):
+        if _LooseVersion(_ipywidgets.__version__) >= _LooseVersion("7.0.0"):
             from ..graph_objs._figurewidget import FigureWidget
-        del LooseVersion
-        del ipywidgets
-    except ImportError:
-        pass
+        else:
+            raise ImportError()
+    except Exception:
+        from ..missing_ipywidgets import FigureWidget
 else:
     __all__.append("FigureWidget")
     orig_getattr = __getattr__
@@ -284,7 +284,11 @@ def __getattr__(import_name):
                     from ..graph_objs._figurewidget import FigureWidget
 
                     return FigureWidget
-            except ImportError:
-                pass
+                else:
+                    raise ImportError()
+            except Exception:
+                from ..missing_ipywidgets import FigureWidget
+
+                return FigureWidget
 
         return orig_getattr(import_name)

diff --git a/packages/python/plotly/plotly/graph_objs/__init__.py b/packages/python/plotly/plotly/graph_objs/__init__.py
--- a/packages/python/plotly/plotly/graph_objs/__init__.py
+++ b/packages/python/plotly/plotly/graph_objs/__init__.py
@@ -261,15 +261,15 @@
 
 if sys.version_info < (3, 7):
     try:
-        import ipywidgets
-        from distutils.version import LooseVersion
+        import ipywidgets as _ipywidgets
+        from distutils.version import LooseVersion as _LooseVersion
 
-        if LooseVersion(ipywidgets.__version__) >= LooseVersion("7.0.0"):
+        if _LooseVersion(_ipywidgets.__version__) >= _LooseVersion("7.0.0"):
             from ..graph_objs._figurewidget import FigureWidget
-        del LooseVersion
-        del ipywidgets
-    except ImportError:
-        pass
+        else:
+            raise ImportError()
+    except Exception:
+        from ..missing_ipywidgets import FigureWidget
 else:
     __all__.append("FigureWidget")
     orig_getattr = __getattr__
@@ -284,7 +284,11 @@ def __getattr__(import_name):
                     from ..graph_objs._figurewidget import FigureWidget
 
                     return FigureWidget
-            except ImportError:
-                pass
+                else:
+                    raise ImportError()
+            except Exception:
+                from ..missing_ipywidgets import FigureWidget
+
+                return FigureWidget
 
         return orig_getattr(import_name)

diff --git a/packages/python/plotly/plotly/missing_ipywidgets.py b/packages/python/plotly/plotly/missing_ipywidgets.py
--- a/packages/python/plotly/plotly/missing_ipywidgets.py
+++ b/packages/python/plotly/plotly/missing_ipywidgets.py
@@ -0,0 +1,15 @@
+from .basedatatypes import BaseFigure
+
+
+class FigureWidget(BaseFigure):
+    """
+    FigureWidget stand-in for use when ipywidgets is not installed. The only purpose
+    of this class is to provide something to import as
+    `plotly.graph_objs.FigureWidget` when ipywidgets is not installed. This class
+    simply raises an informative error message when the constructor is called
+    """
+
+    def __init__(self, *args, **kwargs):
+        raise ImportError(
+            "Please install ipywidgets>=7.0.0 to use the FigureWidget class"
+        )
