diff --git a/astropy/io/fits/hdu/compressed/_tiled_compression.py b/astropy/io/fits/hdu/compressed/_tiled_compression.py
index cc6909aafbd4..4901b936e832 100644
--- a/astropy/io/fits/hdu/compressed/_tiled_compression.py
+++ b/astropy/io/fits/hdu/compressed/_tiled_compression.py
@@ -359,7 +359,9 @@ def decompress_image_data_section(
     else:
         zzero_column = None
 
-    zblank_header = compressed_header.get("ZBLANK", None)
+    zblank_header = compressed_header.get(
+        "ZBLANK", compressed_header.get("BLANK", None)
+    )
 
     gzip_compressed_data_column = None
     gzip_compressed_data_dtype = None
@@ -465,7 +467,7 @@ def decompress_image_data_section(
             if zblank is not None:
                 if not tile_data.flags.writeable:
                     tile_data = tile_data.copy()
-                tile_data[blank_mask] = np.nan
+                tile_data[blank_mask] = zblank_header if zbitpix > 0 else np.nan
 
         image_data[tile_slices] = tile_data
 
diff --git a/astropy/io/fits/hdu/compressed/compressed.py b/astropy/io/fits/hdu/compressed/compressed.py
index 470e35408bc9..9df53d14422a 100644
--- a/astropy/io/fits/hdu/compressed/compressed.py
+++ b/astropy/io/fits/hdu/compressed/compressed.py
@@ -3,6 +3,7 @@
 import ctypes
 import math
 import time
+import warnings
 from contextlib import suppress
 
 import numpy as np
@@ -21,6 +22,7 @@
 )
 from astropy.utils import lazyproperty
 from astropy.utils.decorators import deprecated_renamed_argument
+from astropy.utils.exceptions import AstropyUserWarning
 
 from .header import (
     CompImageHeader,
@@ -362,6 +364,28 @@ def __init__(
         self._orig_bscale = self._bscale
         self._orig_bitpix = self._bitpix
 
+        if (
+            self._bitpix > 0
+            and "BLANK" not in self._header
+            and "ZBLANK" not in self._header
+        ):
+            # check for column named "ZBLANK"
+            for i in range(1, self._header["TFIELDS"] + 1):
+                if self._header[f"TTYPE{i}"] == "ZBLANK":
+                    # required BLANK keyword is missing
+                    # use most negative value as default
+                    self._header["BLANK"] = -(1 << (self._bitpix - 1))
+                    warnings.warn(
+                        f"Setting default value {self._header['BLANK']} for missing BLANK keyword in compressed extension",
+                        AstropyUserWarning,
+                    )
+                    break
+
+        if self._bitpix > 0:
+            self._blank = self._header.get("BLANK", self._header.get("ZBLANK"))
+        else:
+            self._blank = None
+
     def _remove_unnecessary_default_extnames(self, header):
         """Remove default EXTNAME values if they are unnecessary.
 
@@ -537,15 +561,15 @@ def _update_header_data(
             self.name = name
 
     def _scale_data(self, data):
-        if self._orig_bzero != 0 or self._orig_bscale != 1:
-            new_dtype = self._dtype_for_bitpix()
-            data = np.array(data, dtype=new_dtype)
-
-            if "BLANK" in self._header:
-                blanks = data == np.array(self._header["BLANK"], dtype="int32")
+        if self._orig_bzero != 0 or self._orig_bscale != 1 or self._blank is not None:
+            if self._blank is not None:
+                blanks = data == np.array(self._blank, dtype=data.dtype)
             else:
                 blanks = None
 
+            new_dtype = self._dtype_for_bitpix()
+            data = np.array(data, dtype=new_dtype)
+
             if self._orig_bscale != 1:
                 np.multiply(data, self._orig_bscale, data)
             if self._orig_bzero != 0:
@@ -556,7 +580,10 @@ def _scale_data(self, data):
                 np.add(data, self._orig_bzero, out=data, casting="unsafe")
 
             if blanks is not None:
-                data = np.where(blanks, np.nan, data)
+                # use float32 version of nan to reduce data size for uint conversion
+                # result will still be float64 for larger uint sizes (e.g., uint32)
+                # for float types np.where retains the type of the data array
+                data = np.where(blanks, np.float32(np.nan), data)
 
         return data
 
diff --git a/astropy/io/fits/hdu/compressed/header.py b/astropy/io/fits/hdu/compressed/header.py
index 62cf3792d798..cbc811d67e3d 100644
--- a/astropy/io/fits/hdu/compressed/header.py
+++ b/astropy/io/fits/hdu/compressed/header.py
@@ -467,6 +467,14 @@ def _bintable_header_to_image_header(bintable_header):
         del image_header["PCOUNT"]
         del image_header["GCOUNT"]
 
+    # Fill in BLANK keyword if necessary
+    if (
+        image_header["BITPIX"] > 0
+        and "BLANK" not in image_header
+        and "ZBLANK" in bintable_header
+    ):
+        image_header["BLANK"] = bintable_header["ZBLANK"]
+
     # Look to see if there are any blank cards in the table
     # header.  If there are, there should be the same number
     # of blank cards in the image header.  Add blank cards to
diff --git a/astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py b/astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py
index 9d79ec3a35ae..4985ca1df011 100644
--- a/astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py
+++ b/astropy/io/fits/hdu/compressed/tests/test_tiled_compression.py
@@ -8,6 +8,7 @@
 from astropy.io import fits
 from astropy.io.fits.hdu.compressed._codecs import PLIO1
 from astropy.io.fits.hdu.compressed._compression import CfitsioException
+from astropy.utils.exceptions import AstropyUserWarning
 
 from .conftest import fitsio_param_to_astropy_param
 
@@ -65,6 +66,97 @@ def test_zblank_support(canonical_data_base_path, tmp_path):
         assert_equal(np.round(hdul[1].data), reference)
 
 
+def test_integer_blank_support(canonical_data_base_path, tmp_path):
+    # This uses a test 64x64 section of the m13 image to which three NaN values are added.
+    # It is converted to an integer image with blank values and compressed.
+    # Then 3 variations of the BLANK/ZBLANK specification are created by
+    # editing the header and compressed data table.
+    # When these are read in, the result is a float image that should be the same
+    # as the original (including NaNs for blanks).
+
+    with fits.open(canonical_data_base_path / "m13.fits") as hdul:
+        # pick a 64x64 pixel subset
+        data = hdul[0].data[116:180, 116:180]
+        header = hdul[0].header.copy()
+        del header["CHECKSUM"]
+        del header["DATASUM"]
+
+    # float version of image with NaNs as blanks
+    reference = data.astype(np.float32)
+    reference[[1, 2, 3], [1, 2, 3]] = np.nan
+
+    # set blanks in the int16 image and BLANK keyword in header
+    # choose a blank value that differs from the default
+    blank = -16384
+    data[np.isnan(reference)] = blank
+    header["BLANK"] = blank
+
+    # create the compressed file
+    cfile1 = tmp_path / "compressed_with_blank.fits"
+    hdu = fits.CompImageHDU(
+        data=data, header=header, compression_type="RICE_1", tile_shape=(16, 16)
+    )
+    hdu.writeto(cfile1, overwrite=True, checksum=False)
+
+    # replace BLANK keyword in header with ZBLANK keyword
+    cfile2 = tmp_path / "compressed_with_zblank.fits"
+    with fits.open(cfile1, disable_image_compression=True) as hdul:
+        assert "ZBLANK" not in hdul[1].header
+        hdul[1].header["ZBLANK"] = hdul[1].header["BLANK"]
+        del hdul[1].header["BLANK"]
+        hdul.writeto(cfile2, overwrite=True, checksum=False)
+
+    # replace ZBLANK in header with with ZBLANK in table column
+    # This creates a file structure that is unlikely to be encountered in the wild
+    # but that is apparently allowed by the FITS standard.
+    # Two versions are created, one without the BLANK keyword and one with it.
+    cfile3 = tmp_path / "compressed_with_zblank_column.fits"
+    cfile4 = tmp_path / "compressed_with_zblank_column_and_blank.fits"
+    with fits.open(cfile2, disable_image_compression=True) as hdul:
+        phdu = hdul[0]
+        thdu = hdul[1]
+        orig_table = hdul[1].data
+        orig_cols = orig_table.columns
+        zblank = thdu.header["ZBLANK"]
+        new_cols = fits.ColDefs(
+            [
+                fits.Column(
+                    name="COMPRESSED_DATA",
+                    format="1PB()",
+                    array=thdu.data.field("COMPRESSED_DATA"),
+                ),
+                fits.Column(
+                    name="ZBLANK",
+                    format="I",
+                    array=np.zeros(len(orig_table), dtype=np.int32) + zblank,
+                ),
+            ]
+        )
+        new_thdu = fits.BinTableHDU.from_columns(new_cols, header=thdu.header)
+        del new_thdu.header["ZBLANK"]
+        new_hdul = fits.HDUList([phdu, new_thdu])
+        new_hdul.writeto(cfile3, overwrite=True, checksum=False)
+        new_thdu.header["BLANK"] = blank
+        new_hdul = fits.HDUList([phdu, new_thdu])
+        new_hdul.writeto(cfile4, overwrite=True, checksum=False)
+
+    # now test the 4 files to confirm they all uncompress correctly
+    for filename in (cfile1, cfile2, cfile4):
+        with fits.open(canonical_data_base_path / filename) as hdul:
+            assert_equal(hdul[1].data, reference)
+            # ensure that the uncompressed header is created with BLANK keyword
+            assert hdul[1].header.get("BLANK") == blank
+
+    # this one generates an expected warning
+    with pytest.warns(
+        AstropyUserWarning,
+        match="Setting default value -32768 for missing BLANK keyword in compressed extension",
+    ):
+        with fits.open(cfile3) as hdul:
+            assert_equal(hdul[1].data, reference)
+            assert hdul[1].header.get("BLANK") == -32768
+
+
 @pytest.mark.parametrize(
     ("shape", "tile_shape"),
     (
diff --git a/docs/changes/io.fits/16550.bugfix.rst b/docs/changes/io.fits/16550.bugfix.rst
new file mode 100644
index 000000000000..cff2187839a4
--- /dev/null
+++ b/docs/changes/io.fits/16550.bugfix.rst
@@ -0,0 +1 @@
+Fix a spurious exception when reading integer compressed images with blanks.
