diff --git a/homeassistant/components/homee/config_flow.py b/homeassistant/components/homee/config_flow.py
index 44c9b70953bc9..87b23e1bd6516 100644
--- a/homeassistant/components/homee/config_flow.py
+++ b/homeassistant/components/homee/config_flow.py
@@ -11,7 +11,12 @@
 )
 import voluptuous as vol
 
-from homeassistant.config_entries import SOURCE_USER, ConfigFlow, ConfigFlowResult
+from homeassistant.config_entries import (
+    SOURCE_USER,
+    ConfigEntryState,
+    ConfigFlow,
+    ConfigFlowResult,
+)
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
 from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
 
@@ -113,7 +118,22 @@ async def async_step_zeroconf(
         if discovery_info.ip_address.version == 6:
             return self.async_abort(reason="ipv6_address")
 
-        await self.async_set_unique_id(self._name)
+        # If an already configured homee reports with a second IP, abort.
+        existing_entry = await self.async_set_unique_id(self._name)
+        if (
+            existing_entry
+            and existing_entry.state == ConfigEntryState.LOADED
+            and existing_entry.runtime_data.connected
+            and existing_entry.data[CONF_HOST] != self._host
+        ):
+            _LOGGER.debug(
+                "Aborting config flow for discovered homee with IP %s "
+                "since it is already configured at IP %s",
+                self._host,
+                existing_entry.data[CONF_HOST],
+            )
+            return self.async_abort(reason="2nd_ip_address")
+
         self._abort_if_unique_id_configured(updates={CONF_HOST: self._host})
 
         # Cause an auth-error to see if homee is reachable.
diff --git a/homeassistant/components/homee/strings.json b/homeassistant/components/homee/strings.json
index 9187c9956c701..4bb1339ddff6c 100644
--- a/homeassistant/components/homee/strings.json
+++ b/homeassistant/components/homee/strings.json
@@ -1,6 +1,7 @@
 {
   "config": {
     "abort": {
+      "2nd_ip_address": "Your homee is already connected using another IP address",
       "already_configured": "[%key:common::config_flow::abort::already_configured_device%]",
       "reauth_successful": "[%key:common::config_flow::abort::reauth_successful%]",
       "reconfigure_successful": "[%key:common::config_flow::abort::reconfigure_successful%]",
diff --git a/tests/components/homee/test_config_flow.py b/tests/components/homee/test_config_flow.py
index e56294d9092c8..9bd8231a612f7 100644
--- a/tests/components/homee/test_config_flow.py
+++ b/tests/components/homee/test_config_flow.py
@@ -13,12 +13,13 @@
     RESULT_INVALID_AUTH,
     RESULT_UNKNOWN_ERROR,
 )
-from homeassistant.config_entries import SOURCE_USER
+from homeassistant.config_entries import SOURCE_USER, ConfigEntryState
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResultType
 from homeassistant.helpers.service_info.zeroconf import ZeroconfServiceInfo
 
+from . import setup_integration
 from .conftest import (
     HOMEE_ID,
     HOMEE_IP,
@@ -252,12 +253,27 @@ async def test_zeroconf_confirm_errors(
     assert result["type"] is FlowResultType.CREATE_ENTRY
 
 
+@pytest.mark.parametrize(
+    ("ip", "connected", "reason"),
+    [
+        (HOMEE_IP, True, "already_configured"),
+        ("192.168.1.171", True, "2nd_ip_address"),
+        ("192.168.1.171", False, "already_configured"),
+    ],
+)
+@pytest.mark.usefixtures("mock_setup_entry")
 async def test_zeroconf_already_configured(
     hass: HomeAssistant,
     mock_config_entry: MockConfigEntry,
+    ip: str,
+    connected: bool,
+    reason: str,
 ) -> None:
     """Test zeroconf discovery flow when already configured."""
-    mock_config_entry.add_to_hass(hass)
+    mock_config_entry.runtime_data = AsyncMock()
+    mock_config_entry.runtime_data.connected = connected
+    await setup_integration(hass, mock_config_entry)
+    assert mock_config_entry.state is ConfigEntryState.LOADED
 
     result = await hass.config_entries.flow.async_init(
         DOMAIN,
@@ -266,15 +282,15 @@ async def test_zeroconf_already_configured(
             name=f"homee-{HOMEE_ID}._ssh._tcp.local.",
             type="_ssh._tcp.local.",
             hostname=f"homee-{HOMEE_ID}.local.",
-            ip_address=ip_address(HOMEE_IP),
-            ip_addresses=[ip_address(HOMEE_IP)],
+            ip_address=ip_address(ip),
+            ip_addresses=[ip_address(ip)],
             port=22,
             properties={},
         ),
     )
 
     assert result["type"] is FlowResultType.ABORT
-    assert result["reason"] == "already_configured"
+    assert result["reason"] == reason
 
 
 @pytest.mark.parametrize(
