diff --git a/CHANGES.rst b/CHANGES.rst
index ef3f4f46f..a17191f13 100644
--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -22,6 +22,9 @@ Unreleased
     matching. In ``bind_to_environ``, the ``server_name`` parameter is not used
     if ``host_matching`` is enabled. If ``default_subdomain`` is set, it is used
     if a subdomain could not be determined. :issue:`3005`
+-   If a request object is passed to ``Map.bind_to_environ``, the host is
+    validated against ``request.trusted_hosts``. An invalid host will raise a
+    400 error. :issue:`3007`
 
 
 Version 3.1.5
diff --git a/src/werkzeug/routing/map.py b/src/werkzeug/routing/map.py
index ce4799622..c33ecf241 100644
--- a/src/werkzeug/routing/map.py
+++ b/src/werkzeug/routing/map.py
@@ -8,7 +8,6 @@
 from urllib.parse import urljoin
 from urllib.parse import urlunsplit
 
-from .._internal import _get_environ
 from .._internal import _wsgi_decoding_dance
 from ..datastructures import ImmutableDict
 from ..datastructures import MultiDict
@@ -17,6 +16,7 @@
 from ..exceptions import MethodNotAllowed
 from ..exceptions import NotFound
 from ..urls import _urlencode
+from ..wrappers.request import Request
 from ..wsgi import get_host
 from .converters import DEFAULT_CONVERTERS
 from .exceptions import BuildError
@@ -33,7 +33,6 @@
     from _typeshed.wsgi import WSGIApplication
     from _typeshed.wsgi import WSGIEnvironment
 
-    from ..wrappers.request import Request
     from .converters import BaseConverter
     from .rules import RuleFactory
 
@@ -288,7 +287,9 @@ def bind_to_environ(
         alternate domain name, then :attr:`default_subdomain` is used.
 
         :param environ: The WSGI environ for the request. Can also be a
-            ``Request`` with an ``environ`` attribute.
+            ``Request`` with an ``environ`` attribute; in that case, its
+            :attr:`~.Request.host` is accessed to validate its
+            :attr:`~.Request.trusted_hosts`.
         :param server_name: When subdomain matching is enabled and ``subdomain``
             is not given, the subdomain is determined by removing this
             ``host:port`` as a suffix from the request's ``Host``. If the scheme
@@ -307,6 +308,10 @@ def bind_to_environ(
         .. versionchanged:: 3.2
             ``server_name`` is ignored if ``host_matching`` is enabled.
 
+        .. versionchanged:: 3.2
+            If the ``environ`` argument is a ``Request``, access ``request.host``
+            to validate``request.trusted_hosts``.
+
         .. versionchanged:: 1.0.0
             If ``server_name`` specifies port 443, it will match if the scheme
             is ``https`` and ``Host`` does not specify a port.
@@ -321,8 +326,14 @@ def bind_to_environ(
         .. versionchanged:: 0.5
             Removed the ``calculate_subdomain`` parameter which was not used.
         """
-        env = _get_environ(environ)
-        wsgi_server_name = get_host(env).lower()
+        if isinstance(environ, Request):
+            # accessing request.host triggers trusted_hosts validation
+            wsgi_server_name = environ.host.lower()
+            env = environ.environ
+        else:
+            wsgi_server_name = get_host(environ).lower()
+            env = environ
+
         scheme = env["wsgi.url_scheme"]
         upgrade = any(
             v.strip() == "upgrade"
diff --git a/tests/test_routing.py b/tests/test_routing.py
index 645f5e825..e7c266658 100644
--- a/tests/test_routing.py
+++ b/tests/test_routing.py
@@ -12,7 +12,9 @@
 from werkzeug.datastructures import MultiDict
 from werkzeug.exceptions import MethodNotAllowed
 from werkzeug.exceptions import NotFound
+from werkzeug.exceptions import SecurityError
 from werkzeug.test import create_environ
+from werkzeug.wrappers import Request
 from werkzeug.wrappers import Response
 
 
@@ -1694,3 +1696,17 @@ def test_regex():
     )
     adapter = map_.bind("localhost")
     assert adapter.match("/asdfsa.asdfs") == ("regex", {"value": "asdfsa.asdfs"})
+
+
+def test_bind_trusted_host() -> None:
+    req = Request.from_values(base_url="http://example.test")
+    req.trusted_hosts = ["example.test"]
+    r.Map().bind_to_environ(req)
+
+
+def test_bind_untrusted_host() -> None:
+    req = Request.from_values(base_url="http://example.test")
+    req.trusted_hosts = ["other.test"]
+
+    with pytest.raises(SecurityError):
+        r.Map().bind_to_environ(req)
