diff --git a/plotly/basewidget.py b/plotly/basewidget.py
--- a/plotly/basewidget.py
+++ b/plotly/basewidget.py
@@ -142,6 +142,11 @@ def __init__(
         # views of this widget
         self._view_count = 0
 
+        # Initialize widget layout and data for third-party widget integration
+        # --------------------------------------------------------------------
+        self._widget_layout = deepcopy(self._layout_obj._props)
+        self._widget_data = deepcopy(self._data)
+
     def show(self, *args, **kwargs):
         return self
 
