diff --git a/timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py b/timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py
index d98045a19edb..95e30f8e6da9 100644
--- a/timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py
+++ b/timeseries/src/autogluon/timeseries/models/gluonts/torch/models.py
@@ -17,14 +17,6 @@
 # NOTE: We avoid imports for torch and lightning.pytorch at the top level and hide them inside class methods.
 # This is done to skip these imports during multiprocessing (which may cause bugs)
 
-# FIXME: introduces cpflows dependency. We exclude this model until a future release.
-# from gluonts.torch.model.mqf2 import MQF2MultiHorizonEstimator
-
-# FIXME: DeepNPTS does not implement the GluonTS PyTorch API, and does not use
-# PyTorch Lightning. We exclude this model until a future release.
-# from gluonts.torch.model.deep_npts import DeepNPTSEstimator
-
-
 logger = logging.getLogger(__name__)
 
 
@@ -63,8 +55,8 @@ class DeepARModel(AbstractGluonTSModel):
         (if None, defaults to [min(50, (cat+1)//2) for cat in cardinality])
     max_cat_cardinality : int, default = 100
         Maximum number of dimensions to use when one-hot-encoding categorical known_covariates.
-    distr_output : gluonts.torch.distributions.DistributionOutput, default = StudentTOutput()
-        Distribution to use to evaluate observations and sample predictions
+    distr_output : gluonts.torch.distributions.Output, default = StudentTOutput()
+        Distribution output object that defines how the model output is converted to a forecast, and how the loss is computed.
     scaling: bool, default = True
         If True, mean absolute scaling will be applied to each *context window* during training & prediction.
         Note that this is different from the `target_scaler` that is applied to the *entire time series*.
@@ -120,8 +112,8 @@ class SimpleFeedForwardModel(AbstractGluonTSModel):
         Number of time units that condition the predictions
     hidden_dimensions: List[int], default = [20, 20]
         Size of hidden layers in the feedforward network
-    distr_output : gluonts.torch.distributions.DistributionOutput, default = StudentTOutput()
-        Distribution to fit.
+    distr_output : gluonts.torch.distributions.Output, default = StudentTOutput()
+        Distribution output object that defines how the model output is converted to a forecast, and how the loss is computed.
     batch_normalization : bool, default = False
         Whether to use batch normalization
     mean_scaling : bool, default = True
@@ -169,6 +161,8 @@ class TemporalFusionTransformerModel(AbstractGluonTSModel):
     ----------------
     context_length : int, default = max(64, 2 * prediction_length)
         Number of past values used for prediction.
+    distr_output : gluonts.torch.distributions.Output, default = QuantileOutput()
+        Distribution output object that defines how the model output is converted to a forecast, and how the loss is computed.
     disable_static_features : bool, default = False
         If True, static features won't be used by the model even if they are present in the dataset.
         If False, static features will be used by the model if they are present in the dataset.
@@ -235,6 +229,10 @@ def _get_estimator_init_args(self) -> Dict[str, Any]:
             init_kwargs["past_dynamic_cardinalities"] = self.past_feat_dynamic_cat_cardinality
 
         init_kwargs.setdefault("time_features", get_time_features_for_frequency(self.freq))
+
+        # 'distr_output' and 'quantiles' shouldn't be included at the same time (otherwise an exception will be raised)
+        if "distr_output" in init_kwargs:
+            init_kwargs.pop("quantiles", None)
         return init_kwargs
 
 
@@ -256,8 +254,8 @@ class DLinearModel(AbstractGluonTSModel):
         Number of time units that condition the predictions
     hidden_dimension: int, default = 20
         Size of hidden layers in the feedforward network
-    distr_output : gluonts.torch.distributions.DistributionOutput, default = StudentTOutput()
-        Distribution to fit.
+    distr_output : gluonts.torch.distributions.Output, default = StudentTOutput()
+        Distribution output object that defines how the model output is converted to a forecast, and how the loss is computed.
     scaling : {"mean", "std", None}, default = "mean"
         Scaling applied to each *context window* during training & prediction.
         One of ``"mean"`` (mean absolute scaling), ``"std"`` (standardization), ``None`` (no scaling).
@@ -320,8 +318,8 @@ class PatchTSTModel(AbstractGluonTSModel):
         Number of attention heads in the Transformer encoder which must divide d_model.
     num_encoder_layers : int, default = 2
         Number of layers in the Transformer encoder.
-    distr_output : gluonts.torch.distributions.DistributionOutput, default = StudentTOutput()
-        Distribution to fit.
+    distr_output : gluonts.torch.distributions.Output, default = StudentTOutput()
+        Distribution output object that defines how the model output is converted to a forecast, and how the loss is computed.
     scaling : {"mean", "std", None}, default = "mean"
         Scaling applied to each *context window* during training & prediction.
         One of ``"mean"`` (mean absolute scaling), ``"std"`` (standardization), ``None`` (no scaling).
diff --git a/timeseries/tests/unittests/models/test_gluonts.py b/timeseries/tests/unittests/models/test_gluonts.py
index 7e81f2bb1578..8e256908cee4 100644
--- a/timeseries/tests/unittests/models/test_gluonts.py
+++ b/timeseries/tests/unittests/models/test_gluonts.py
@@ -4,7 +4,9 @@
 import numpy as np
 import pytest
 from gluonts.model.predictor import Predictor as GluonTSPredictor
+from gluonts.torch.distributions import StudentTOutput
 
+from autogluon.timeseries.dataset.ts_dataframe import TimeSeriesDataFrame
 from autogluon.timeseries.models.gluonts import (
     DeepARModel,
     DLinearModel,
@@ -402,3 +404,18 @@ def test_when_model_is_initialized_then_covariate_scaler_is_created(gluonts_mode
     model = gluonts_model_class(freq=df.freq, metadata=metadata)
     model.initialize()
     assert model.covariate_scaler is not None
+
+
+def test_when_distr_output_passed_to_tft_then_model_can_fit_and_predict():
+    data = DUMMY_TS_DATAFRAME.copy()
+    quantile_levels = [0.15, 0.4, 0.94]
+    model = TemporalFusionTransformerModel(
+        freq=data.freq,
+        prediction_length=4,
+        quantile_levels=quantile_levels,
+        hyperparameters={"distr_output": StudentTOutput(), **DUMMY_HYPERPARAMETERS},
+    )
+    model.fit(train_data=data)
+    predictions = model.predict(data)
+    assert isinstance(predictions, TimeSeriesDataFrame)
+    assert set(predictions.columns) == set(["mean"] + [str(q) for q in quantile_levels])
