diff --git a/wandb/cli/beta_sync.py b/wandb/cli/beta_sync.py
--- a/wandb/cli/beta_sync.py
+++ b/wandb/cli/beta_sync.py
@@ -13,7 +13,7 @@
 from wandb.errors import term
 from wandb.proto.wandb_sync_pb2 import ServerSyncResponse
 from wandb.sdk import wandb_setup
-from wandb.sdk.lib import asyncio_compat
+from wandb.sdk.lib import asyncio_compat, wbauth
 from wandb.sdk.lib.printer import Printer, new_printer
 from wandb.sdk.lib.progress import progress_printer
 from wandb.sdk.lib.service.service_connection import ServiceConnection
@@ -88,6 +88,15 @@ def sync(
     if ask_for_confirmation and not term.confirm("Sync the listed runs?"):
         return
 
+    # Authenticate the session. This updates the singleton settings credentials.
+    if not wbauth.authenticate_session(
+        host=singleton.settings.base_url,
+        source="wandb sync",
+        no_offline=True,
+    ):
+        term.termlog("Not authenticated.")
+        return
+
     service = singleton.ensure_service()
     printer = new_printer()
     singleton.asyncer.run(
