diff --git a/timeseries/src/autogluon/timeseries/models/toto/hf_pretrained_model.py b/timeseries/src/autogluon/timeseries/models/toto/hf_pretrained_model.py
index 319b03ffc7ab..9bc3f031242e 100644
--- a/timeseries/src/autogluon/timeseries/models/toto/hf_pretrained_model.py
+++ b/timeseries/src/autogluon/timeseries/models/toto/hf_pretrained_model.py
@@ -1,4 +1,7 @@
+import json
 import logging
+import os
+from pathlib import Path
 
 from transformers import PretrainedConfig, PreTrainedModel
 
@@ -68,12 +71,10 @@ def __init__(self, config: TotoConfig):
             scale_factor_exponent=config.scale_factor_exponent,
             **getattr(config, "extra_kwargs", {}),
         )
-        self._register_load_state_dict_pre_hook(self._remap_state_dict_keys_hook)
         self.post_init()
 
-    def _remap_state_dict_keys_hook(
-        self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs
-    ):
+    @staticmethod
+    def _remap_state_dict_keys(state_dict):
         remap = {
             "mlp.0.w12.weight": "mlp.0.weight",
             "mlp.0.w12.bias": "mlp.0.bias",
@@ -81,6 +82,7 @@ def _remap_state_dict_keys_hook(
             "mlp.0.w3.bias": "mlp.2.bias",
         }
 
+        new_state = {}
         keys_to_remap = []
         for key in list(state_dict.keys()):
             for old, new in remap.items():
@@ -89,11 +91,81 @@ def _remap_state_dict_keys_hook(
                     keys_to_remap.append((key, new_key))
                     break
 
+        new_state = state_dict.copy()
         for old_key, new_key in keys_to_remap:
-            state_dict[new_key] = state_dict.pop(old_key)
+            new_state[new_key] = new_state.pop(old_key)
+
+        return new_state
 
     @classmethod
-    def from_pretrained(cls, model_name_or_path, config=None, torch_dtype=None, device_map=None, **kwargs):
+    def load_from_checkpoint(
+        cls,
+        checkpoint_path,
+        device_map: str = "cpu",
+        strict=True,
+        **model_kwargs,
+    ):
+        """
+        Custom checkpoint loading. Used to load a local
+        safetensors checkpoint with an optional config.json file.
+        """
+        import safetensors.torch as safetorch
+
+        if os.path.isdir(checkpoint_path):
+            safetensors_file = os.path.join(checkpoint_path, "model.safetensors")
+        else:
+            safetensors_file = checkpoint_path
+
+        if os.path.exists(safetensors_file):
+            model_state = safetorch.load_file(safetensors_file, device=device_map)
+        else:
+            raise FileNotFoundError(f"Model checkpoint not found at: {safetensors_file}")
+
+        # Load configuration from config.json if it exists.
+        config_file = os.path.join(checkpoint_path, "config.json")
+        config = {}
+        if os.path.exists(config_file):
+            with open(config_file, "r") as f:
+                config = json.load(f)
+
+        # Merge any extra kwargs into the configuration.
+        config.update(model_kwargs)
+
+        remapped_state_dict = cls._remap_state_dict_keys(model_state)
+
+        instance = cls(**config)
+
+        # Filter out unexpected keys
+        filtered_remapped_state_dict = {
+            k: v
+            for k, v in remapped_state_dict.items()
+            if k in instance.state_dict() and not k.endswith("rotary_emb.freqs")
+        }
+
+        instance.load_state_dict(filtered_remapped_state_dict, strict=strict)
+        instance.to(device_map)  # type: ignore
+
+        return instance
+
+    @classmethod
+    def from_pretrained(
+        cls,
+        *,
+        model_id: str,
+        revision: str | None = None,
+        cache_dir: Path | str | None = None,
+        force_download: bool = False,
+        proxies: dict | None = None,
+        resume_download: bool | None = None,
+        local_files_only: bool = False,
+        token: str | bool | None = None,
+        device_map: str = "cpu",
+        strict: bool = False,
+        **model_kwargs,
+    ):
+        """Load Pytorch pretrained weights and return the loaded model."""
+        from huggingface_hub import constants, hf_hub_download
+
         transformers_logger = logging.getLogger("transformers.modeling_utils")
         original_level = transformers_logger.level
 
@@ -102,13 +174,23 @@ def from_pretrained(cls, model_name_or_path, config=None, torch_dtype=None, devi
             # remapping hook is only called after the initial model loading.
             transformers_logger.setLevel(logging.ERROR)
 
-            # Transformers follows a different load path that does not call load_state_dict hooks when
-            # loading with explicit device maps. Here, we first load the model with no device maps and
-            # move it.
-            model = super().from_pretrained(model_name_or_path, config=config, torch_dtype=torch_dtype, **kwargs)
-            if device_map is not None:
-                model = model.to(device_map)
-
+            if os.path.isdir(model_id):
+                print("Loading weights from local directory")
+                model_file = os.path.join(model_id, constants.SAFETENSORS_SINGLE_FILE)
+                model = cls.load_from_checkpoint(model_file, device_map, strict, **model_kwargs)
+            else:
+                model_file = hf_hub_download(
+                    repo_id=model_id,
+                    filename=constants.SAFETENSORS_SINGLE_FILE,
+                    revision=revision,
+                    cache_dir=cache_dir,
+                    force_download=force_download,
+                    proxies=proxies,
+                    resume_download=resume_download,
+                    token=token,
+                    local_files_only=local_files_only,
+                )
+                model = cls.load_from_checkpoint(model_file, device_map, strict, **model_kwargs)
         finally:
             transformers_logger.setLevel(original_level)
 
diff --git a/timeseries/src/autogluon/timeseries/models/toto/model.py b/timeseries/src/autogluon/timeseries/models/toto/model.py
index 1fc442eeb116..95ceb1468161 100644
--- a/timeseries/src/autogluon/timeseries/models/toto/model.py
+++ b/timeseries/src/autogluon/timeseries/models/toto/model.py
@@ -126,7 +126,7 @@ def load_forecaster(self):
 
         hyperparameters = self.get_hyperparameters()
         pretrained_model = TotoPretrainedModel.from_pretrained(
-            self.model_path,
+            model_id=self.model_path,
             config=TotoConfig.from_pretrained(self.model_path),
             device_map=hyperparameters["device"],
         )
@@ -147,9 +147,21 @@ def _get_default_hyperparameters(self) -> dict:
             "num_samples": 256,
             "device": "cuda",
             "context_length": 4096,
-            "compile_model": True,
+            "compile_model": False,
         }
 
+    def _get_sample_batch_size(self) -> int:
+        num_samples = self.get_hyperparameter("num_samples")
+        batch_size = num_samples
+        while batch_size > 32:
+            for factor in range(2, int(batch_size**0.5) + 1):
+                if batch_size % factor == 0:
+                    batch_size //= factor
+                    break
+            else:  # batch_size is prime
+                return batch_size
+        return batch_size
+
     @property
     def allowed_hyperparameters(self) -> list[str]:
         return super().allowed_hyperparameters + [
@@ -198,6 +210,7 @@ def _predict(
         dataset = TotoInferenceDataset(
             target_df=data.fill_missing_values("auto"),
             max_context_length=hyperparameters["context_length"],
+            target_column=self.target,
         )
         loader = TotoDataLoader(
             dataset,
@@ -214,7 +227,7 @@ def _predict(
                     masked_timeseries,
                     prediction_length=self.prediction_length,
                     num_samples=hyperparameters["num_samples"],
-                    samples_per_batch=32,
+                    samples_per_batch=self._get_sample_batch_size(),
                 )
 
                 batch_means.append(forecast.mean.cpu().numpy())
diff --git a/timeseries/tests/conftest.py b/timeseries/tests/conftest.py
index fc185b6a5df8..c72a59873cc7 100644
--- a/timeseries/tests/conftest.py
+++ b/timeseries/tests/conftest.py
@@ -8,14 +8,15 @@
     "autogluon/chronos-t5-tiny",
     "autogluon/chronos-bolt-tiny",
     "autogluon/chronos-2-small",
+    "Datadog/Toto-Open-Base-1.0",
 ]
 
 
 def download_and_cache_hf_hub_dependencies():
-    from transformers import AutoModel
+    from huggingface_hub import snapshot_download
 
     for dependency in _HF_HUB_DEPENDENCIES:
-        _ = AutoModel.from_pretrained(dependency)
+        _ = snapshot_download(dependency)
 
 
 def pytest_addoption(parser):
diff --git a/timeseries/tests/smoketests/test_all_models.py b/timeseries/tests/smoketests/test_all_models.py
index be026d0b5ee3..2bbf24265a01 100644
--- a/timeseries/tests/smoketests/test_all_models.py
+++ b/timeseries/tests/smoketests/test_all_models.py
@@ -88,6 +88,7 @@ def generate_train_and_test_data(
     "SimpleFeedForward": DUMMY_MODEL_HPARAMS,
     "TemporalFusionTransformer": DUMMY_MODEL_HPARAMS,
     "TiDE": DUMMY_MODEL_HPARAMS,
+    "Toto": {"num_samples": 5},
     "WaveNet": DUMMY_MODEL_HPARAMS,
     "Zero": DUMMY_MODEL_HPARAMS,
     # Override default hyperparameters for faster training
