diff --git a/pandas/tests/frame/test_arithmetic.py b/pandas/tests/frame/test_arithmetic.py
index 862ee82ed304b..5aac1f29fe82c 100644
--- a/pandas/tests/frame/test_arithmetic.py
+++ b/pandas/tests/frame/test_arithmetic.py
@@ -2152,6 +2152,33 @@ def test_arithmetic_multiindex_column_align_with_fillvalue():
     tm.assert_frame_equal(result, expected)
 
 
+def test_arithmetic_multiindex_add_with_mixed_string_datetime_index():
+    # GH#26558
+    df1 = DataFrame(
+        data=[10],
+        index=MultiIndex.from_tuples([("Z", "2019-05-31")]),
+        columns=["A"],
+    )
+    df2 = DataFrame(
+        data=[20],
+        index=MultiIndex.from_tuples([("Z", datetime(2019, 5, 31))]),
+        columns=["A"],
+    )
+
+    expected = DataFrame(
+        data=[20.0, 10.0],
+        index=MultiIndex.from_tuples(
+            [("Z", datetime(2019, 5, 31)), ("Z", "2019-05-31")]
+        ),
+        columns=["A"],
+    )
+
+    with tm.assert_produces_warning(RuntimeWarning, match="are unorderable"):
+        result = df1.add(df2, fill_value=0)
+
+    tm.assert_frame_equal(result, expected)
+
+
 def test_bool_frame_mult_float():
     # GH 18549
     df = DataFrame(True, list("ab"), list("cd"))
