diff --git a/great_expectations/render/renderer/suite_scaffold_notebook_renderer.py b/great_expectations/render/renderer/suite_scaffold_notebook_renderer.py
--- a/great_expectations/render/renderer/suite_scaffold_notebook_renderer.py
+++ b/great_expectations/render/renderer/suite_scaffold_notebook_renderer.py
@@ -130,7 +130,7 @@ def render(self, batch_kwargs=None, **kwargs) -> nbformat.NotebookNode:
 
 This is highly configurable depending on your goals. You can include or exclude
 columns, and include or exclude expectation types (when applicable). [The
-Expectation Glossary](http://docs.greatexpectations.io/en/latest/expectation_glossary.html)
+Expectation Glossary](https://docs.greatexpectations.io/en/latest/reference/glossary_of_expectations.html?utm_source=notebook&utm_medium=scaffold_expectations)
 contains a list of possible expectations."""
         )
         self._add_scaffold_cell()
