diff --git a/statsmodels/genmod/generalized_estimating_equations.py b/statsmodels/genmod/generalized_estimating_equations.py
--- a/statsmodels/genmod/generalized_estimating_equations.py
+++ b/statsmodels/genmod/generalized_estimating_equations.py
@@ -2938,7 +2938,7 @@ def setup_nominal(self, endog, exog, groups, time, offset):
                 u = np.zeros(len(endog_cuts), dtype=np.float64)
                 u[thresh_ix] = 1
                 exog_out[jrow, :] = np.kron(u, exog_row)
-                endog_out[jrow] = int(endog_value == thresh)
+                endog_out[jrow] = int(np.squeeze(endog_value == thresh))
                 groups_out[jrow] = group_value
                 time_out[jrow] = time_value
                 offset_out[jrow] = offset_value

diff --git a/statsmodels/stats/descriptivestats.py b/statsmodels/stats/descriptivestats.py
--- a/statsmodels/stats/descriptivestats.py
+++ b/statsmodels/stats/descriptivestats.py
@@ -408,7 +408,7 @@ def _mode(ser):
             if np.isscalar(mode_res[0]):
                 return float(mode_res[0]), mode_res[1]
             if mode_res[0].shape[0] > 0:
-                return [float(val) for val in mode_res]
+                return [float(np.squeeze(val)) for val in mode_res]
             return np.nan, np.nan
 
         mode_values = df.apply(_mode).T
