diff --git a/test/distributed/tensor/test_tensor_ops.py b/test/distributed/tensor/test_tensor_ops.py
index c59c7aba540d5..692f074a91665 100644
--- a/test/distributed/tensor/test_tensor_ops.py
+++ b/test/distributed/tensor/test_tensor_ops.py
@@ -19,12 +19,7 @@
 from torch.distributed.tensor._sharding_prop import ShardingPropagator
 from torch.distributed.tensor.debug import CommDebugMode
 from torch.testing._internal.common_distributed import skip_if_lt_x_gpu
-from torch.testing._internal.common_utils import (
-    MI200_ARCH,
-    run_tests,
-    serialTest,
-    skipIfRocmArch,
-)
+from torch.testing._internal.common_utils import MI200_ARCH, run_tests, skipIfRocmArch
 from torch.testing._internal.distributed._tensor.common_dtensor import (
     create_local_tensor_test_class,
     DTensorConverter,
@@ -619,7 +614,6 @@ def test_gather(self):
 
     @skipIfRocmArch(MI200_ARCH)
     @with_comms
-    @serialTest()
     def test_index(self):
         meshes = [
             self.build_device_mesh(),  # 1D mesh
