diff --git a/numpy/linalg/tests/test_regression.py b/numpy/linalg/tests/test_regression.py
index 053e7130da63..12d10bc7eecc 100644
--- a/numpy/linalg/tests/test_regression.py
+++ b/numpy/linalg/tests/test_regression.py
@@ -7,6 +7,7 @@
 from numpy import arange, array, dot, float64, linalg, transpose
 from numpy.testing import (
     assert_,
+    assert_almost_equal,
     assert_array_almost_equal,
     assert_array_equal,
     assert_array_less,
@@ -180,3 +181,10 @@ def test_openblas_threading(self):
             if mismatches != 0:
                 assert False, ("unexpected result from matmul, "
                     "probably due to OpenBLAS threading issues")
+
+    def test_norm_linux_arm(self):
+        # gh-30816
+        a = np.arange(20000) / 50000
+        b = a + 1j * np.roll(np.flip(a), 12345)
+        norm = np.linalg.norm(b)
+        assert_almost_equal(norm, 46.18628948075393)
diff --git a/requirements/ci32_requirements.txt b/requirements/ci32_requirements.txt
index a19f6b3b9e7b..02ff529de09c 100644
--- a/requirements/ci32_requirements.txt
+++ b/requirements/ci32_requirements.txt
@@ -1,3 +1,3 @@
 spin
 # Keep this in sync with ci_requirements.txt
-scipy-openblas32==0.3.31.126.1
+scipy-openblas32==0.3.31.188.0
diff --git a/requirements/ci_requirements.txt b/requirements/ci_requirements.txt
index f200891aa560..397fa703e28d 100644
--- a/requirements/ci_requirements.txt
+++ b/requirements/ci_requirements.txt
@@ -1,4 +1,4 @@
 spin
 # Keep this in sync with ci32_requirements.txt
-scipy-openblas32==0.3.31.126.1
-scipy-openblas64==0.3.31.126.1
+scipy-openblas32==0.3.31.188.0
+scipy-openblas64==0.3.31.188.0
