diff --git a/pymc/backends/arviz.py b/pymc/backends/arviz.py
--- a/pymc/backends/arviz.py
+++ b/pymc/backends/arviz.py
@@ -618,11 +618,13 @@ def dataset_to_point_list(
     for vn in var_names:
         if not isinstance(vn, str):
             raise ValueError(f"Variable names must be str, but dataset key {vn} is a {type(vn)}.")
+
     num_sample_dims = len(sample_dims)
     stacked_dims = {dim_name: ds[var_names[0]][dim_name] for dim_name in sample_dims}
     transposed_dict = {vn: da.transpose(*sample_dims, ...) for vn, da in ds.items()}
+    stacked_size = np.prod(transposed_dict[var_names[0]].shape[:num_sample_dims], dtype=int)
     stacked_dict = {
-        vn: da.values.reshape((-1, *da.shape[num_sample_dims:]))
+        vn: da.values.reshape((stacked_size, *da.shape[num_sample_dims:]))
         for vn, da in transposed_dict.items()
     }
     points = [
