diff --git a/tests/datasource/fluent/integration/test_sql_datasources.py b/tests/datasource/fluent/integration/test_sql_datasources.py
index 903331a48029..36d1795d806d 100644
--- a/tests/datasource/fluent/integration/test_sql_datasources.py
+++ b/tests/datasource/fluent/integration/test_sql_datasources.py
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import logging
+import uuid
 from pprint import pformat as pf
 from typing import Final, Generator, Literal, Protocol
 
@@ -96,6 +97,11 @@ def __call__(
         ...
 
 
+def get_random_identifier_name() -> str:
+    guid = uuid.uuid4()
+    return f"i{guid.hex}"
+
+
 @pytest.fixture(scope="function")
 def capture_engine_logs(caplog: pytest.LogCaptureFixture) -> pytest.LogCaptureFixture:
     """Capture SQLAlchemy engine logs and display them if the test fails."""
@@ -158,6 +164,8 @@ def _table_factory(
                 schema = table["schema"]
                 qualified_table_name = f"{schema}.{name}" if schema else name
                 conn.execute(TextClause(f"DROP TABLE IF EXISTS {qualified_table_name}"))
+            if schema:
+                conn.execute(TextClause(f"DROP SCHEMA IF EXISTS {schema}"))
             if SQLA_VERSION >= Version("2.0"):
                 conn.commit()
 
@@ -232,9 +240,7 @@ def test_postgres(
         if not table_name:
             pytest.skip(f"no '{asset_name}' table_name for databricks")
         # create table
-        table_factory(
-            engine=postgres_ds.get_engine(), table_names={table_name}, schema="public"
-        )
+        table_factory(engine=postgres_ds.get_engine(), table_names={table_name})
 
         table_names: list[str] = inspect(postgres_ds.get_engine()).get_table_names()
         print(f"postgres tables:\n{pf(table_names)}))")
@@ -271,7 +277,11 @@ def test_snowflake(
         if not table_name:
             pytest.skip(f"no '{asset_name}' table_name for databricks")
         # create table
-        table_factory(engine=snowflake_ds.get_engine(), table_names={table_name})
+        table_factory(
+            engine=snowflake_ds.get_engine(),
+            table_names={table_name},
+            schema=get_random_identifier_name(),
+        )
 
         table_names: list[str] = inspect(snowflake_ds.get_engine()).get_table_names()
         print(f"snowflake tables:\n{pf(table_names)}))")
@@ -279,11 +289,13 @@ def test_snowflake(
         snowflake_ds.add_table_asset(asset_name, table_name=table_name)
 
     @pytest.mark.parametrize(
-        "datasource_type",
+        "datasource_type,schema",
         [
-            param("trino", marks=[pytest.mark.trino]),
-            param("postgres", marks=[pytest.mark.postgresql]),
-            param("snowflake", marks=[pytest.mark.snowflake]),
+            param("trino", None, marks=[pytest.mark.trino]),
+            param("postgres", None, marks=[pytest.mark.postgresql]),
+            param(
+                "snowflake", get_random_identifier_name(), marks=[pytest.mark.snowflake]
+            ),
         ],
     )
     def test_checkpoint_run(
@@ -293,6 +305,7 @@ def test_checkpoint_run(
         table_factory: TableFactory,
         asset_name: str,
         datasource_type: str,
+        schema: str | None,
     ):
         datasource: SQLDatasource = request.getfixturevalue(f"{datasource_type}_ds")
 
@@ -301,9 +314,13 @@ def test_checkpoint_run(
             pytest.skip(f"no '{asset_name}' table_name for {datasource_type}")
 
         # create table
-        table_factory(engine=datasource.get_engine(), table_names={table_name})
+        table_factory(
+            engine=datasource.get_engine(), table_names={table_name}, schema=schema
+        )
 
-        asset = datasource.add_table_asset(asset_name, table_name=table_name)
+        asset = datasource.add_table_asset(
+            asset_name, table_name=table_name, schema_name=schema
+        )
 
         suite = context.add_expectation_suite(
             expectation_suite_name=f"{datasource.name}-{asset.name}"
