diff --git a/keras/src/backend/common/variables.py b/keras/src/backend/common/variables.py
--- a/keras/src/backend/common/variables.py
+++ b/keras/src/backend/common/variables.py
@@ -276,7 +276,7 @@ def value(self):
         return self._maybe_autocast(self._value)
 
     def assign(self, value):
-        value = self._convert_to_tensor(value, dtype=self.dtype)
+        value = self._convert_to_tensor(value, dtype=self._dtype)
         if not shape_equal(value.shape, self.shape):
             raise ValueError(
                 "The shape of the target variable and "
