diff --git a/homeassistant/components/snapcast/media_player.py b/homeassistant/components/snapcast/media_player.py
index d4d5f98211e8b..d43129af054b4 100644
--- a/homeassistant/components/snapcast/media_player.py
+++ b/homeassistant/components/snapcast/media_player.py
@@ -160,7 +160,10 @@ def state(self) -> MediaPlayerState | None:
         if self._device.connected:
             if self.is_volume_muted or self._current_group.muted:
                 return MediaPlayerState.IDLE
-            return STREAM_STATUS.get(self._current_group.stream_status)
+            try:
+                return STREAM_STATUS.get(self._current_group.stream_status)
+            except KeyError:
+                pass
         return MediaPlayerState.OFF
 
     @property
@@ -275,10 +278,15 @@ async def async_unjoin_player(self) -> None:
     @property
     def metadata(self) -> Mapping[str, Any]:
         """Get metadata from the current stream."""
-        if metadata := self.coordinator.server.stream(
-            self._current_group.stream
-        ).metadata:
-            return metadata
+        try:
+            if metadata := self.coordinator.server.stream(
+                self._current_group.stream
+            ).metadata:
+                return metadata
+        except (
+            KeyError
+        ):  # the stream function raises KeyError if the stream does not exist
+            pass
 
         # Fallback to an empty dict
         return {}
@@ -333,11 +341,15 @@ def media_duration(self) -> int | None:
     @property
     def media_position(self) -> int | None:
         """Position of current playing media in seconds."""
-        # Position is part of properties object, not metadata object
-        if properties := self.coordinator.server.stream(
-            self._current_group.stream
-        ).properties:
-            if (value := properties.get("position")) is not None:
-                return int(value)
-
+        try:
+            # Position is part of properties object, not metadata object
+            if properties := self.coordinator.server.stream(
+                self._current_group.stream
+            ).properties:
+                if (value := properties.get("position")) is not None:
+                    return int(value)
+        except (
+            KeyError
+        ):  # the stream function raises KeyError if the stream does not exist
+            pass
         return None
diff --git a/tests/components/snapcast/conftest.py b/tests/components/snapcast/conftest.py
index 282429b110a3f..5ce92a54811e9 100644
--- a/tests/components/snapcast/conftest.py
+++ b/tests/components/snapcast/conftest.py
@@ -1,7 +1,7 @@
 """Test the snapcast config flow."""
 
 from collections.abc import Generator
-from unittest.mock import AsyncMock, patch
+from unittest.mock import AsyncMock, PropertyMock, patch
 
 import pytest
 from snapcast.control.client import Snapclient
@@ -53,6 +53,8 @@ def mock_create_server(
         mock_server.streams = [mock_stream_1, mock_stream_2]
 
         def get_stream(identifier: str) -> AsyncMock:
+            if len(mock_server.streams) == 0:
+                raise KeyError(identifier)
             return {s.identifier: s for s in mock_server.streams}[identifier]
 
         def get_group(identifier: str) -> AsyncMock:
@@ -94,7 +96,7 @@ def mock_group_1(mock_stream_1: AsyncMock, streams: dict[str, AsyncMock]) -> Asy
     group.friendly_name = "Test Group 1"
     group.stream = mock_stream_1.identifier
     group.muted = False
-    group.stream_status = mock_stream_1.status
+    type(group).stream_status = PropertyMock(return_value=mock_stream_1.status)
     group.volume = 48
     group.streams_by_name.return_value = {s.friendly_name: s for s in streams.values()}
     return group
@@ -109,7 +111,7 @@ def mock_group_2(mock_stream_2: AsyncMock, streams: dict[str, AsyncMock]) -> Asy
     group.friendly_name = "Test Group 2"
     group.stream = mock_stream_2.identifier
     group.muted = False
-    group.stream_status = mock_stream_2.status
+    type(group).stream_status = PropertyMock(return_value=mock_stream_2.status)
     group.volume = 65
     group.streams_by_name.return_value = {s.friendly_name: s for s in streams.values()}
     return group
diff --git a/tests/components/snapcast/test_media_player.py b/tests/components/snapcast/test_media_player.py
index e43005ac75875..908b48cfa5278 100644
--- a/tests/components/snapcast/test_media_player.py
+++ b/tests/components/snapcast/test_media_player.py
@@ -1,6 +1,6 @@
 """Test the snapcast media player implementation."""
 
-from unittest.mock import AsyncMock, patch
+from unittest.mock import AsyncMock, PropertyMock, patch
 
 import pytest
 from syrupy.assertion import SnapshotAssertion
@@ -137,3 +137,41 @@ async def test_join_exception(
 
     # Ensure that the group did not attempt to add a non-Snapcast client
     mock_group_1.add_client.assert_not_awaited()
+
+
+async def test_stream_not_found(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_create_server: AsyncMock,
+    mock_group_2: AsyncMock,
+) -> None:
+    """Test server.stream call KeyError."""
+    mock_create_server.streams = []
+
+    with patch("secrets.token_hex", return_value="mock_token"):
+        await setup_integration(hass, mock_config_entry)
+        assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    state = hass.states.get("media_player.test_client_2_snapcast_client")
+    assert "media_position" not in state.attributes
+    assert "metadata" not in state.attributes
+
+
+async def test_state_stream_not_found(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_create_server: AsyncMock,
+    mock_group_1: AsyncMock,
+) -> None:
+    """Test state returns OFF when stream is not found."""
+
+    type(mock_group_1).stream_status = PropertyMock(
+        side_effect=KeyError("Stream not found")
+    )
+
+    with patch("secrets.token_hex", return_value="mock_token"):
+        await setup_integration(hass, mock_config_entry)
+        assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    state = hass.states.get("media_player.test_client_1_snapcast_client")
+    assert state.state == "off"
