diff --git a/.github/pull_request_template.md b/.github/pull_request_template.md
index 9a815bcf9..c90d18448 100644
--- a/.github/pull_request_template.md
+++ b/.github/pull_request_template.md
@@ -1,18 +1,24 @@
 # Overview
+
 Description
 
 ## Addressed issues
+
 *  Issues addressed
 
 ## Showcase
+
 A short / one-liner example to highlight the (new) feature
-```
+
+```python
 import splinepy
 
 new, feature = splinepy.tataratat()
 ```
 
 ## Checklists
+
+* [ ] Set target branch to develop
 * [ ] Documentations are up-to-date.
 * [ ] Added example(s)
 * [ ] Added test(s)
diff --git a/include/splinepy/py/py_multipatch.hpp b/include/splinepy/py/py_multipatch.hpp
index aaaee7553..8f05462e6 100644
--- a/include/splinepy/py/py_multipatch.hpp
+++ b/include/splinepy/py/py_multipatch.hpp
@@ -393,6 +393,13 @@ class PyMultipatch {
   py::array_t<int> GetInterfaceOrientations(const double tolerance,
                                             const int n_threads);
 
+  /// @brief Set the Boundary Orientations object
+  ///
+  /// Performs sanity checks
+  /// @param interface_orientations
+  /// @return
+  void SetInterfaceOrientations(const py::array_t<int>& interface_orientations);
+
   /// @brief Gets IDs of boundary patch
   /// @param bid boundary ID to be extracted
   py::array_t<int> BoundaryPatchIds(const int bid);
diff --git a/splinepy/multipatch.py b/splinepy/multipatch.py
index ccd3cdd0f..c8a5d48e9 100644
--- a/splinepy/multipatch.py
+++ b/splinepy/multipatch.py
@@ -152,6 +152,34 @@ def interface_orientations(self, tolerance=None, nthreads=None):
             nthreads=_default_if_none(nthreads, _settings.NTHREADS),
         )
 
+    def set_interface_orientations(self, interface_orientations):
+        """
+        Interface orientations are usually computed using the jacobian at face
+        centers. This is not always possible, e.g., if the spline is
+        representing a scalar field on a geometry. This function can be used to
+        manually set the orientations to prepare the spline for export.
+
+        Notes
+        -----
+        The interfaces **must** be set before the orientations, as the
+        interface orientations count the interfaces between splines to ensure
+        they match. This function does not include any checks to verify
+        whether the interfaces exist, since performing such a check would
+        trigger their computation — which might not be possible for scalar
+        splines (the main intended use case for this function).
+
+        Parameters
+        ----------
+        interface_orientations : np.array (np.int64)
+          interface orientations between patches
+          shape {n_interfaces , 4 + 2 * para_dim}
+
+        Returns
+        -------
+        None
+        """
+        super().set_interface_orientations(interface_orientations)
+
     @property
     def boundaries(self):
         """
diff --git a/src/py/py_multipatch.cpp b/src/py/py_multipatch.cpp
index 8e8cf0b95..7c8aa9790 100644
--- a/src/py/py_multipatch.cpp
+++ b/src/py/py_multipatch.cpp
@@ -602,18 +602,24 @@ void GetInterfaceOrientation(
   const int& dim_ = pyspline_start->SplinepyDim();
 
   // Checks
+  if (para_dim_ < 2) {
+    splinepy::utils::PrintAndThrowError(
+        "Curves are not supported for automatic computatation of interface "
+        "orientations");
+  }
+  if (dim_ < 2) {
+    splinepy::utils::PrintAndThrowError(
+        "Scalar splines are not supported for automatic computation of "
+        "interface orientations");
+  }
   if ((para_dim_ != pyspline_end->SplinepyParaDim())
       || (dim_ != pyspline_end->SplinepyDim())) {
     splinepy::utils::PrintAndThrowError(
-        "Spline Orientation can not be checked, as they have mismatching"
-        "dimensionality start spline has dimensions ",
+        "Spline Orientation can not be checked, as they have mismatching "
+        "dimensionality. Current spline has dimensions ",
         para_dim_,
         "D -> ",
         dim_,
-        "D, the adjacent one has dimensions ",
-        pyspline_end->SplinepyParaDim(),
-        "D -> ",
-        pyspline_end->SplinepyDim(),
         "D.");
   }
 
@@ -696,6 +702,42 @@ void GetInterfaceOrientation(
   }
 }
 
+void PyMultipatch::SetInterfaceOrientations(
+    const py::array_t<int>& interface_orientations) {
+
+  // Check if interfaces are already available
+  if (interfaces_.size() == 0) {
+    splinepy::utils::PrintAndThrowError(
+        "Interfaces need to be computed or set first, because orientations "
+        "will be reset upon their recomputation");
+  }
+
+  // Check if the shape of the interface orientation is usable
+  int para_dim = ParaDim();
+  const int n_faces_per_element = 2 * para_dim;
+  // 4 includes - patch id start, face id start, patch id end, face id end
+  const int n_entries_per_line = 4 + n_faces_per_element;
+  // Loop over interfaces to compute number of interfaces
+  const int* interfaces_ptr =
+      static_cast<int*>(GetInterfaces(false).request().ptr);
+  int number_of_interfaces{};
+  for (int i_interface{}; i_interface < interfaces_.size(); i_interface++) {
+    if (interfaces_ptr[i_interface] >= 0) {
+      // interface found
+      if (interfaces_ptr[i_interface] > i_interface) {
+        number_of_interfaces++;
+      }
+    }
+  }
+
+  // size check
+  splinepy::py::CheckPyArrayShape(interface_orientations,
+                                  {number_of_interfaces, n_entries_per_line},
+                                  true);
+
+  interface_orientations_ = interface_orientations;
+}
+
 py::array_t<int> PyMultipatch::GetInterfaceOrientations(const double tolerance,
                                                         const int n_threads) {
 
@@ -711,7 +753,7 @@ py::array_t<int> PyMultipatch::GetInterfaceOrientations(const double tolerance,
 
   const int n_faces_per_element = 2 * para_dim;
   // 4 includes - patch id start, face id start, patch id end, face id end
-  const int n_entries_per_line = 4 + 2 * para_dim;
+  const int n_entries_per_line = 4 + n_faces_per_element;
 
   // Make sure Interfaces are available
   const int* interfaces_ptr =
@@ -1544,6 +1586,9 @@ void PyMultipatch::AddFields(py::list& fields,
   const auto n_current_fields = static_cast<int>(field_multipatches_.size());
   const auto n_new_fields = static_cast<int>(fields.size());
 
+  // Reference to original interfaces (required for extraction)
+  const py::array_t<int> geometric_interfaces = GetInterfaces(false);
+
   // some hint values for null splines
   const int para_dim = ParaDim();
 
@@ -1562,6 +1607,7 @@ void PyMultipatch::AddFields(py::list& fields,
 
       field_ptrs[i] =
           std::make_shared<PyMultipatch>(casted_list, 1, para_dim, field_dim);
+      field_ptrs[i]->SetInterfaces(geometric_interfaces);
 
       // propagate same_parametric_bounds_ flag
       field_ptrs[i]->same_parametric_bounds_ = same_parametric_bounds_;
@@ -1750,6 +1796,9 @@ void init_multipatch(py::module_& m) {
            &PyMultipatch::GetInterfaceOrientations,
            py::arg("tolerance"),
            py::arg("nthreads") = 1)
+      .def("set_interface_orientations",
+           &PyMultipatch::SetInterfaceOrientations,
+           py::arg("interface_orientations"))
       //.def("", &PyMultipatch::)
       ;
 }
