diff --git a/common/src/autogluon/common/features/feature_metadata.py b/common/src/autogluon/common/features/feature_metadata.py
index efb4fa6bb73d..a16355f51e48 100644
--- a/common/src/autogluon/common/features/feature_metadata.py
+++ b/common/src/autogluon/common/features/feature_metadata.py
@@ -51,6 +51,19 @@ def __init__(self, type_map_raw: Dict[str, str], type_group_map_special: Dict[st
 
         self._validate()
 
+    def __eq__(self, other) -> bool:
+        if set(self.type_map_raw.keys()) != set(other.type_map_raw.keys()):
+            return False
+        for k in self.type_map_raw.keys():
+            if self.type_map_raw[k] != other.type_map_raw[k]:
+                return False
+        if set(self.type_group_map_special.keys()) != set(other.type_group_map_special.keys()):
+            return False
+        for k in self.type_group_map_special.keys():
+            if set(self.type_group_map_special[k]) != set(other.type_group_map_special[k]):
+                return False
+        return True
+
     # Confirms if inputs are valid
     def _validate(self):
         type_group_map_special_expanded = []
diff --git a/core/src/autogluon/core/models/abstract/abstract_model.py b/core/src/autogluon/core/models/abstract/abstract_model.py
index c0a03c37a2a8..7c83887b4ec9 100644
--- a/core/src/autogluon/core/models/abstract/abstract_model.py
+++ b/core/src/autogluon/core/models/abstract/abstract_model.py
@@ -454,9 +454,10 @@ def _preprocess_set_features(self, X: pd.DataFrame, feature_metadata: FeatureMet
             self.features = list(X.columns)
         # TODO: Consider changing how this works or where it is done
         if feature_metadata is None:
-            feature_metadata = FeatureMetadata.from_df(X)
+            feature_metadata = self._infer_feature_metadata(X=X)
         else:
             feature_metadata = copy.deepcopy(feature_metadata)
+        feature_metadata = self._update_feature_metadata(X=X, feature_metadata=feature_metadata)
         get_features_kwargs = self.params_aux.get("get_features_kwargs", None)
         if get_features_kwargs is not None:
             valid_features = feature_metadata.get_features(**get_features_kwargs)
@@ -504,6 +505,16 @@ def _preprocess_set_features(self, X: pd.DataFrame, feature_metadata: FeatureMet
         if error_if_no_features and not self._features_internal:
             raise NoValidFeatures
 
+    def _update_feature_metadata(self, X: pd.DataFrame, feature_metadata: FeatureMetadata) -> FeatureMetadata:
+        """
+        [Advanced] Method that performs updates to feature_metadata during initialization.
+        Primarily present for use in stacker models.
+        """
+        return feature_metadata
+
+    def _infer_feature_metadata(self, X: pd.DataFrame) -> FeatureMetadata:
+        return FeatureMetadata.from_df(X)
+
     def _preprocess_fit_args(self, **kwargs) -> dict:
         sample_weight = kwargs.get("sample_weight", None)
         if sample_weight is not None and isinstance(sample_weight, str):
@@ -562,11 +573,11 @@ def _initialize(self, X=None, y=None, feature_metadata=None, num_classes=None, *
 
         self._init_misc(X=X, y=y, feature_metadata=feature_metadata, num_classes=num_classes, **kwargs)
 
+        self._init_params()
+
         if X is not None:
             self._preprocess_set_features(X=X, feature_metadata=feature_metadata)
 
-        self._init_params()
-
     def _init_misc(self, **kwargs):
         """Initialize parameters that depend on self.params_aux being initialized"""
         if self.eval_metric is None:
diff --git a/core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py b/core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py
index 83315462d098..70aebe9328ef 100644
--- a/core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py
+++ b/core/src/autogluon/core/models/ensemble/stacker_ensemble_model.py
@@ -1,16 +1,17 @@
+from __future__ import annotations
+
 import copy
 import logging
 import os
 import time
 from collections import defaultdict
-from typing import Dict
+from typing import Dict, List
 
 import numpy as np
 import pandas as pd
 
 from autogluon.common.features.feature_metadata import FeatureMetadata
 from autogluon.common.features.types import R_FLOAT, S_STACK
-from autogluon.common.utils.path_converter import PathConverter
 
 from ...constants import MULTICLASS, QUANTILE, SOFTCLASS
 from ..abstract.abstract_model import AbstractModel
@@ -35,12 +36,12 @@ class StackerEnsembleModel(BaggedEnsembleModel):
 
     def __init__(
         self,
-        base_model_names=None,
-        base_models_dict=None,
-        base_model_paths_dict=None,
-        base_model_types_dict=None,
-        base_model_types_inner_dict=None,
-        base_model_performances_dict=None,
+        base_model_names: List[str] | None = None,
+        base_models_dict: Dict[str, AbstractModel] | None = None,
+        base_model_paths_dict: Dict[str, str] = None,
+        base_model_types_dict: dict | None = None,
+        base_model_types_inner_dict: dict | None = None,
+        base_model_performances_dict: Dict[str, float] | None = None,
         **kwargs,
     ):
         super().__init__(**kwargs)
@@ -61,17 +62,22 @@ def __init__(
         self._base_model_performances_dict = base_model_performances_dict
         self._base_model_types_inner_dict = base_model_types_inner_dict
 
-    def _initialize(self, **kwargs):
-        super()._initialize(**kwargs)
+    def _update_feature_metadata(self, X: pd.DataFrame, feature_metadata: FeatureMetadata) -> FeatureMetadata:
+        """
+        Updates base_model_names and feature_metadata to reflect the used base models.
+        """
         base_model_performances_dict = self._base_model_performances_dict
         base_model_types_inner_dict = self._base_model_types_inner_dict
         if (base_model_performances_dict is not None) and (base_model_types_inner_dict is not None):
-            if self.params["max_base_models_per_type"] > 0:
+            max_base_models_per_type = self.params["max_base_models_per_type"]
+            if isinstance(max_base_models_per_type, str):
+                max_base_models_per_type = self._get_dynamic_max_base_models_per_type(X=X)
+            if max_base_models_per_type > 0:
                 self.base_model_names = self.limit_models_per_type(
                     models=self.base_model_names,
                     model_types=base_model_types_inner_dict,
                     model_scores=base_model_performances_dict,
-                    max_base_models_per_type=self.params["max_base_models_per_type"],
+                    max_base_models_per_type=max_base_models_per_type,
                 )
             if self.params["max_base_models"] > 0:
                 self.base_model_names = self.limit_models(
@@ -88,7 +94,49 @@ def _initialize(self, **kwargs):
             stack_column_prefix: self.base_model_names[i] for i, stack_column_prefix in enumerate(self.stack_column_prefix_lst)
         }
 
-        self._add_stack_to_feature_metadata()
+        feature_metadata = self._remove_unused_stack_in_feature_metadata(feature_metadata=feature_metadata)
+        return feature_metadata
+
+    def _get_dynamic_max_base_models_per_type(self, X: pd.DataFrame):
+        num_rows = len(X)
+        if num_rows < 1000:
+            max_models_per_type = 1
+        elif num_rows < 5000:
+            max_models_per_type = 2
+        elif num_rows < 10000:
+            max_models_per_type = 3
+        elif num_rows < 15000:
+            max_models_per_type = 4
+        elif num_rows < 20000:
+            max_models_per_type = 5
+        elif num_rows < 25000:
+            max_models_per_type = 6
+        elif num_rows < 30000:
+            max_models_per_type = 7
+        elif num_rows < 35000:
+            max_models_per_type = 8
+        elif num_rows < 40000:
+            max_models_per_type = 9
+        elif num_rows < 45000:
+            max_models_per_type = 10
+        elif num_rows < 50000:
+            max_models_per_type = 11
+        else:
+            max_models_per_type = 12
+        return max_models_per_type
+
+    def _infer_feature_metadata(self, X: pd.DataFrame) -> FeatureMetadata:
+        """
+        Additionally adds the stack feature special types to the inferred feature_metadata.
+        """
+        feature_metadata = super()._infer_feature_metadata(X=X)
+        stack_column_prefix_lst = copy.deepcopy(self.base_model_names)
+        stack_columns, num_pred_cols_per_model = self.set_stack_columns(stack_column_prefix_lst=stack_column_prefix_lst)
+        type_map_raw = {column: R_FLOAT for column in stack_columns}
+        type_group_map_special = {S_STACK: stack_columns}
+        stacker_feature_metadata = FeatureMetadata(type_map_raw=type_map_raw, type_group_map_special=type_group_map_special)
+        feature_metadata = feature_metadata.add_special_types(stacker_feature_metadata.get_type_map_special())
+        return feature_metadata
 
     @staticmethod
     def limit_models_per_type(models, model_types, model_scores, max_base_models_per_type):
@@ -108,7 +156,7 @@ def limit_models(self, models, model_scores, max_base_models):
         return self.limit_models_per_type(models=models, model_types=model_types, model_scores=model_scores, max_base_models_per_type=max_base_models)
 
     def _set_default_params(self):
-        default_params = {"use_orig_features": True, "max_base_models": 25, "max_base_models_per_type": 5}
+        default_params = {"use_orig_features": True, "max_base_models": 0, "max_base_models_per_type": "auto"}
         for param, val in default_params.items():
             self._set_default_param_value(param, val)
         super()._set_default_params()
@@ -215,19 +263,19 @@ def get_info(self):
         info["children_info"] = children_info  # Ensure children_info is last in order
         return info
 
-    def _add_stack_to_feature_metadata(self):
-        if len(self.models) == 0:
-            type_map_raw = {column: R_FLOAT for column in self.stack_columns}
-            type_group_map_special = {S_STACK: self.stack_columns}
-            stacker_feature_metadata = FeatureMetadata(type_map_raw=type_map_raw, type_group_map_special=type_group_map_special)
-            if self.feature_metadata is None:  # TODO: This is probably not the best way to do this
-                self.feature_metadata = stacker_feature_metadata
-            else:
-                # FIXME: This is a hack, stack feature special types should be already present in feature_metadata, not added here
-                existing_stack_features = self.feature_metadata.get_features(required_special_types=[S_STACK])
-                # HACK: Currently AutoGluon auto-adds all base learner prediction features into self.feature_metadata.
-                # The two lines below are here because if feature pruning would crash if it prunes a base learner prediction feature.
-                existing_features = self.feature_metadata.get_features()
-                stacker_feature_metadata = stacker_feature_metadata.keep_features([feature for feature in existing_features if feature in type_map_raw])
-                if set(stacker_feature_metadata.get_features()) != set(existing_stack_features):
-                    self.feature_metadata = self.feature_metadata.add_special_types(stacker_feature_metadata.get_type_map_special())
+    def _remove_unused_stack_in_feature_metadata(self, feature_metadata: FeatureMetadata) -> FeatureMetadata:
+        """
+        Updates `self.feature_metadata` to only contain stack features specified in `self.stack_columns`.
+        """
+        assert feature_metadata is not None, f"feature_metadata must be specified prior to adding stack feature information."
+        # Trust feature metadata
+        original_stack_features = feature_metadata.get_features(required_special_types=[S_STACK])
+        current_stack_features = self.stack_columns
+        for stack_feature in current_stack_features:
+            assert stack_feature in original_stack_features, (
+                f"Missing expected stack feature '{stack_feature}' in original feature_metadata. "
+                f"Stack features in feature_metadata: {original_stack_features}"
+            )
+        stack_features_to_remove = [stack_feature for stack_feature in original_stack_features if stack_feature not in current_stack_features]
+        feature_metadata = feature_metadata.remove_features(features=stack_features_to_remove)
+        return feature_metadata
diff --git a/core/src/autogluon/core/trainer/abstract_trainer.py b/core/src/autogluon/core/trainer/abstract_trainer.py
index 62d6b8c90230..670de529085f 100644
--- a/core/src/autogluon/core/trainer/abstract_trainer.py
+++ b/core/src/autogluon/core/trainer/abstract_trainer.py
@@ -16,6 +16,7 @@
 import pandas as pd
 
 from autogluon.common.features.feature_metadata import FeatureMetadata
+from autogluon.common.features.types import R_FLOAT, S_STACK
 from autogluon.common.utils.lite import disable_if_lite_mode
 from autogluon.common.utils.log_utils import convert_time_in_s_to_log_friendly
 from autogluon.common.utils.path_converter import PathConverter
@@ -694,6 +695,8 @@ def stack_new_level_core(
                     "base_model_names": base_model_names,
                     "base_model_paths_dict": base_model_paths,
                     "base_model_types_dict": base_model_types,
+                    "base_model_types_inner_dict": self.get_models_attribute_dict(attribute="type_inner", models=base_model_names),
+                    "base_model_performances_dict": self.get_models_attribute_dict(attribute="val_score", models=base_model_names),
                     "random_state": level + self.random_state,
                 }
                 get_models_kwargs.update(
@@ -713,6 +716,7 @@ def stack_new_level_core(
                 kwargs["hyperparameter_tune_kwargs"] = hyperparameter_tune_kwargs
         logger.log(20, f"Fitting {len(models)} L{level} models ...")
         X_init = self.get_inputs_to_stacker(X, base_models=base_model_names, fit=True)
+        feature_metadata = self.get_feature_metadata(use_orig_features=True, base_models=base_model_names)
         if X_val is not None:
             X_val = self.get_inputs_to_stacker(X_val, base_models=base_model_names, fit=False, use_val_cache=True)
         compute_score = not refit_full
@@ -724,7 +728,10 @@ def stack_new_level_core(
         if X_unlabeled is not None:
             X_unlabeled = self.get_inputs_to_stacker(X_unlabeled, base_models=base_model_names, fit=False)
 
-        fit_kwargs = dict(num_classes=self.num_classes)
+        fit_kwargs = dict(
+            num_classes=self.num_classes,
+            feature_metadata=feature_metadata,
+        )
 
         # FIXME: TODO: v0.1 X_unlabeled isn't cached so it won't be available during refit_full or fit_extra.
         return self._train_multi(
@@ -1239,6 +1246,7 @@ def get_inputs_to_stacker(
     ) -> pd.DataFrame:
         """
         Returns the valid X input for a stacker model with base models equal to `base_models`.
+        Pairs with `feature_metadata = self.get_feature_metadata(...)`. The contents of the returned `X` should reflect `feature_metadata`.
 
         Parameters
         ----------
@@ -1291,6 +1299,48 @@ def get_inputs_to_stacker(
             X = X_stacker
         return X
 
+    def get_feature_metadata(self, use_orig_features: bool = True, model: str | None = None, base_models: List[str] | None = None) -> FeatureMetadata:
+        """
+        Returns the FeatureMetadata input to a `model.fit` call.
+        Pairs with `X = self.get_inputs_to_stacker(...)`. The returned FeatureMetadata should reflect the contents of `X`.
+
+        Parameters
+        ----------
+        use_orig_features : bool, default = True
+            If True, will include the original features in the FeatureMetadata.
+            If False, will only include the stack features in the FeatureMetadata.
+        model : str, default = None
+            If specified, it must be an already existing model.
+            `base_models` will be set to the base models of `model`.
+        base_models : List[str], default = None
+            If specified, will add the stack features of the `base_models` to FeatureMetadata.
+
+        Returns
+        -------
+        FeatureMetadata
+            The FeatureMetadata that should be passed into a `model.fit` call.
+        """
+        if model is not None and base_models is not None:
+            raise AssertionError("Only one of `model`, `base_models` is allowed to be set.")
+        if model is not None and base_models is None:
+            base_models = self.get_base_model_names(model)
+
+        feature_metadata = None
+        if use_orig_features:
+            feature_metadata = self.feature_metadata
+        if base_models:
+            stack_column_names, _ = self._get_stack_column_names(models=base_models)
+            stacker_type_map_raw = {column: R_FLOAT for column in stack_column_names}
+            stacker_type_group_map_special = {S_STACK: stack_column_names}
+            stacker_feature_metadata = FeatureMetadata(type_map_raw=stacker_type_map_raw, type_group_map_special=stacker_type_group_map_special)
+            if feature_metadata is not None:
+                feature_metadata = feature_metadata.join_metadata(stacker_feature_metadata)
+            else:
+                feature_metadata = stacker_feature_metadata
+        if feature_metadata is None:
+            feature_metadata = FeatureMetadata(type_map_raw={})
+        return feature_metadata
+
     def _get_stack_column_names(self, models: List[str]) -> Tuple[List[str], int]:
         """
         Get the stack column names generated when the provided models are used as base models in a stack ensemble.
@@ -1781,6 +1831,8 @@ def generate_weighted_ensemble(
             else:
                 save_bag_folds = True
 
+        feature_metadata = self.get_feature_metadata(use_orig_features=False, base_models=base_model_names)
+
         base_model_paths_dict = self.get_models_attribute_dict(attribute="path", models=base_model_names)
         base_model_paths_dict = {key: os.path.join(self.path, val) for key, val in base_model_paths_dict.items()}
         weighted_ensemble_model, _ = get_models_func(
@@ -1822,7 +1874,7 @@ def generate_weighted_ensemble(
             level=level,
             time_limit=time_limit,
             ens_sample_weight=w,
-            fit_kwargs=dict(num_classes=self.num_classes, groups=None),  # FIXME: Is this the right way to do this?
+            fit_kwargs=dict(feature_metadata=feature_metadata, num_classes=self.num_classes, groups=None),  # FIXME: Is this the right way to do this?
         )
         for weighted_ensemble_model_name in models:
             if check_if_best and weighted_ensemble_model_name in self.get_model_names():
@@ -3688,21 +3740,9 @@ def _get_model_fit_kwargs(
             if k_fold == self.k_fold:  # don't do this on refit full
                 model_fit_kwargs["groups"] = self._groups
 
-        #######################
-        # FIXME: This section is a hack, compute genuine feature_metadata for each stack level instead
-        #  Don't do this here, do this upstream so it isn't recomputed for each model
-        #  Add feature_metadata to model_fit_kwargs
         # FIXME: Sample weight `extract_column` is a hack, have to compute feature_metadata here because sample weight column could be in X upstream, extract sample weight column upstream instead.
-        # FIXME: This doesn't assign proper special types to stack features, relying on a hack in StackerEnsembleModel to assign S_STACK to feature metadata, don't do this.
-        #  Remove hack in StackerEnsembleModel
-        feature_metadata = self.feature_metadata
-        features_base = self.feature_metadata.get_features()
-        features_new = [feature for feature in X.columns if feature not in features_base]
-        if features_new:
-            feature_metadata_new = FeatureMetadata.from_df(X[features_new])
-            feature_metadata = feature_metadata.join_metadata(feature_metadata_new).keep_features(list(X.columns))
-        model_fit_kwargs["feature_metadata"] = feature_metadata
-        #######################
+        if "feature_metadata" not in model_fit_kwargs:
+            raise AssertionError(f"Missing expected parameter 'feature_metadata'.")
         return model_fit_kwargs
 
     def _get_bagged_model_fit_kwargs(self, k_fold: int, k_fold_start: int, k_fold_end: int, n_repeats: int, n_repeat_start: int) -> dict:
diff --git a/features/tests/features/test_feature_metadata.py b/features/tests/features/test_feature_metadata.py
index a72a44c07807..6c0737b2bc78 100644
--- a/features/tests/features/test_feature_metadata.py
+++ b/features/tests/features/test_feature_metadata.py
@@ -196,3 +196,30 @@ def test_feature_metadata_get_features():
     assert feature_metadata.get_features(valid_raw_types=["2", "3"], required_special_types=["s1"], required_exact=True) == []
     assert feature_metadata.get_features(valid_raw_types=["2", "3"], required_special_types=["s1", "s3"]) == ["b"]
     assert feature_metadata.get_features(valid_raw_types=["2", "3"], required_special_types=["s1", "s3"], required_exact=True) == ["b"]
+
+
+def test_feature_metadata_equals():
+    type_map_raw = dict(
+        a="1",
+        b="2",
+        c="3",
+        d="1",
+        e="1",
+        f="4",
+    )
+    type_group_map_special = {"s1": ["a", "b", "d"], "s2": ["a", "e"], "s3": ["a", "b"], "s4": ["f"]}
+    feature_metadata = FeatureMetadata(type_map_raw=type_map_raw, type_group_map_special=type_group_map_special)
+    feature_metadata_2 = FeatureMetadata(type_map_raw=type_map_raw, type_group_map_special=type_group_map_special)
+    assert feature_metadata == feature_metadata_2
+    feature_metadata_2 = feature_metadata_2.remove_features(features=["a"])
+    assert feature_metadata != feature_metadata_2
+
+    feature_metadata_3 = FeatureMetadata(type_map_raw=dict(a="1"))
+    feature_metadata_2 = feature_metadata_2.join_metadata(feature_metadata_3)
+    assert feature_metadata != feature_metadata_2
+
+    feature_metadata_2 = feature_metadata_2.add_special_types(type_map_special={"a": ["s1"]})
+    assert feature_metadata != feature_metadata_2
+
+    feature_metadata_2 = feature_metadata_2.add_special_types(type_map_special={"a": ["s2", "s3"]})
+    assert feature_metadata == feature_metadata_2
diff --git a/tabular/tests/unittests/models/advanced/test_stack_feature_usage.py b/tabular/tests/unittests/models/advanced/test_stack_feature_usage.py
new file mode 100644
index 000000000000..fb900fcb301d
--- /dev/null
+++ b/tabular/tests/unittests/models/advanced/test_stack_feature_usage.py
@@ -0,0 +1,145 @@
+"""
+Unit tests to ensure correctness of internal stacking logic.
+"""
+import shutil
+
+from autogluon.common.features.types import S_STACK
+from autogluon.core.models.ensemble.stacker_ensemble_model import StackerEnsembleModel
+from autogluon.tabular.predictor import TabularPredictor
+
+
+def test_stack_feature_usage_binary(fit_helper):
+    """Tests that stacker models use base model predictions as features correctly for binary"""
+    dataset_name = "adult"
+    expected_ancestors = {"LightGBM_BAG_L1", "DummyModel_BAG_L1"}
+    _fit_predictor_stack_feature_usage(
+        dataset_name=dataset_name,
+        max_base_models_per_type=1,
+        max_base_models=2,
+        sample_size=100,
+        expected_ancestors=expected_ancestors,
+        fit_helper=fit_helper,
+    )
+
+
+def test_stack_feature_usage_multiclass(fit_helper):
+    """Tests that stacker models use base model predictions as features correctly for multiclass"""
+    dataset_name = "covertype_small"
+    expected_ancestors = {"LightGBM_BAG_L1", "DummyModel_BAG_L1"}
+    _fit_predictor_stack_feature_usage(
+        dataset_name=dataset_name,
+        max_base_models_per_type=1,
+        max_base_models=2,
+        sample_size=100,
+        expected_ancestors=expected_ancestors,
+        fit_helper=fit_helper,
+    )
+
+
+def test_stack_feature_usage_regression(fit_helper):
+    """Tests that stacker models use base model predictions as features correctly for regression"""
+    dataset_name = "ames"
+    expected_ancestors = {"LightGBM_BAG_L1", "KNeighbors_BAG_L1"}
+    _fit_predictor_stack_feature_usage(
+        dataset_name=dataset_name,
+        max_base_models_per_type=1,
+        max_base_models=2,
+        sample_size=100,
+        expected_ancestors=expected_ancestors,
+        fit_helper=fit_helper,
+    )
+
+
+def test_stack_feature_usage_binary_all(fit_helper):
+    """Tests that stacker models use base model predictions as features correctly for binary"""
+    dataset_name = "adult"
+    expected_ancestors = {"LightGBM_BAG_L1", "LightGBM_2_BAG_L1", "KNeighbors_BAG_L1", "DummyModel_BAG_L1"}
+    _fit_predictor_stack_feature_usage(
+        dataset_name=dataset_name,
+        max_base_models_per_type=0,  # uncapped
+        max_base_models=0,  # uncapped
+        sample_size=100,
+        expected_ancestors=expected_ancestors,
+        fit_helper=fit_helper,
+    )
+
+
+def test_stack_feature_usage_binary_only_max_models(fit_helper):
+    """Tests that stacker models use base model predictions as features correctly for binary"""
+    dataset_name = "adult"
+    expected_ancestors = {"LightGBM_BAG_L1", "LightGBM_2_BAG_L1"}
+    _fit_predictor_stack_feature_usage(
+        dataset_name=dataset_name,
+        max_base_models_per_type=0,  # uncapped
+        max_base_models=2,
+        sample_size=100,
+        expected_ancestors=expected_ancestors,
+        fit_helper=fit_helper,
+    )
+
+
+def _fit_predictor_stack_feature_usage(
+    dataset_name: str,
+    max_base_models_per_type: int,
+    max_base_models: int,
+    sample_size: int,
+    expected_ancestors: set,
+    fit_helper,
+):
+    """Tests that stacker models use base model predictions as features correctly"""
+    fit_args = dict(
+        hyperparameters={
+            "GBM": [
+                {"num_iterations": 200},
+                {"num_iterations": 50},
+            ],
+            "KNN": {},
+            "DUMMY": {},
+        },
+        ag_args_ensemble={
+            "max_base_models_per_type": max_base_models_per_type,
+            "max_base_models": max_base_models,
+            "fold_fitting_strategy": "sequential_local",
+        },
+        num_bag_folds=2,
+        num_stack_levels=1,
+        fit_weighted_ensemble=False,
+    )
+    predictor = fit_helper.fit_and_validate_dataset(
+        dataset_name=dataset_name,
+        fit_args=fit_args,
+        refit_full=False,
+        sample_size=sample_size,
+        expected_model_count=7,
+        delete_directory=False,
+    )
+    _assert_stack_features(predictor=predictor, expected_ancestors=expected_ancestors)
+    shutil.rmtree(predictor.path, ignore_errors=True)
+
+
+def _assert_stack_features(predictor: TabularPredictor, expected_ancestors: set):
+    """
+    Verifies that stack features are correctly set and used
+    """
+    leaderboard = predictor.leaderboard(extra_info=True)
+    leaderboard_l2 = leaderboard[leaderboard["stack_level"] == 2]
+    leaderboard_l2 = leaderboard_l2.set_index("model")
+    for m in leaderboard_l2.index:
+        ancestors = set(leaderboard_l2.loc[m, "ancestors"])
+        assert expected_ancestors == ancestors
+        model = predictor._trainer.load_model(model_name=m)
+        assert isinstance(model, StackerEnsembleModel)
+        assert expected_ancestors == set(model.base_model_names)
+        assert expected_ancestors == set(predictor._trainer.get_minimum_model_set(model=m, include_self=False))
+        stack_columns = model.stack_columns
+        stack_columns_fm = model.feature_metadata.get_features(required_special_types=[S_STACK])
+        assert set(stack_columns) == set(stack_columns_fm)
+        features = model.feature_metadata.get_features()
+        assert set(features) == set(model.features)
+        # Asserts that child models use the same features as parent models
+        for child_model_name in model.models:
+            child_model = model.load_child(model=child_model_name)
+            child_stack_columns_fm = child_model.feature_metadata.get_features(required_special_types=[S_STACK])
+            assert set(stack_columns) == set(child_stack_columns_fm)
+            assert set(features) == set(child_model.features)
+            assert model.feature_metadata == child_model.feature_metadata
