diff --git a/dashboard/head.py b/dashboard/head.py
--- a/dashboard/head.py
+++ b/dashboard/head.py
@@ -3,6 +3,7 @@
 import json
 import asyncio
 import logging
+import ipaddress
 
 import aiohttp
 import aiohttp.web
@@ -210,6 +211,8 @@ async def _async_notify():
             raise Exception(f"Failed to find a valid port for dashboard after "
                             f"{self.http_port_retries} retries: {last_ex}")
         http_host, http_port, *_ = site._server.sockets[0].getsockname()
+        http_host = self.ip if ipaddress.ip_address(
+            http_host).is_unspecified else http_host
         logger.info("Dashboard head http address: %s:%s", http_host, http_port)
 
         # Write the dashboard head port to redis.
