diff --git a/scipy/special/tests/test_legendre.py b/scipy/special/tests/test_legendre.py
index c02da3dbc5c0..7625313ef258 100644
--- a/scipy/special/tests/test_legendre.py
+++ b/scipy/special/tests/test_legendre.py
@@ -321,6 +321,27 @@ def test_all_limits(self, m_max, n_max, x):
             np.testing.assert_allclose(p_jac[:, m], 0)
             np.testing.assert_allclose(p_jac[:, -m], 0)
 
+    @pytest.mark.parametrize("n", np.arange(0, 20))
+    def test_assoc_legendre_p_norm_m0(self, n):
+        # gh 24099
+        m = 0
+        z = np.linspace(-1, 1, 101)
+        p = assoc_legendre_p(n, m, z, norm=False)
+        p_norm = assoc_legendre_p(n, m, z, norm=True)
+        factor = np.sqrt((2 * n + 1) / 2)
+        np.testing.assert_allclose(p_norm, factor * p)
+
+    @pytest.mark.parametrize("n", np.arange(0, 20))
+    def test_assoc_legendre_p_all_norm_m0(self, n):
+        # gh 24099
+        m = 0
+        z = np.linspace(-1, 1, 101)
+        p_all = assoc_legendre_p_all(n, m, z, norm=False)[0, : n + 1, 0, :]
+        p_norm_all = assoc_legendre_p_all(n, m, z, norm=True)[0, : n + 1, 0, :]
+        factor = np.sqrt((2 * np.arange(n + 1) + 1) / 2)
+        factor = factor[:, np.newaxis]
+        np.testing.assert_allclose(p_norm_all, factor * p_all)
+
 
 class TestMultiAssocLegendreP:
     @pytest.mark.parametrize("shape", [(1000,), (4, 9), (3, 5, 7)])
