diff --git a/keras/src/metrics/reduction_metrics.py b/keras/src/metrics/reduction_metrics.py
--- a/keras/src/metrics/reduction_metrics.py
+++ b/keras/src/metrics/reduction_metrics.py
@@ -199,6 +199,9 @@ def __init__(self, fn, name=None, dtype=None, **kwargs):
             self._direction = "down"
 
     def update_state(self, y_true, y_pred, sample_weight=None):
+        y_true = backend.cast(y_true, self.dtype)
+        y_pred = backend.cast(y_pred, self.dtype)
+
         mask = backend.get_keras_mask(y_pred)
         values = self._fn(y_true, y_pred, **self._fn_kwargs)
         if sample_weight is not None and mask is not None:
