diff --git a/docs/news.rst b/docs/news.rst
index 47aca8c9f..ab9c83af3 100644
--- a/docs/news.rst
+++ b/docs/news.rst
@@ -61,6 +61,8 @@ Psycopg 3.2.12
 
 - Allow copy to pass different types per column, as long as the database can
   convert them. Regression introduced in 3.2.11 (:ticket:`#1192`).
+- Show the host name in the error message in case of name resolution error
+  (:ticket:`#1205`).
 
 
 Psycopg 3.2.11
diff --git a/psycopg/psycopg/_conninfo_attempts.py b/psycopg/psycopg/_conninfo_attempts.py
index f853e0537..b619ae78e 100644
--- a/psycopg/psycopg/_conninfo_attempts.py
+++ b/psycopg/psycopg/_conninfo_attempts.py
@@ -44,13 +44,15 @@ def conninfo_attempts(params: ConnMapping) -> list[ConnDict]:
         try:
             attempts.extend(_resolve_hostnames(attempt))
         except OSError as ex:
-            logger.debug("failed to resolve host %r: %s", attempt.get("host"), ex)
-            last_exc = ex
+            last_exc = e.OperationalError(
+                f"failed to resolve host {attempt.get('host')!r}: {ex}"
+            )
+            logger.debug("%s", last_exc)
 
     if not attempts:
         assert last_exc
         # We couldn't resolve anything
-        raise e.OperationalError(str(last_exc))
+        raise last_exc
 
     if get_param(params, "load_balance_hosts") == "random":
         shuffle(attempts)
diff --git a/psycopg/psycopg/_conninfo_attempts_async.py b/psycopg/psycopg/_conninfo_attempts_async.py
index 08f363d1a..21228b4d4 100644
--- a/psycopg/psycopg/_conninfo_attempts_async.py
+++ b/psycopg/psycopg/_conninfo_attempts_async.py
@@ -42,13 +42,15 @@ async def conninfo_attempts_async(params: ConnMapping) -> list[ConnDict]:
         try:
             attempts.extend(await _resolve_hostnames(attempt))
         except OSError as ex:
-            logger.debug("failed to resolve host %r: %s", attempt.get("host"), ex)
-            last_exc = ex
+            last_exc = e.OperationalError(
+                f"failed to resolve host {attempt.get('host')!r}: {ex}"
+            )
+            logger.debug("%s", last_exc)
 
     if not attempts:
         assert last_exc
         # We couldn't resolve anything
-        raise e.OperationalError(str(last_exc))
+        raise last_exc
 
     if get_param(params, "load_balance_hosts") == "random":
         shuffle(attempts)
diff --git a/tests/test_connection.py b/tests/test_connection.py
index 6b7ee93b8..6b842b142 100644
--- a/tests/test_connection.py
+++ b/tests/test_connection.py
@@ -506,6 +506,14 @@ def fake_connect(conninfo, *, timeout=0.0):
     conn.close()
 
 
+def test_resolve_error_host(conn_cls):
+    host = "nosuchhost.example.com"
+    with pytest.raises(psycopg.OperationalError) as ex:
+        conn_cls.connect(host=host)
+
+    assert host in str(ex.value)
+
+
 @pytest.mark.parametrize(
     "args, kwargs, exctype",
     [
diff --git a/tests/test_connection_async.py b/tests/test_connection_async.py
index 8e70947af..6a8f05349 100644
--- a/tests/test_connection_async.py
+++ b/tests/test_connection_async.py
@@ -502,6 +502,14 @@ def fake_connect(conninfo, *, timeout=0.0):
     await conn.close()
 
 
+async def test_resolve_error_host(aconn_cls):
+    host = "nosuchhost.example.com"
+    with pytest.raises(psycopg.OperationalError) as ex:
+        await aconn_cls.connect(host=host)
+
+    assert host in str(ex.value)
+
+
 @pytest.mark.parametrize(
     "args, kwargs, exctype",
     [
