diff --git a/src/napari/layers/_layer_actions.py b/src/napari/layers/_layer_actions.py
index a790b4cacd3..bb92f8654d3 100644
--- a/src/napari/layers/_layer_actions.py
+++ b/src/napari/layers/_layer_actions.py
@@ -61,7 +61,10 @@ def _convert(ll: LayerList, type_: str) -> None:
     for lay in list(ll.selection):
         idx = ll.index(lay)
         if isinstance(lay, Shapes) and type_ == 'labels':
-            data = lay.to_labels()
+            ll_shape = (
+                ll._extent_world_augmented[1] - ll._extent_world_augmented[0]
+            )
+            data = lay.to_labels(labels_shape=lay.world_to_data(ll_shape))
             idx += 1
         elif (
             not np.issubdtype(lay.data.dtype, np.integer) and type_ == 'labels'
diff --git a/src/napari/layers/_tests/test_layer_actions.py b/src/napari/layers/_tests/test_layer_actions.py
index 706bd5d7a77..2b320979308 100644
--- a/src/napari/layers/_tests/test_layer_actions.py
+++ b/src/napari/layers/_tests/test_layer_actions.py
@@ -313,7 +313,36 @@ def test_convert_layer(layer, type_):
         )  # check array data not copied unnecessarily
 
 
-def test_convert_warns_with_projecton_mode():
+@pytest.mark.parametrize(
+    ('scale', 'translate', 'xfail'),
+    [
+        ((1.0, 1.0), (0.0, 0.0), False),  # default
+        ((1.0, 1.0), (30.0, 30.0), True),  # translated, currently fails
+        ((5.0, 5.0), (0.0, 0.0), False),  # scaled
+    ],
+)
+def test_make_label_from_shape_param(scale, translate, xfail):
+    """Tests that label shape matches the maximum extent of added shape, with optional scale and translate."""
+    ll = LayerList()
+    # add an image
+    layer = Image(np.zeros((20, 20)))
+    layer.scale = np.array(scale)
+    layer.translate = np.array(translate)
+    ll.append(layer)
+    # add a shape within the image
+    shape = Shapes([np.array([[5, 5], [5, 25], [25, 5], [25, 25]])])
+    shape.scale = np.array(scale)
+    shape.translate = np.array(translate)
+    ll.append(shape)
+    # Create a label based on the shape.
+    if xfail:
+        pytest.xfail('Converting layers with translations does not work')
+    _convert(ll, 'labels')
+    # the label layer should match the layer list extent
+    assert np.array_equal(ll[-1].extent.world, ll.extent.world)
+
+
+def test_convert_warns_with_projection_mode():
     # inplace
     ll = LayerList(
         [Image(np.random.rand(10, 10).astype(int), projection_mode='mean')]
