diff --git a/.pre-commit-config.yaml b/.pre-commit-config.yaml
index 8549b0163b..b1c2a87615 100644
--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -2,12 +2,18 @@ repos:
   - repo: https://github.com/astral-sh/ruff-pre-commit
     rev: 45ef068da5f21267bb2a7ec4a623092959f09ce5  # frozen: v0.14.14
     hooks:
-      - id: ruff
+      - id: ruff-check
       - id: ruff-format
   - repo: https://github.com/astral-sh/uv-pre-commit
     rev: 8afdb18d56d3bd3edea0c4ddde96965d6894f5f3  # frozen: 0.9.26
     hooks:
       - id: uv-lock
+  - repo: https://github.com/codespell-project/codespell
+    rev: 63c8f8312b7559622c0d82815639671ae42132ac  # frozen v2.4.1
+    hooks:
+      - id: codespell
+        additional_dependencies:
+          - tomli
   - repo: https://github.com/pre-commit/pre-commit-hooks
     rev: 3e8a8703264a2f4a69428a0aa4dcb512790b2c8c  # frozen: v6.0.0
     hooks:
diff --git a/pyproject.toml b/pyproject.toml
index 660239f9c1..20424de48d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -166,6 +166,9 @@ select = [
 force-single-line = true
 order-by-type = false
 
+[tool.codespell]
+ignore-words-list = "te"
+
 [tool.tox]
 env_list = [
     "py3.14", "py3.14t",
diff --git a/src/flask/app.py b/src/flask/app.py
index 1232b03dd4..c6dd7833ee 100644
--- a/src/flask/app.py
+++ b/src/flask/app.py
@@ -275,7 +275,7 @@ def __init__(
                 f"{self.static_url_path}/<path:filename>",
                 endpoint="static",
                 host=static_host,
-                view_func=lambda **kw: self_ref().send_static_file(**kw),  # type: ignore # noqa: B950
+                view_func=lambda **kw: self_ref().send_static_file(**kw),  # type: ignore
             )
 
     def get_send_file_max_age(self, filename: str | None) -> int | None:
@@ -1512,7 +1512,7 @@ def wsgi_app(
             except Exception as e:
                 error = e
                 response = self.handle_exception(e)
-            except:  # noqa: B001
+            except:
                 error = sys.exc_info()[1]
                 raise
             return response(environ, start_response)
diff --git a/src/flask/sessions.py b/src/flask/sessions.py
index c1d978d3ec..1841d8826a 100644
--- a/src/flask/sessions.py
+++ b/src/flask/sessions.py
@@ -107,7 +107,7 @@ def _fail(self, *args: t.Any, **kwargs: t.Any) -> t.NoReturn:
             "application to something unique and secret."
         )
 
-    __setitem__ = __delitem__ = clear = pop = popitem = update = setdefault = _fail  # noqa: B950
+    __setitem__ = __delitem__ = clear = pop = popitem = update = setdefault = _fail
     del _fail
 
 
