diff --git a/sklearn/tree/_export.py b/sklearn/tree/_export.py
--- a/sklearn/tree/_export.py
+++ b/sklearn/tree/_export.py
@@ -308,6 +308,7 @@ def node_to_str(self, tree, node_id, criterion):
             # Always write node decision criteria, except for leaves
             if self.feature_names is not None:
                 feature = self.feature_names[tree.feature[node_id]]
+                feature = self.str_escape(feature)
             else:
                 feature = "x%s%s%s" % (
                     characters[1],
@@ -383,6 +384,7 @@ def node_to_str(self, tree, node_id, criterion):
                 node_string += "class = "
             if self.class_names is not True:
                 class_name = self.class_names[np.argmax(value)]
+                class_name = self.str_escape(class_name)
             else:
                 class_name = "y%s%s%s" % (
                     characters[1],
@@ -397,6 +399,9 @@ def node_to_str(self, tree, node_id, criterion):
 
         return node_string + characters[5]
 
+    def str_escape(self, string):
+        return string
+
 
 class _DOTTreeExporter(_BaseTreeExporter):
     def __init__(
@@ -571,6 +576,10 @@ def recurse(self, tree, node_id, criterion, parent=None, depth=0):
                 # Add edge to parent
                 self.out_file.write("%d -> %d ;\n" % (parent, node_id))
 
+    def str_escape(self, string):
+        # override default escaping for graphviz
+        return string.replace('"', r"\"")
+
 
 class _MPLTreeExporter(_BaseTreeExporter):
     def __init__(
