diff --git a/keras/src/layers/convolutional/base_conv_transpose.py b/keras/src/layers/convolutional/base_conv_transpose.py
--- a/keras/src/layers/convolutional/base_conv_transpose.py
+++ b/keras/src/layers/convolutional/base_conv_transpose.py
@@ -151,6 +151,19 @@ def __init__(
                 f"dilation_rate={self.dilation_rate}"
             )
 
+        if self.output_padding is not None:
+            for i, (op, s) in enumerate(zip(self.output_padding, self.strides)):
+                if op >= s:
+                    raise ValueError(
+                        "Invalid `output_padding` argument. "
+                        "Each value in `output_padding` must be strictly "
+                        "less than the corresponding `strides` value.\n"
+                        f"At index {i}, `output_padding` is {op} and `strides` "
+                        f"is {s}.\n"
+                        f"Received: output_padding={self.output_padding}, "
+                        f"strides={self.strides}."
+                    )
+
     def build(self, input_shape):
         if self.data_format == "channels_last":
             channel_axis = -1
