diff --git a/custom_components/opnsense/pyopnsense/__init__.py b/custom_components/opnsense/pyopnsense/__init__.py
index 5ed6506f..f562a57f 100644
--- a/custom_components/opnsense/pyopnsense/__init__.py
+++ b/custom_components/opnsense/pyopnsense/__init__.py
@@ -1272,6 +1272,9 @@ async def _get_telemetry_system(self) -> MutableMapping[str, Any]:
         time_info = await self._safe_dict_post("/api/diagnostics/system/systemTime")
         # _LOGGER.debug(f"[get_telemetry_system] time_info: {time_info}")
         system: MutableMapping[str, Any] = {}
+
+        systemtime: datetime = datetime.strptime(time_info["datetime"], "%a %b %d %H:%M:%S %Z %Y")
+
         pattern = re.compile(r"^(?:(\d+)\s+days?,\s+)?(\d{2}):(\d{2}):(\d{2})$")
         match = pattern.match(time_info.get("uptime", ""))
         if match:
@@ -1280,9 +1283,19 @@ async def _get_telemetry_system(self) -> MutableMapping[str, Any]:
             hours = OPNsenseClient._try_to_int(hours_str, 0) or 0
             minutes = OPNsenseClient._try_to_int(minutes_str, 0) or 0
             seconds = OPNsenseClient._try_to_int(seconds_str, 0) or 0
-            system["uptime"] = days * 86400 + hours * 3600 + minutes * 60 + seconds
 
-            boottime: datetime = datetime.now() - timedelta(seconds=system["uptime"])
+            uptime = days * 86400 + hours * 3600 + minutes * 60 + seconds
+
+        if "boottime" in time_info:
+            boottime: datetime = datetime.strptime(time_info["boottime"], "%a %b %d %H:%M:%S %Z %Y")
+            system["boottime"] = boottime.timestamp()
+            if match:
+                system["uptime"] = uptime
+            else:
+                system["uptime"] = int((systemtime - boottime).total_seconds())
+        elif match:
+            system["uptime"] = uptime
+            boottime = systemtime - timedelta(seconds=system["uptime"])
             system["boottime"] = boottime.timestamp()
         else:
             _LOGGER.warning("Invalid uptime format")
