diff --git a/sklearn/svm/_base.py b/sklearn/svm/_base.py
--- a/sklearn/svm/_base.py
+++ b/sklearn/svm/_base.py
@@ -616,6 +616,13 @@ def _validate_for_predict(self, X):
                     "the number of samples at training time"
                     % (X.shape[1], self.shape_fit_[0])
                 )
+        # Fixes https://nvd.nist.gov/vuln/detail/CVE-2020-28975
+        # Check that _n_support is consistent with support_vectors
+        sv = self.support_vectors_
+        if not self._sparse and sv.size > 0 and self.n_support_.sum() != sv.shape[0]:
+            raise ValueError(
+                f"The internal representation of {self.__class__.__name__} was altered"
+            )
         return X
 
     @property
